{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "defd54a4-e796-4cf2-9cd6-80fb2511a6f7",
   "metadata": {},
   "source": [
    "# 02_231105_TutorialPRS\n",
    "\n",
    "Polygenic Risk Scoreの計算実習をしてみましょう\n",
    "\n",
    "## 使用データの展開"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "86aab064-4d84-467f-af87-1fb5504a2875",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'GWA_tutorial'...\n",
      "remote: Enumerating objects: 315, done.\u001b[K\n",
      "remote: Counting objects: 100% (19/19), done.\u001b[K\n",
      "remote: Compressing objects: 100% (17/17), done.\u001b[K\n",
      "remote: Total 315 (delta 7), reused 12 (delta 2), pack-reused 296\u001b[K\n",
      "Receiving objects: 100% (315/315), 31.71 MiB | 25.71 MiB/s, done.\n",
      "Resolving deltas: 100% (121/121), done.\n"
     ]
    }
   ],
   "source": [
    "git clone https://github.com/MareesAT/GWA_tutorial.git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2ca4f684-5145-49ee-bccd-9329124f9019",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "02_231105_TutorialPRS.ipynb  \u001b[0m\u001b[01;34mGWA_tutorial\u001b[0m  README.md\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3e72074e-8760-4869-b2f0-dc4955e09415",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;31m1_QC_GWAS.zip\u001b[0m                    \u001b[01;31m3_Association_GWAS.zip\u001b[0m  README.md\n",
      "\u001b[01;31m2_Population_stratification.zip\u001b[0m  4_PRS.doc\n"
     ]
    }
   ],
   "source": [
    "ls GWA_tutorial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2df51214-cf6f-4adb-9d9e-226fc0fb51a5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;31m1_QC_GWAS.zip\u001b[0m                    \u001b[01;31m3_Association_GWAS.zip\u001b[0m  README.md\n",
      "\u001b[01;31m2_Population_stratification.zip\u001b[0m  4_PRS.doc\n"
     ]
    }
   ],
   "source": [
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial\n",
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5ed35dba-2432-4457-a5f7-c0c389da94ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Archive:  1_QC_GWAS.zip\n",
      "  inflating: 1_QC_GWAS/1_Main_script_QC_GWAS.txt  \n",
      "  inflating: 1_QC_GWAS/check_heterozygosity_rate.R  \n",
      "  inflating: 1_QC_GWAS/gender_check.R  \n",
      "  inflating: 1_QC_GWAS/HapMap_3_r3_1.bed  \n",
      "  inflating: 1_QC_GWAS/HapMap_3_r3_1.bim  \n",
      "  inflating: 1_QC_GWAS/HapMap_3_r3_1.fam  \n",
      "  inflating: 1_QC_GWAS/heterozygosity_outliers_list.R  \n",
      "  inflating: 1_QC_GWAS/hist_miss.R   \n",
      "  inflating: 1_QC_GWAS/hwe.R         \n",
      "  inflating: 1_QC_GWAS/inversion.txt  \n",
      "  inflating: 1_QC_GWAS/MAF_check.R   \n",
      "  inflating: 1_QC_GWAS/pops_HapMap_3_r3  \n",
      "  inflating: 1_QC_GWAS/Relatedness.R  \n",
      "Archive:  2_Population_stratification.zip\n",
      "  inflating: 2_Population_stratification/1_Main_script_QC_GWAS.txt  \n",
      "  inflating: 2_Population_stratification/2_Main_script_MDS.txt  \n",
      "  inflating: 2_Population_stratification/MDS_merged.R  \n",
      "Archive:  3_Association_GWAS.zip\n",
      "  inflating: 3_Association_GWAS/3_Main_script_association_GWAS.txt  \n",
      "  inflating: 3_Association_GWAS/Manhattan_plot.R  \n",
      "  inflating: 3_Association_GWAS/QQ_plot.R  \n"
     ]
    }
   ],
   "source": [
    "unzip 1_QC_GWAS.zip\n",
    "unzip 2_Population_stratification.zip\n",
    "unzip 3_Association_GWAS.zip"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9321c87-eb0a-48a0-b4e7-f65b3fd5fa67",
   "metadata": {},
   "source": [
    "## 1.GWASに必要なQC\n",
    "\n",
    "まずディレクトリに移動します。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9bcf8a9d-03d7-44df-9c0c-d3d341f57169",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1_Main_script_QC_GWAS.txt    HapMap_3_r3_1.fam               MAF_check.R\n",
      "check_heterozygosity_rate.R  heterozygosity_outliers_list.R  pops_HapMap_3_r3\n",
      "gender_check.R               hist_miss.R                     Relatedness.R\n",
      "HapMap_3_r3_1.bed            hwe.R\n",
      "HapMap_3_r3_1.bim            inversion.txt\n"
     ]
    }
   ],
   "source": [
    "cd 1_QC_GWAS\n",
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1715073-9116-4238-b9f8-af63c3b93aab",
   "metadata": {},
   "source": [
    "これからやるスクリプトはこんな感じ。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4ce62670-f0bb-481f-a517-055baacd4867",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################ Explanation of the main script ##########################\n",
      "\n",
      "# This tutorial uses freely available HapMap data: hapmap3_r3_b36_fwd.consensus.qc. We simulated a binary outcome measure (i.e., a binary phenotypic trait) and added this to the dataset. The outcome measure was only simulated for the founders in the HapMap data. This data set will be referred to as HapMap_3_r3_1. \n",
      "# The HapMap data, without our simulated outcome measure, can also be obtained from http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/2010-05_phaseIII/plink_format/ \n",
      "# It is essential for the execution of the tutorial that that all scripts belonging to this tutorial are in the same directory on your UNIX workstation.\n",
      "# Many scripts include comments which explain how these scripts work. Note, in order to complete the tutorial it is essential to execute all commands in this tutorial.\n",
      "# This script can also be used for your own data analysis, to use it as such, replace the name of the HapMap file with the name of your own data file. \n",
      "# Furthermore, this script is based on a binary outcome measure, and is therefore not applicable for quantitative outcome measures (this would require some adaptations)\n",
      "# Note, most GWAS studies are performed on an ethnic homogenous population, in which population outliers are removed. The HapMap data, used for this tutorial, contains multiple distinct ethnic groups, which makes it problematic for analysis.\n",
      "# Therefore, we have selected only the EUR individuals of the complete HapMap sample for the tutorials 1-3. This selection is already performed in the HapMap_3_r3_1 file from our GitHub page.\n",
      "# The Rscripts used in this tutorial are all executed from the Unix command line.\n",
      "# Therefore, this tutorial and the other tutorials from our GitHub page, can be completed simply by copy-and-pasting all commands from the �main scripts� into the Unix terminal.\n",
      "# For a thorough theoretical explanation of all QC steps we refer to the article accompanying this tutorial entitled �A tutorial on conducting Genome-Wide-Association Studies: Quality control and statistical analysis� (https://www.ncbi.nlm.nih.gov/pubmed/29484742).\n",
      "\n",
      "\n",
      "##############################################################\n",
      "############### START ANALISIS ###############################\n",
      "##############################################################\n",
      "\n",
      "# Change directory to a folder on your UNIX device containing all files from �1_QC_GWAS.zip�. \n",
      "cd HOME/{user}/{path/folder containing your files}\n",
      "\n",
      "### Step 1 ### \n",
      "\n",
      "# Investigate missingness per individual and per SNP and make histograms.\n",
      "plink --bfile HapMap_3_r3_1 --missing    \n",
      "# output: plink.imiss and plink.lmiss, these files show respectively the proportion of missing SNPs per individual and the proportion of missing individuals per SNP.\n",
      "\n",
      "\n",
      "# Generate plots to visualize the missingness results.\n",
      "Rscript --no-save hist_miss.R\n",
      "\n",
      "# Delete SNPs and individuals with high levels of missingness, explanation of this and all following steps can be found in box 1 and table 1 of the article mentioned in the comments of this script.\n",
      "# The following two QC commands will not remove any SNPs or individuals. However, it is good practice to start the QC with these non-stringent thresholds.  \n",
      "# Delete SNPs with missingness >0.2.\n",
      "plink --bfile HapMap_3_r3_1 --geno 0.2 --make-bed --out HapMap_3_r3_2\n",
      "\n",
      "# Delete individuals with missingness >0.2.\n",
      "plink --bfile HapMap_3_r3_2 --mind 0.2 --make-bed --out HapMap_3_r3_3\n",
      "\n",
      "# Delete SNPs with missingness >0.02.\n",
      "plink --bfile HapMap_3_r3_3 --geno 0.02 --make-bed --out HapMap_3_r3_4\n",
      "\n",
      "# Delete individuals with missingness >0.02.\n",
      "plink --bfile HapMap_3_r3_4 --mind 0.02 --make-bed --out HapMap_3_r3_5\n",
      "\n",
      "###################################################################\n",
      "### Step2 ####\n",
      "\n",
      "# Check for sex discrepancy.\n",
      "# Subjects who were a priori determined as females must have a F value of <0.2, and subjects who were a priori determined as males must have a F value >0.8. This F value is based on the X chromosome inbreeding (homozygosity) estimate.\n",
      "# Subjects who do not fulfil these requirements are flagged \"PROBLEM\" by PLINK.\n",
      "\n",
      "plink --bfile HapMap_3_r3_5 --check-sex \n",
      "\n",
      "# Generate plots to visualize the sex-check results.\n",
      "Rscript --no-save gender_check.R\n",
      "# These checks indicate that there is one woman with a sex discrepancy, F value of 0.99. (When using other datasets often a few discrepancies will be found). \n",
      "\n",
      "# The following two scripts can be used to deal with individuals with a sex discrepancy.\n",
      "# Note, please use one of the two options below to generate the bfile hapmap_r23a_6, this file we will use in the next step of this tutorial.\n",
      "\n",
      "# 1) Delete individuals with sex discrepancy.\n",
      "grep \"PROBLEM\" plink.sexcheck| awk '{print$1,$2}'> sex_discrepancy.txt\n",
      "# This command generates a list of individuals with the status �PROBLEM�.\n",
      "plink --bfile HapMap_3_r3_5 --remove sex_discrepancy.txt --make-bed --out HapMap_3_r3_6 \n",
      "# This command removes the list of individuals with the status �PROBLEM�.\n",
      "\n",
      "# 2) impute-sex.\n",
      "#plink --bfile HapMap_3_r3_5 --impute-sex --make-bed --out HapMap_3_r3_6\n",
      "# This imputes the sex based on the genotype information into your data set.\n",
      "\n",
      "###################################################\n",
      "### Step 3 ### \n",
      "\n",
      "# Generate a bfile with autosomal SNPs only and delete SNPs with a low minor allele frequency (MAF).\n",
      "\n",
      "# Select autosomal SNPs only (i.e., from chromosomes 1 to 22).\n",
      "awk '{ if ($1 >= 1 && $1 <= 22) print $2 }' HapMap_3_r3_6.bim > snp_1_22.txt\n",
      "plink --bfile HapMap_3_r3_6 --extract snp_1_22.txt --make-bed --out HapMap_3_r3_7\n",
      "\n",
      "# Generate a plot of the MAF distribution.\n",
      "plink --bfile HapMap_3_r3_7 --freq --out MAF_check\n",
      "Rscript --no-save MAF_check.R\n",
      "\n",
      "# Remove SNPs with a low MAF frequency.\n",
      "plink --bfile HapMap_3_r3_7 --maf 0.05 --make-bed --out HapMap_3_r3_8\n",
      "# 1073226 SNPs are left\n",
      "# A conventional MAF threshold for a regular GWAS is between 0.01 or 0.05, depending on sample size.\n",
      "\n",
      "\n",
      "####################################################\n",
      "### Step 4 ###\n",
      "\n",
      "# Delete SNPs which are not in Hardy-Weinberg equilibrium (HWE).\n",
      "# Check the distribution of HWE p-values of all SNPs.\n",
      "\n",
      "plink --bfile HapMap_3_r3_8 --hardy\n",
      "# Selecting SNPs with HWE p-value below 0.00001, required for one of the two plot generated by the next Rscript, allows to zoom in on strongly deviating SNPs. \n",
      "awk '{ if ($9 <0.00001) print $0 }' plink.hwe>plinkzoomhwe.hwe\n",
      "Rscript --no-save hwe.R\n",
      "\n",
      "# By default the --hwe option in plink only filters for controls.\n",
      "# Therefore, we use two steps, first we use a stringent HWE threshold for controls, followed by a less stringent threshold for the case data.\n",
      "plink --bfile HapMap_3_r3_8 --hwe 1e-6 --make-bed --out HapMap_hwe_filter_step1\n",
      "\n",
      "# The HWE threshold for the cases filters out only SNPs which deviate extremely from HWE. \n",
      "# This second HWE step only focusses on cases because in the controls all SNPs with a HWE p-value < hwe 1e-6 were already removed\n",
      "plink --bfile HapMap_hwe_filter_step1 --hwe 1e-10 --hwe-all --make-bed --out HapMap_3_r3_9\n",
      "\n",
      "# Theoretical background for this step is given in our accompanying article: https://www.ncbi.nlm.nih.gov/pubmed/29484742 .\n",
      "\n",
      "############################################################\n",
      "### step 5 ###\n",
      "\n",
      "# Generate a plot of the distribution of the heterozygosity rate of your subjects.\n",
      "# And remove individuals with a heterozygosity rate deviating more than 3 sd from the mean.\n",
      "\n",
      "# Checks for heterozygosity are performed on a set of SNPs which are not highly correlated.\n",
      "# Therefore, to generate a list of non-(highly)correlated SNPs, we exclude high inversion regions (inversion.txt [High LD regions]) and prune the SNPs using the command --indep-pairwise�.\n",
      "# The parameters �50 5 0.2� stand respectively for: the window size, the number of SNPs to shift the window at each step, and the multiple correlation coefficient for a SNP being regressed on all other SNPs simultaneously.\n",
      "\n",
      "plink --bfile HapMap_3_r3_9 --exclude inversion.txt --range --indep-pairwise 50 5 0.2 --out indepSNP\n",
      "# Note, don't delete the file indepSNP.prune.in, we will use this file in later steps of the tutorial.\n",
      "\n",
      "plink --bfile HapMap_3_r3_9 --extract indepSNP.prune.in --het --out R_check\n",
      "# This file contains your pruned data set.\n",
      "\n",
      "# Plot of the heterozygosity rate distribution\n",
      "Rscript --no-save check_heterozygosity_rate.R\n",
      "\n",
      "# The following code generates a list of individuals who deviate more than 3 standard deviations from the heterozygosity rate mean.\n",
      "# For data manipulation we recommend using UNIX. However, when performing statistical calculations R might be more convenient, hence the use of the Rscript for this step:\n",
      "Rscript --no-save heterozygosity_outliers_list.R\n",
      "\n",
      "# Output of the command above: fail-het-qc.txt .\n",
      "# When using our example data/the HapMap data this list contains 2 individuals (i.e., two individuals have a heterozygosity rate deviating more than 3 SD's from the mean).\n",
      "# Adapt this file to make it compatible for PLINK, by removing all quotation marks from the file and selecting only the first two columns.\n",
      "sed 's/\"// g' fail-het-qc.txt | awk '{print$1, $2}'> het_fail_ind.txt\n",
      "\n",
      "# Remove heterozygosity rate outliers.\n",
      "plink --bfile HapMap_3_r3_9 --remove het_fail_ind.txt --make-bed --out HapMap_3_r3_10\n",
      "\n",
      "\n",
      "############################################################\n",
      "### step 6 ###\n",
      "\n",
      "# It is essential to check datasets you analyse for cryptic relatedness.\n",
      "# Assuming a random population sample we are going to exclude all individuals above the pihat threshold of 0.2 in this tutorial.\n",
      "\n",
      "# Check for relationships between individuals with a pihat > 0.2.\n",
      "plink --bfile HapMap_3_r3_10 --extract indepSNP.prune.in --genome --min 0.2 --out pihat_min0.2\n",
      "\n",
      "# The HapMap dataset is known to contain parent-offspring relations. \n",
      "# The following commands will visualize specifically these parent-offspring relations, using the z values. \n",
      "awk '{ if ($8 >0.9) print $0 }' pihat_min0.2.genome>zoom_pihat.genome\n",
      "\n",
      "# Generate a plot to assess the type of relationship.\n",
      "Rscript --no-save Relatedness.R\n",
      "\n",
      "# The generated plots show a considerable amount of related individuals (explentation plot; PO = parent-offspring, UN = unrelated individuals) in the Hapmap data, this is expected since the dataset was constructed as such.\n",
      "# Normally, family based data should be analyzed using specific family based methods. In this tutorial, for demonstrative purposes, we treat the relatedness as cryptic relatedness in a random population sample.\n",
      "# In this tutorial, we aim to remove all 'relatedness' from our dataset.\n",
      "# To demonstrate that the majority of the relatedness was due to parent-offspring we only include founders (individuals without parents in the dataset).\n",
      "\n",
      "plink --bfile HapMap_3_r3_10 --filter-founders --make-bed --out HapMap_3_r3_11\n",
      "\n",
      "# Now we will look again for individuals with a pihat >0.2.\n",
      "plink --bfile HapMap_3_r3_11 --extract indepSNP.prune.in --genome --min 0.2 --out pihat_min0.2_in_founders\n",
      "# The file 'pihat_min0.2_in_founders.genome' shows that, after exclusion of all non-founders, only 1 individual pair with a pihat greater than 0.2 remains in the HapMap data.\n",
      "# This is likely to be a full sib or DZ twin pair based on the Z values. Noteworthy, they were not given the same family identity (FID) in the HapMap data.\n",
      "\n",
      "# For each pair of 'related' individuals with a pihat > 0.2, we recommend to remove the individual with the lowest call rate. \n",
      "plink --bfile HapMap_3_r3_11 --missing\n",
      "# Use an UNIX text editor (e.g., vi(m) ) to check which individual has the highest call rate in the 'related pair'. \n",
      "\n",
      "# Generate a list of FID and IID of the individual(s) with a Pihat above 0.2, to check who had the lower call rate of the pair.\n",
      "# In our dataset the individual 13291  NA07045 had the lower call rate.\n",
      "vi 0.2_low_call_rate_pihat.txt\n",
      "i \n",
      "13291  NA07045\n",
      "# Press esc on keyboard!\n",
      ":x\n",
      "# Press enter on keyboard\n",
      "# In case of multiple 'related' pairs, the list generated above can be extended using the same method as for our lone 'related' pair.\n",
      "\n",
      "# Delete the individuals with the lowest call rate in 'related' pairs with a pihat > 0.2 \n",
      "plink --bfile HapMap_3_r3_11 --remove 0.2_low_call_rate_pihat.txt --make-bed --out HapMap_3_r3_12\n",
      "\n",
      "################################################################################################################################\n",
      "\n",
      "# CONGRATULATIONS!! You've just succesfully completed the first tutorial! You are now able to conduct a proper genetic QC. \n",
      "\n",
      "# For the next tutorial, using the script: 2_Main_script_MDS.txt, you need the following files:\n",
      "# - The bfile HapMap_3_r3_12 (i.e., HapMap_3_r3_12.fam,HapMap_3_r3_12.bed, and HapMap_3_r3_12.bim\n",
      "# - indepSNP.prune.in\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cat 1_Main_script_QC_GWAS.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35b08337-83ef-418d-842b-1d094f9591b1",
   "metadata": {},
   "source": [
    "主に使うのはplinkなので起動チェックしておく"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "f7f50342-a55a-41c5-afa8-fce3debd0151",
   "metadata": {},
   "outputs": [],
   "source": [
    "#plinkのアドレスのチェック\n",
    "plink=~/local/plink/plink"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c5759d93-8995-4311-b8de-766def4949f4",
   "metadata": {},
   "source": [
    "### Step 1：欠損データのチェック\n",
    "\n",
    "Investigate missingness per individual and per SNP and make histograms.*個体ごと、SNPごとの欠損度を調査し、ヒストグラムを作成する。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_1 --missing    \n",
    "\n",
    "output: plink.imiss and plink.lmiss, these files show respectively the proportion of missing SNPs per individual and the proportion of missing individuals per SNP.*これらのファイルはそれぞれ、個体ごとの欠損SNPの割合と、SNPごとの欠損個体の割合を示している。*　\n",
    "\n",
    "Generate plots to visualize the missingness results.*missingness（欠損）の結果を可視化するプロットを作成する。*　　\n",
    "\n",
    "    Rscript --no-save hist_miss.R\n",
    "\n",
    "Delete SNPs and individuals with high levels of missingness, explanation of this and all following steps can be found in box 1 and table 1 of the article mentioned in the comments of this script.*欠損の多いSNPと個体を削除する。このステップとそれに続くすべてのステップの説明は、このスクリプトのコメントに記載されている論文のbox 1とtable 1にあります。*\n",
    "\n",
    "The following two QC commands will not remove any SNPs or individuals. However, it is good practice to start the QC with these non-stringent thresholds.*以下の2つのQCコマンドでは、SNPや個体は削除されない。しかし、これらの厳しくないしきい値でQCを開始するのは良い習慣である。*\n",
    "\n",
    "Delete SNPs with missingness >0.2.\n",
    "\n",
    "    plink --bfile HapMap_3_r3_1 --geno 0.2 --make-bed --out HapMap_3_r3_2\n",
    "\n",
    "Delete individuals with missingness >0.2.\n",
    "\n",
    "    plink --bfile HapMap_3_r3_2 --mind 0.2 --make-bed --out HapMap_3_r3_3\n",
    "\n",
    "Delete SNPs with missingness >0.02.\n",
    "\n",
    "    plink --bfile HapMap_3_r3_3 --geno 0.02 --make-bed --out HapMap_3_r3_4\n",
    "\n",
    "Delete individuals with missingness >0.02.\n",
    "\n",
    "    plink --bfile HapMap_3_r3_4 --mind 0.02 --make-bed --out HapMap_3_r3_5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "cff7ff1e-d141-47c6-9ee4-b56f4c0bd7ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plink.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_1\n",
      "  --missing\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1457897 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 225 het. haploid genotypes present (see plink.hh ); many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.997378.\n",
      "--missing: Sample missing data report written to plink.imiss, and variant-based\n",
      "missing data report written to plink.lmiss.\n"
     ]
    }
   ],
   "source": [
    "#欠損値調査\n",
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/1_QC_GWAS\n",
    "# Investigate missingness per individual and per SNP and make histograms.\n",
    "$plink --bfile HapMap_3_r3_1 --missing    \n",
    "# output: plink.imiss and plink.lmiss, these files show respectively the proportion of missing SNPs per individual and the proportion of missing individuals per SNP."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "6be9a08e-692c-4df7-b80c-9ec3b30c5894",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "indmiss<-read.table(file=\"plink.imiss\", header=TRUE)\n",
      "snpmiss<-read.table(file=\"plink.lmiss\", header=TRUE)\n",
      "# read data into R \n",
      "\n",
      "pdf(\"histimiss.pdf\") #indicates pdf format and gives title to file\n",
      "hist(indmiss[,6],main=\"Histogram individual missingness\") #selects column 6, names header of file\n",
      "\n",
      "pdf(\"histlmiss.pdf\") \n",
      "hist(snpmiss[,5],main=\"Histogram SNP missingness\")  \n",
      "dev.off() # shuts down the current device\n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n"
     ]
    }
   ],
   "source": [
    "#欠損値チェックの結果出力\n",
    "# Generate plots to visualize the missingness results.\n",
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/1_QC_GWAS\n",
    "cat hist_miss.R\n",
    "#Rscript --no-save hist_miss.R #dev.off()はまめにした方が良い\n",
    "\n",
    "echo '#change file name and plot' > hist_miss_HM.R\n",
    "echo 'indmiss<-read.table(file=\"plink.imiss\", header=TRUE)' >> hist_miss_HM.R\n",
    "echo 'snpmiss<-read.table(file=\"plink.lmiss\", header=TRUE)' >> hist_miss_HM.R\n",
    "echo '# read data into R' >> hist_miss_HM.R\n",
    "echo '' >> hist_miss_HM.R\n",
    "echo 'pdf(\"histimiss.pdf\") #indicates pdf format and gives title to file' >> hist_miss_HM.R\n",
    "echo 'hist(indmiss[,6],main=\"Histogram individual missingness\") #selects column 6, names header of file' >> hist_miss_HM.R\n",
    "echo 'dev.off() # shuts down the current device' >> hist_miss_HM.R\n",
    "echo '' >> hist_miss_HM.R\n",
    "echo 'pdf(\"histlmiss.pdf\") #indicates pdf format and gives title to file' >> hist_miss_HM.R\n",
    "echo 'hist(snpmiss[,5],main=\"Histogram SNP missingness\")' >> hist_miss_HM.R\n",
    "echo 'dev.off() # shuts down the current device' >> hist_miss_HM.R\n",
    "Rscript --no-save hist_miss_HM.R"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a113f8c0-d327-4b7a-8663-108fc8cb1fd4",
   "metadata": {},
   "source": [
    "【ここから少々話がそれる】"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d5b1010c-f3fd-47b2-a10f-22b657bddaf5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running `brew update --auto-update`...\n",
      "Installing from the API is now the default behaviour!\n",
      "You can save space and time by running:\n",
      "  brew untap homebrew/core\n",
      "\u001b[34m==>\u001b[0m \u001b[1mDownloading https://ghcr.io/v2/homebrew/core/libsixel/manifests/1.10.3_1\u001b[0m\n",
      "#=#=#                                                                          ######################################################################### 100.0%\n",
      "\u001b[32m==>\u001b[0m \u001b[1mFetching dependencies for libsixel: \u001b[32mjpeg-turbo\u001b[39m and \u001b[32mlibpng\u001b[39m\u001b[0m\n",
      "\u001b[34m==>\u001b[0m \u001b[1mDownloading https://ghcr.io/v2/homebrew/core/jpeg-turbo/manifests/3.0.0\u001b[0m\n",
      "######################################################################### 100.0%####################################################################      94.7%\n",
      "\u001b[32m==>\u001b[0m \u001b[1mFetching \u001b[32mjpeg-turbo\u001b[39m\u001b[0m\n",
      "\u001b[34m==>\u001b[0m \u001b[1mDownloading https://ghcr.io/v2/homebrew/core/jpeg-turbo/blobs/sha256:44943f3\u001b[0m\n",
      "######################################################################### 100.0%                                                                           0.1%\n",
      "\u001b[34m==>\u001b[0m \u001b[1mDownloading https://ghcr.io/v2/homebrew/core/libpng/manifests/1.6.40\u001b[0m\n",
      "#=#=#                                                                          ######################################################################### 100.0%\n",
      "\u001b[32m==>\u001b[0m \u001b[1mFetching \u001b[32mlibpng\u001b[39m\u001b[0m\n",
      "\u001b[34m==>\u001b[0m \u001b[1mDownloading https://ghcr.io/v2/homebrew/core/libpng/blobs/sha256:71da4cbafaa\u001b[0m\n",
      "######################################################################### 100.0%                                                                           1.7%\n",
      "\u001b[32m==>\u001b[0m \u001b[1mFetching \u001b[32mlibsixel\u001b[39m\u001b[0m\n",
      "\u001b[34m==>\u001b[0m \u001b[1mDownloading https://ghcr.io/v2/homebrew/core/libsixel/blobs/sha256:f496efefb\u001b[0m\n",
      "######################################################################### 100.0%#                                                                          2.9%\n",
      "\u001b[32m==>\u001b[0m \u001b[1mInstalling dependencies for libsixel: \u001b[32mjpeg-turbo\u001b[39m and \u001b[32mlibpng\u001b[39m\u001b[0m\n",
      "\u001b[32m==>\u001b[0m \u001b[1mInstalling libsixel dependency: \u001b[32mjpeg-turbo\u001b[39m\u001b[0m\n",
      "\u001b[34m==>\u001b[0m \u001b[1mDownloading https://ghcr.io/v2/homebrew/core/jpeg-turbo/manifests/3.0.0\u001b[0m\n",
      "Already downloaded: /home/hmiwa/.cache/Homebrew/downloads/325e745bcb7840d6a83df7bfcafaa4dbc30b29140f252fb82b7714493df29d6d--jpeg-turbo-3.0.0.bottle_manifest.json\n",
      "\u001b[34m==>\u001b[0m \u001b[1mPouring jpeg-turbo--3.0.0.x86_64_linux.bottle.tar.gz\u001b[0m\n",
      "🍺  /home/hmiwa/homebrew/Cellar/jpeg-turbo/3.0.0: 44 files, 5.2MB\n",
      "\u001b[32m==>\u001b[0m \u001b[1mInstalling libsixel dependency: \u001b[32mlibpng\u001b[39m\u001b[0m\n",
      "\u001b[34m==>\u001b[0m \u001b[1mDownloading https://ghcr.io/v2/homebrew/core/libpng/manifests/1.6.40\u001b[0m\n",
      "Already downloaded: /home/hmiwa/.cache/Homebrew/downloads/d2caacc64e32be045b902fa78d2bb4aea7fba8f088cef85cd34b1beab15b755a--libpng-1.6.40.bottle_manifest.json\n",
      "\u001b[34m==>\u001b[0m \u001b[1mPouring libpng--1.6.40.x86_64_linux.bottle.tar.gz\u001b[0m\n",
      "🍺  /home/hmiwa/homebrew/Cellar/libpng/1.6.40: 28 files, 1.5MB\n",
      "\u001b[32m==>\u001b[0m \u001b[1mInstalling \u001b[32mlibsixel\u001b[39m\u001b[0m\n",
      "\u001b[34m==>\u001b[0m \u001b[1mPouring libsixel--1.10.3_1.x86_64_linux.bottle.tar.gz\u001b[0m\n",
      "🍺  /home/hmiwa/homebrew/Cellar/libsixel/1.10.3_1: 19 files, 1.1MB\n",
      "\u001b[34m==>\u001b[0m \u001b[1mRunning `brew cleanup libsixel`...\u001b[0m\n",
      "Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.\n",
      "Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).\n"
     ]
    }
   ],
   "source": [
    "#画像表示のためのアプリケーションをインストールしてみる\n",
    "cd \n",
    "brew=~/homebrew/bin/brew\n",
    "$brew install libsixel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e0aca6fb-f532-4cf7-a6c8-ba05bbcf5ca3",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/1_QC_GWAS\n",
    "img2sixel=~/homebrew/bin/img2sixel\n",
    "pdftoppm -png histlmiss.pdf histlmiss\n",
    "#$img2sixel histlmiss-1.png #環境には合わなかったようだ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "019c16cf-4403-4789-a280-ed34af184c31",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Rのパッケージでもやってみようかと思ったけど難しそうだった\n",
    "#Rscript 'install.packages(\"imager\")'\n",
    "#Rscript -e 'library(\"imager\");img = load.image(\"myimage.png\");plot(img)'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8ee837ad-9f2b-4ca3-813e-cc716b02093a",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/1_QC_GWAS\n",
    "img2sixel=~/homebrew/bin/img2sixel\n",
    "#eog histlmiss-1.png #eogも難しそう"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "86604414-6756-43d3-b64c-00ecbbf311a6",
   "metadata": {},
   "outputs": [],
   "source": [
    "pdftoppm -png histlmiss.pdf histlmiss\n",
    "pdftoppm -png histimiss.pdf histimiss\n",
    "histlmiss=`base64 histlmiss-1.png`\n",
    "histimiss=`base64 histimiss-1.png`\n",
    "echo $histlmiss　> histlmiss-1.txt\n",
    "echo $histimiss　> histimiss-1.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "13e4883a-6444-4459-8195-488c4ba5bb30",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic\n",
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/1_QC_GWAS\n",
    "cp histlmiss-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic\n",
    "cp histimiss-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "57df2e58-2d22-40c2-8abc-85e856a22b17",
   "metadata": {},
   "source": [
    "最初からマークダウンで処理すればよかったのでは【解決！】\n",
    "\n",
    "<img src=\"pic/histlmiss-1.png\" width=\"45%\">　<img src=\"pic/histimiss-1.png\" width=\"45%\">"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f16cd6e9-76b5-4334-80ae-a6b3c4be5f00",
   "metadata": {},
   "source": [
    "【画像表示問題は解決したのでここから本題に戻ります。】"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fc25c3e1-a013-4f23-a073-bb265e3391a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_2.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_1\n",
      "  --geno 0.2\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_2\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1457897 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 225 het. haploid genotypes present (see HapMap_3_r3_2.hh ); many\n",
      "commands treat these as missing.\n",
      "Total genotyping rate is 0.997378.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "1457897 variants and 165 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (53 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_2.bed + HapMap_3_r3_2.bim + HapMap_3_r3_2.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_3.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_2\n",
      "  --make-bed\n",
      "  --mind 0.2\n",
      "  --out HapMap_3_r3_3\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1457897 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 225 het. haploid genotypes present (see HapMap_3_r3_3.hh ); many\n",
      "commands treat these as missing.\n",
      "Total genotyping rate is 0.997378.\n",
      "1457897 variants and 165 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (53 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_3.bed + HapMap_3_r3_3.bim + HapMap_3_r3_3.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_4.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_3\n",
      "  --geno 0.02\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_4\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1457897 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 225 het. haploid genotypes present (see HapMap_3_r3_4.hh ); many\n",
      "commands treat these as missing.\n",
      "Total genotyping rate is 0.997378.\n",
      "27454 variants removed due to missing genotype data (--geno).\n",
      "1430443 variants and 165 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (53 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_4.bed + HapMap_3_r3_4.bim + HapMap_3_r3_4.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_5.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_4\n",
      "  --make-bed\n",
      "  --mind 0.02\n",
      "  --out HapMap_3_r3_5\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1430443 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 179 het. haploid genotypes present (see HapMap_3_r3_5.hh ); many\n",
      "commands treat these as missing.\n",
      "Total genotyping rate is 0.997899.\n",
      "1430443 variants and 165 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (53 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_5.bed + HapMap_3_r3_5.bim + HapMap_3_r3_5.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# QC処理\n",
    "# Delete SNPs and individuals with high levels of missingness, explanation of this and all following steps can be found in box 1 and table 1 of the article mentioned in the comments of this script.\n",
    "# The following two QC commands will not remove any SNPs or individuals. However, it is good practice to start the QC with these non-stringent thresholds.  \n",
    "# Delete SNPs with missingness >0.2.\n",
    "$plink --bfile HapMap_3_r3_1 --geno 0.2 --make-bed --out HapMap_3_r3_2\n",
    "\n",
    "# Delete individuals with missingness >0.2.\n",
    "$plink --bfile HapMap_3_r3_2 --mind 0.2 --make-bed --out HapMap_3_r3_3\n",
    "\n",
    "# Delete SNPs with missingness >0.02.\n",
    "$plink --bfile HapMap_3_r3_3 --geno 0.02 --make-bed --out HapMap_3_r3_4\n",
    "\n",
    "# Delete individuals with missingness >0.02.\n",
    "$plink --bfile HapMap_3_r3_4 --mind 0.02 --make-bed --out HapMap_3_r3_5"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9e5b53e-dac4-4f27-97dc-9a8e00e0c362",
   "metadata": {},
   "source": [
    "### Step 2：性別チェック\n",
    "\n",
    "Check for sex discrepancy.*性別の不一致をチェックする。*\n",
    "\n",
    "Subjects who were a priori determined as females must have a F value of <0.2, and subjects who were a priori determined as males must have a F value >0.8. This F value is based on the X chromosome inbreeding (homozygosity) estimate.*先験的に女性と判定された被験者はF値が<0.2でなければならず、先験的に男性と判定された被験者はF値が>0.8でなければならない。このF値はX染色体の近親交配（ホモ接合性）の推定値に基づいている。*\n",
    "\n",
    "Subjects who do not fulfil these requirements are flagged \"PROBLEM\" by PLINK.*これらの条件を満たさない被験者は、PLINKによって \"PROBLEM \"のフラグが立てられます。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_5 --check-sex \n",
    "\n",
    "Generate plots to visualize the sex-check results.*性チェック結果を可視化するプロットを作成します。*\n",
    "\n",
    "    Rscript --no-save gender_check.R\n",
    "\n",
    "These checks indicate that there is one woman with a sex discrepancy, F value of 0.99. (When using other datasets often a few discrepancies will be found). *これらのチェックは、F値が0.99の雌雄不一致の女性が1人いることを示している。(他のデータセットを使用する場合、しばしば数個の不一致が見つかる)。*\n",
    "\n",
    "The following two scripts can be used to deal with individuals with a sex discrepancy.*以下の2つのスクリプトは、性別の不一致がある個人を処理するのに使える。*\n",
    "\n",
    "Note, please use one of the two options below to generate the bfile hapmap_r23a_6, this file we will use in the next step of this tutorial.*このチュートリアルの次のステップで使用するファイル haap_r23a_6 を生成するために、以下の2つのオプションのどちらかを使用してください。*\n",
    "\n",
    "1. Delete individuals with sex discrepancy.\n",
    "\n",
    "        grep \"PROBLEM\" plink.sexcheck| awk '{print$1,$2}'> sex_discrepancy.txt\n",
    "\n",
    "    This command generates a list of individuals with the status �PROBLEM�.\n",
    "\n",
    "        plink --bfile HapMap_3_r3_5 --remove sex_discrepancy.txt --make-bed --out HapMap_3_r3_6 \n",
    "\n",
    "    This command removes the list of individuals with the status �PROBLEM�.\n",
    "\n",
    "2. impute-sex.\n",
    "\n",
    "        plink --bfile HapMap_3_r3_5 --impute-sex --make-bed --out HapMap_3_r3_6\n",
    "   \n",
    "    This imputes the sex based on the genotype information into your data set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e57ba06f-3b14-43b2-9791-b794866ec5b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plink.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_5\n",
      "  --check-sex\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1430443 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 179 het. haploid genotypes present (see plink.hh ); many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.997899.\n",
      "1430443 variants and 165 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (53 phenotypes\n",
      "are missing.)\n",
      "--check-sex: 23424 Xchr and 0 Ychr variant(s) scanned, 1 problem detected.\n",
      "Report written to plink.sexcheck .\n"
     ]
    }
   ],
   "source": [
    "# サンプルの性別チェック\n",
    "# Check for sex discrepancy.\n",
    "# Subjects who were a priori determined as females must have a F value of <0.2, and subjects who were a priori determined as males must have a F value >0.8. This F value is based on the X chromosome inbreeding (homozygosity) estimate.\n",
    "# Subjects who do not fulfil these requirements are flagged \"PROBLEM\" by PLINK.\n",
    "$plink --bfile HapMap_3_r3_5 --check-sex "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "c501ad47-3e86-4d38-8602-59e375f56582",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gender <- read.table(\"plink.sexcheck\", header=T,as.is=T)\n",
      "\n",
      "pdf(\"Gender_check.pdf\")\n",
      "hist(gender[,6],main=\"Gender\", xlab=\"F\")\n",
      "dev.off()\n",
      "\n",
      "pdf(\"Men_check.pdf\")\n",
      "male=subset(gender, gender$PEDSEX==1)\n",
      "hist(male[,6],main=\"Men\",xlab=\"F\")\n",
      "dev.off()\n",
      "\n",
      "pdf(\"Women_check.pdf\")\n",
      "female=subset(gender, gender$PEDSEX==2)\n",
      "hist(female[,6],main=\"Women\",xlab=\"F\")\n",
      "dev.off()\n",
      "\n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n"
     ]
    }
   ],
   "source": [
    "#性別チェック結果の表示\n",
    "# Generate plots to visualize the sex-check results.\n",
    "cat gender_check.R\n",
    "Rscript --no-save gender_check.R\n",
    "# These checks indicate that there is one woman with a sex discrepancy, F value of 0.99. (When using other datasets often a few discrepancies will be found). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "de2d1354-08af-467a-86f4-f0259e4474ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "pdftoppm -png Gender_check.pdf Gender_check\n",
    "pdftoppm -png Men_check.pdf Men_check\n",
    "pdftoppm -png Women_check.pdf Women_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "e9685e8d-2b27-4c09-9e83-910a6c081383",
   "metadata": {},
   "outputs": [],
   "source": [
    "cp Gender_check-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic\n",
    "cp Men_check-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic\n",
    "cp Women_check-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d10cfbdf-42d3-4059-8b14-0500fb2e281d",
   "metadata": {},
   "source": [
    "<img src=\"pic/Gender_check-1.png\" width=\"33%\"><img src=\"pic/Men_check-1.png\" width=\"33%\"><img src=\"pic/Women_check-1.png\" width=\"33%\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "ba12b07a-0401-4031-b1cf-49b9238f0202",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_6.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_5\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_6\n",
      "  --remove sex_discrepancy.txt\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1430443 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "--remove: 164 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 179 het. haploid genotypes present (see HapMap_3_r3_6.hh ); many\n",
      "commands treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.99798.\n",
      "1430443 variants and 164 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (52 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_6.bed + HapMap_3_r3_6.bim + HapMap_3_r3_6.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# 問題のあるサンプルの除去／修正\n",
    "# The following two scripts can be used to deal with individuals with a sex discrepancy.\n",
    "# Note, please use one of the two options below to generate the bfile hapmap_r23a_6, this file we will use in the next step of this tutorial.\n",
    "\n",
    "# 1) Delete individuals with sex discrepancy.\n",
    "grep \"PROBLEM\" plink.sexcheck| awk '{print$1,$2}'> sex_discrepancy.txt\n",
    "# This command generates a list of individuals with the status �PROBLEM�.\n",
    "$plink --bfile HapMap_3_r3_5 --remove sex_discrepancy.txt --make-bed --out HapMap_3_r3_6 \n",
    "# This command removes the list of individuals with the status �PROBLEM�.\n",
    "\n",
    "# 2) impute-sex.\n",
    "#$plink --bfile HapMap_3_r3_5 --impute-sex --make-bed --out HapMap_3_r3_6\n",
    "# This imputes the sex based on the genotype information into your data set."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8a9a2e8c-0c0e-4299-93a5-7d0dbe62b49f",
   "metadata": {},
   "source": [
    "### Step 3：常染色体SNP選択・レアアリル削除\n",
    "\n",
    "Generate a bfile with autosomal SNPs only and delete SNPs with a low minor allele frequency (MAF).  *常染色体のSNPのみを含むbfileを作成し、マイナーアレル頻度（MAF）の低いSNPを削除する。*\n",
    "\n",
    "Select autosomal SNPs only (i.e., from chromosomes 1 to 22).  *常染色体のSNPのみ（つまり1番から22番までのSNP）を選択します。*\n",
    "\n",
    "    awk '{ if ($1 >= 1 && $1 <= 22) print $2 }' HapMap_3_r3_6.bim > snp_1_22.txt\n",
    "    plink --bfile HapMap_3_r3_6 --extract snp_1_22.txt --make-bed --out HapMap_3_r3_7\n",
    "\n",
    "Generate a plot of the MAF distribution.  *MAF分布のプロットを作成する。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_7 --freq --out MAF_check\n",
    "    Rscript --no-save MAF_check.R\n",
    "\n",
    "Remove SNPs with a low MAF frequency.*MAF頻度の低いSNPを削除する。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_7 --maf 0.05 --make-bed --out HapMap_3_r3_8\n",
    "\n",
    "1073226 SNPs are left  *1073226SNPが残っている*\n",
    "\n",
    "A conventional MAF threshold for a regular GWAS is between 0.01 or 0.05, depending on sample size.  *通常のGWASのMAF閾値は、サンプルサイズにもよるが、0.01か0.05の間である。*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "18690481-c54c-472f-9680-2f5cf4884e91",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_7.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_6\n",
      "  --extract snp_1_22.txt\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_7\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1430443 variants loaded from .bim file.\n",
      "164 people (80 males, 84 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "--extract: 1398544 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998052.\n",
      "1398544 variants and 164 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (52 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_7.bed + HapMap_3_r3_7.bim + HapMap_3_r3_7.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#常染色体SNP選択\n",
    "awk '{ if ($1 >= 1 && $1 <= 22) print $2 }' HapMap_3_r3_6.bim > snp_1_22.txt\n",
    "plink --bfile HapMap_3_r3_6 --extract snp_1_22.txt --make-bed --out HapMap_3_r3_7"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "a972b08b-a68b-4a6e-be8e-243db4ffbf06",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to MAF_check.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_7\n",
      "  --freq\n",
      "  --out MAF_check\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1398544 variants loaded from .bim file.\n",
      "164 people (80 males, 84 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998052.\n",
      "--freq: Allele frequencies (founders only) written to MAF_check.frq .\n",
      "maf_freq <- read.table(\"MAF_check.frq\", header =TRUE, as.is=T)\n",
      "pdf(\"MAF_distribution.pdf\")\n",
      "hist(maf_freq[,5],main = \"MAF distribution\", xlab = \"MAF\")\n",
      "dev.off()\n",
      "\n",
      "\n",
      "null device \n",
      "          1 \n"
     ]
    }
   ],
   "source": [
    "#MAF分布のプロットを作成する。\n",
    "plink --bfile HapMap_3_r3_7 --freq --out MAF_check\n",
    "cat MAF_check.R\n",
    "Rscript --no-save MAF_check.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "30ae344e-5016-4385-a1d5-149aa633fe48",
   "metadata": {},
   "outputs": [],
   "source": [
    "pdftoppm -png MAF_distribution.pdf MAF_distribution\n",
    "cp MAF_distribution-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b9e2c626-fb84-480c-9947-76e31a4318c2",
   "metadata": {},
   "source": [
    "<img src=\"pic/MAF_distribution-1.png\" width=\"50%\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "08825e30-24ae-47d5-8a63-45439cf5afcf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_8.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_7\n",
      "  --maf 0.05\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_8\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1398544 variants loaded from .bim file.\n",
      "164 people (80 males, 84 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998052.\n",
      "325318 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "1073226 variants and 164 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (52 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_8.bed + HapMap_3_r3_8.bim + HapMap_3_r3_8.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#MAF頻度の低いSNPを削除する。\n",
    "plink --bfile HapMap_3_r3_7 --maf 0.05 --make-bed --out HapMap_3_r3_8"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1443ffc9-28ee-4ff4-b571-970b5b01f661",
   "metadata": {},
   "source": [
    "### Step 4：ハーディーワインベルグ平衡(HWE)にないSNPを削除\n",
    "\n",
    "Delete SNPs which are not in Hardy-Weinberg equilibrium (HWE).  *ハーディーワインベルグ平衡(HWE)にないSNPを削除する。*\n",
    "\n",
    "Check the distribution of HWE p-values of all SNPs.  *すべてのSNPのHWE p値の分布をチェックする。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_8 --hardy\n",
    "\n",
    "Selecting SNPs with HWE p-value below 0.00001, required for one of the two plot generated by the next Rscript, allows to zoom in on strongly deviating SNPs.   *HWE p値が0.00001以下のSNPを選択すると強く乖離したSNPにズームインできる。（次のRscriptで生成される2つのプロットのうちの1つに必要）*\n",
    "\n",
    "    awk '{ if ($9 <0.00001) print $0 }' plink.hwe>plinkzoomhwe.hwe\n",
    "    Rscript --no-save hwe.R\n",
    "\n",
    "By default the --hwe option in plink only filters for controls.  *デフォルトでは、plinkの--hweオプションはコントロールだけをフィルターする。*\n",
    "\n",
    "Therefore, we use two steps, first we use a stringent HWE threshold for controls, followed by a less stringent threshold for the case data.  *そのため、2つのステップを使用する。まずコントロールに対して厳しいHWE閾値を使用し、次にケースデータに対してより厳しくない閾値を使用する。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_8 --hwe 1e-6 --make-bed --out HapMap_hwe_filter_step1\n",
    "\n",
    "The HWE threshold for the cases filters out only SNPs which deviate extremely from HWE.   *症例のHWE閾値は、HWEから極端に逸脱するSNPのみをフィルタリングする。*\n",
    "\n",
    "This second HWE step only focusses on cases because in the controls all SNPs with a HWE p-value < hwe 1e-6 were already removed. *対照群ではHWE p値 < hwe 1e-6のSNPはすでに除去されているので、この2番目のHWEステップは症例にのみ焦点を当てる。*\n",
    "\n",
    "    plink --bfile HapMap_hwe_filter_step1 --hwe 1e-10 --hwe-all --make-bed --out HapMap_3_r3_9\n",
    "\n",
    "Theoretical background for this step is given in our accompanying article: https://www.ncbi.nlm.nih.gov/pubmed/29484742 .  *このステップの理論的背景は、添付の記事*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "f02dc5cd-9e6f-44cb-af7e-5dbdc2e07255",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plink.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_8\n",
      "  --hardy\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "164 people (80 males, 84 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998039.\n",
      "--hardy: Writing Hardy-Weinberg report (founders only) to plink.hwe ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#すべてのSNPのHWE p値の分布をチェック\n",
    "plink --bfile HapMap_3_r3_8 --hardy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "c8d85823-7a0e-4e94-8fd1-53c5a5cef04a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hwe<-read.table (file=\"plink.hwe\", header=TRUE)\n",
      "pdf(\"histhwe.pdf\")\n",
      "hist(hwe[,9],main=\"Histogram HWE\")\n",
      "dev.off()\n",
      "\n",
      "hwe_zoom<-read.table (file=\"plinkzoomhwe.hwe\", header=TRUE)\n",
      "pdf(\"histhwe_below_theshold.pdf\")\n",
      "hist(hwe_zoom[,9],main=\"Histogram HWE: strongly deviating SNPs only\")\n",
      "dev.off()\n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n"
     ]
    }
   ],
   "source": [
    "#HWE p値が0.00001以下のSNPを選択すると強く乖離したSNPにズームインできる。\n",
    "awk '{ if ($9 <0.00001) print $0 }' plink.hwe > plinkzoomhwe.hwe\n",
    "cat hwe.R\n",
    "Rscript --no-save hwe.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "e76389f4-6473-488a-90ba-003264a1b71c",
   "metadata": {},
   "outputs": [],
   "source": [
    "filename=\"histhwe\"\n",
    "pdftoppm -png $filename.pdf $filename\n",
    "cp $filename-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "18e4a98e-9464-464f-824e-9cdf92a7b13b",
   "metadata": {},
   "outputs": [],
   "source": [
    "filename=\"histhwe_below_theshold\"\n",
    "pdftoppm -png $filename.pdf $filename\n",
    "cp $filename-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "46c6d904-6c7d-4511-a362-af807b332c92",
   "metadata": {},
   "source": [
    "<img src=\"pic/histhwe-1.png\" width=\"45%\"> <img src=\"pic/histhwe_below_theshold-1.png\" width=\"45%\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "7b6a1d89-37ef-447e-9365-c20778b3a54a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_hwe_filter_step1.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_8\n",
      "  --hwe 1e-6\n",
      "  --make-bed\n",
      "  --out HapMap_hwe_filter_step1\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "164 people (80 males, 84 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998039.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "1073226 variants and 164 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (52 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_hwe_filter_step1.bed + HapMap_hwe_filter_step1.bim +\n",
      "HapMap_hwe_filter_step1.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#デフォルトでは、plinkの--hweオプションはコントロールだけをフィルターする。\n",
    "#そのため、2つのステップを使用する。\n",
    "#まずコントロールに対して厳しいHWE閾値を使用し、次にケースデータに対してより厳しくない閾値を使用する。\n",
    "plink --bfile HapMap_3_r3_8 --hwe 1e-6 --make-bed --out HapMap_hwe_filter_step1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "369e0182-f00b-4f83-abbc-9f451d14c8d7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_9.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_hwe_filter_step1\n",
      "  --hwe 1e-10\n",
      "  --hwe-all\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_9\n",
      "\n",
      "Note: --hwe-all flag deprecated.  Use \"--hwe include-nonctrl\".\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "164 people (80 males, 84 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998039.\n",
      "--hwe: 0 variants removed due to Hardy-Weinberg exact test.\n",
      "1073226 variants and 164 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (52 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_9.bed + HapMap_3_r3_9.bim + HapMap_3_r3_9.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#症例のHWE閾値は、HWEから極端に逸脱するSNPのみをフィルタリングする。\n",
    "#対照群ではHWE p値 < hwe 1e-6のSNPはすでに除去されているので、この2番目のHWEステップは症例にのみ焦点を当てる。\n",
    "plink --bfile HapMap_hwe_filter_step1 --hwe 1e-10 --hwe-all --make-bed --out HapMap_3_r3_9"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cc1fe348-616c-419e-b796-e70204953dca",
   "metadata": {},
   "source": [
    "### Step 5：ヘテロ接合度チェック\n",
    "\n",
    "Generate a plot of the distribution of the heterozygosity rate of your subjects.  *被験者のヘテロ接合率の分布をプロットする。*\n",
    "\n",
    "And remove individuals with a heterozygosity rate deviating more than 3 sd from the mean.  *そしてヘテロ接合率が平均値から3 sd以上ずれている個体を削除する。*\n",
    "\n",
    "Checks for heterozygosity are performed on a set of SNPs which are not highly correlated.  *ヘテロ接合のチェックは、相関の高くないSNPの集合に対して行われる。*\n",
    "\n",
    "Therefore, to generate a list of non-(highly)correlated SNPs, we exclude high inversion regions (inversion.txt [High LD regions]) and prune the SNPs using the command --indep-pairwise�.　　\n",
    "\n",
    "The parameters �50 5 0.2� stand respectively for: the window size, the number of SNPs to shift the window at each step, and the multiple correlation coefficient for a SNP being regressed on all other SNPs simultaneously.  *パラメータ 50 5 0.2 はそれぞれ、窓の大きさ、各ステップで窓をシフトさせるSNPの数、SNPが他のSNPと同時に回帰される場合の重相関係数を表す。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_9 --exclude inversion.txt --range --indep-pairwise 50 5 0.2 --out indepSNP\n",
    "\n",
    "Note, don't delete the file indepSNP.prune.in, we will use this file in later steps of the tutorial.  *indepSNP.prune.inは削除しないでください。このファイルはチュートリアルの後のステップで使用します。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_9 --extract indepSNP.prune.in --het --out R_check\n",
    "\n",
    "This file contains your pruned data set.\n",
    "\n",
    "Plot of the heterozygosity rate distribution *ヘテロ接合率分布のプロット*\n",
    "\n",
    "    Rscript --no-save check_heterozygosity_rate.R\n",
    "\n",
    "The following code generates a list of individuals who deviate more than 3 standard deviations from the heterozygosity rate mean.  *以下のコードは、ヘテロ接合率の平均値から3標準偏差以上ずれた個体のリストを生成する。*\n",
    "\n",
    "For data manipulation we recommend using UNIX. However, when performing statistical calculations R might be more convenient, hence the use of the Rscript for this step:\n",
    "\n",
    "    Rscript --no-save heterozygosity_outliers_list.R\n",
    "\n",
    "Output of the command above: fail-het-qc.txt .\n",
    "When using our example data/the HapMap data this list contains 2 individuals (i.e., two individuals have a heterozygosity rate deviating more than 3 SD's from the mean).  *私たちの例のデータ／HapMapデータを使用する場合、このリストには2個体が含まれる（すなわち、平均から3SD以上乖離したヘテロ接合率を持つ個体が2個体ある）。*\n",
    "Adapt this file to make it compatible for PLINK, by removing all quotation marks from the file and selecting only the first two columns.  *このファイルをPLINKに適合させるために、ファイルから引用符をすべて取り除き、最初の2列だけを選択する。*\n",
    "\n",
    "    sed 's/\"// g' fail-het-qc.txt | awk '{print$1, $2}'> het_fail_ind.txt\n",
    "\n",
    "Remove heterozygosity rate outliers.  *ヘテロ接合率の異常値を取り除く。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_9 --remove het_fail_ind.txt --make-bed --out HapMap_3_r3_10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "e44c781a-0a1a-4fac-9ba0-7e8e262b0037",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to indepSNP.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_9\n",
      "  --exclude inversion.txt\n",
      "  --indep-pairwise 50 5 0.2\n",
      "  --out indepSNP\n",
      "  --range\n",
      "\n",
      "Note: --range flag deprecated.  Use e.g. \"--extract range <filename>\".\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "164 people (80 males, 84 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "--exclude range: 9893 variants excluded.\n",
      "--exclude range: 1063333 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998034.\n",
      "1063333 variants and 164 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (52 phenotypes\n",
      "are missing.)\n",
      "Pruned 79323 variants from chromosome 1, leaving 8397.\n",
      "Pruned 81833 variants from chromosome 2, leaving 7965.\n",
      "Pruned 67905 variants from chromosome 3, leaving 6957.\n",
      "Pruned 59914 variants from chromosome 4, leaving 6215.\n",
      "Pruned 62681 variants from chromosome 5, leaving 6336.\n",
      "Pruned 59440 variants from chromosome 6, leaving 6092.\n",
      "Pruned 53506 variants from chromosome 7, leaving 5598.\n",
      "Pruned 51227 variants from chromosome 8, leaving 5025.\n",
      "Pruned 45349 variants from chromosome 9, leaving 4983.\n",
      "Pruned 51843 variants from chromosome 10, leaving 5382.\n",
      "Pruned 50263 variants from chromosome 11, leaving 5021.\n",
      "Pruned 47315 variants from chromosome 12, leaving 5250.\n",
      "Pruned 36460 variants from chromosome 13, leaving 4030.\n",
      "Pruned 31645 variants from chromosome 14, leaving 3499.\n",
      "Pruned 28604 variants from chromosome 15, leaving 3404.\n",
      "Pruned 30134 variants from chromosome 16, leaving 3685.\n",
      "Pruned 24471 variants from chromosome 17, leaving 3366.\n",
      "Pruned 28242 variants from chromosome 18, leaving 3413.\n",
      "Pruned 16854 variants from chromosome 19, leaving 2808.\n",
      "Pruned 24897 variants from chromosome 20, leaving 3051.\n",
      "Pruned 13769 variants from chromosome 21, leaving 1713.\n",
      "Pruned 13514 variants from chromosome 22, leaving 1954.\n",
      "Pruning complete.  959189 of 1063333 variants removed.\n",
      "Marker lists written to indepSNP.prune.in and indepSNP.prune.out .\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to R_check.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_9\n",
      "  --extract indepSNP.prune.in\n",
      "  --het\n",
      "  --out R_check\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "164 people (80 males, 84 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "--extract: 104144 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998028.\n",
      "104144 variants and 164 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (52 phenotypes\n",
      "are missing.)\n",
      "--het: 104144 variants scanned, report written to R_check.het .\n",
      "het <- read.table(\"R_check.het\", head=TRUE)\n",
      "pdf(\"heterozygosity.pdf\")\n",
      "het$HET_RATE = (het$\"N.NM.\" - het$\"O.HOM.\")/het$\"N.NM.\"\n",
      "hist(het$HET_RATE, xlab=\"Heterozygosity Rate\", ylab=\"Frequency\", main= \"Heterozygosity Rate\")\n",
      "dev.off()\n",
      "null device \n",
      "          1 \n"
     ]
    }
   ],
   "source": [
    "#ヘテロ接合率分布のプロット\n",
    "plink --bfile HapMap_3_r3_9 --exclude inversion.txt --range --indep-pairwise 50 5 0.2 --out indepSNP\n",
    "plink --bfile HapMap_3_r3_9 --extract indepSNP.prune.in --het --out R_check\n",
    "cat check_heterozygosity_rate.R\n",
    "Rscript --no-save check_heterozygosity_rate.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "28b7e24b-0a9d-4c8d-a91d-cd76dfa0b074",
   "metadata": {},
   "outputs": [],
   "source": [
    "filename=\"heterozygosity\"\n",
    "pdftoppm -png $filename.pdf $filename\n",
    "cp $filename-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02535b40-3895-4cb6-9310-14070911187d",
   "metadata": {},
   "source": [
    "<img src=\"pic/heterozygosity-1.png\" width=\"50%\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "7e15f1e4-a04b-44b4-815f-2fdd7ed120ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "het <- read.table(\"R_check.het\", head=TRUE)\n",
      "het$HET_RATE = (het$\"N.NM.\" - het$\"O.HOM.\")/het$\"N.NM.\"\n",
      "het_fail = subset(het, (het$HET_RATE < mean(het$HET_RATE)-3*sd(het$HET_RATE)) | (het$HET_RATE > mean(het$HET_RATE)+3*sd(het$HET_RATE)));\n",
      "het_fail$HET_DST = (het_fail$HET_RATE-mean(het$HET_RATE))/sd(het$HET_RATE);\n",
      "write.table(het_fail, \"fail-het-qc.txt\", row.names=FALSE)\n",
      "\"FID\" \"IID\" \"O.HOM.\" \"E.HOM.\" \"N.NM.\" \"F\" \"HET_RATE\" \"HET_DST\"\n",
      "1330 \"NA12342\" 68049 67240 103571 0.02229 0.342972453679119 -3.66711854374478\n",
      "1459 \"NA12874\" 68802 67560 104068 0.0339 0.338874582004074 -5.04839854982741\n"
     ]
    }
   ],
   "source": [
    "#以下のコードは、ヘテロ接合率の平均値から3標準偏差以上ずれた個体のリストを生成する。\n",
    "cat heterozygosity_outliers_list.R\n",
    "Rscript --no-save heterozygosity_outliers_list.R\n",
    "\n",
    "cat fail-het-qc.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "b9badf3b-6628-4ab0-a8de-8e7d60dadd0a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_10.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_9\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_10\n",
      "  --remove het_fail_ind.txt\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "164 people (80 males, 84 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "--remove: 162 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 110 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.998063.\n",
      "1073226 variants and 162 people pass filters and QC.\n",
      "Among remaining phenotypes, 55 are cases and 55 are controls.  (52 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_10.bed + HapMap_3_r3_10.bim + HapMap_3_r3_10.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#ファイルから引用符をすべて取り除き、最初の2列だけを選択する。\n",
    "sed 's/\"// g' fail-het-qc.txt | awk '{print$1, $2}'> het_fail_ind.txt\n",
    "#ヘテロ接合率の異常値を取り除く。\n",
    "plink --bfile HapMap_3_r3_9 --remove het_fail_ind.txt --make-bed --out HapMap_3_r3_10"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38f54ae5-40f8-4dc6-96e2-b93fb70737dd",
   "metadata": {},
   "source": [
    "### Step 6：血縁関係チェック\n",
    "\n",
    "It is essential to check datasets you analyse for cryptic relatedness.　　*分析するデータセットに隠微な血縁関係がないかチェックすることは不可欠である。*\n",
    "\n",
    "Assuming a random population sample we are going to exclude all individuals above the pihat threshold of 0.2 in this tutorial.　　*このチュートリアルでは、無作為母集団標本を仮定して、pihatのしきい値0.2以上の個体はすべて除外します。*\n",
    "\n",
    "Check for relationships between individuals with a pihat > 0.2.   *pihat > 0.2の個体間の関係を調べます。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_10 --extract indepSNP.prune.in --genome --min 0.2 --out pihat_min0.2\n",
    "\n",
    "The HapMap dataset is known to contain parent-offspring relations.  *HapMap データセットには親子関係が含まれていることが知られている。*\n",
    "\n",
    "The following commands will visualize specifically these parent-offspring relations, using the z values.   *以下のコマンドは、これらの親子関係を z 値を使って具体的に可視化する。*\n",
    "\n",
    "    awk '{ if ($8 >0.9) print $0 }' pihat_min0.2.genome>zoom_pihat.genome\n",
    "\n",
    "Generate a plot to assess the type of relationship.  *血縁関係のタイプを評価するプロットを生成する。*\n",
    "\n",
    "    Rscript --no-save Relatedness.R\n",
    "\n",
    "The generated plots show a considerable amount of related individuals (explentation plot; PO = parent-offspring, UN = unrelated individuals) in the Hapmap data, this is expected since the dataset was constructed as such.　　　*生成されたプロットはHapmapデータにかなりの数の関連個体（説明プロット；PO = parent-offspring, UN = unrelated individuals）が含まれていることを示しているが、これはデータセットがそのように構築されていることから予想されることである。*\n",
    "\n",
    "Normally, family based data should be analyzed using specific family based methods. In this tutorial, for demonstrative purposes, we treat the relatedness as cryptic relatedness in a random population sample.   *通常、ファミリー・ベースのデータは、特定のファミリー・ベースの手法を用いて解析されるべきです。このチュートリアルでは、デモンストレーションのために、ランダムな母集団サンプルにおける隠微な血縁関係を扱います。*\n",
    "\n",
    "In this tutorial, we aim to remove all 'relatedness' from our dataset. *このチュートリアルでは、データセットからすべての「血縁」を取り除くことを目的とする。*\n",
    "\n",
    "To demonstrate that the majority of the relatedness was due to parent-offspring we only include founders (individuals without parents in the dataset).  *血縁関係の大部分が親-子によるものであることを実証するために、ファウンダー（データセットに親のいない個体）のみを含める。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_10 --filter-founders --make-bed --out HapMap_3_r3_11\n",
    "\n",
    "Now we will look again for individuals with a pihat >0.2.  *ここでもう一度、0.2以上のpihatを持つ個体について調べる。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_11 --extract indepSNP.prune.in --genome --min 0.2 --out pihat_min0.2_in_founders\n",
    "\n",
    "The file 'pihat_min0.2_in_founders.genome' shows that, after exclusion of all non-founders, only 1 individual pair with a pihat greater than 0.2 remains in the HapMap data. *ファイル'pihat_min0.2_in_founders.genome'を見ると、すべての非ファウンダーを除外した後、0.2より大きいpihatを持つ個体ペアが1組だけHapMapデータに残っている。*\n",
    "\n",
    "This is likely to be a full sib or DZ twin pair based on the Z values. Noteworthy, they were not given the same family identity (FID) in the HapMap data. *これは、Z値から見て、完全な兄弟姉妹か二卵性双生児のペアである可能性が高い。注目すべきは、HapMapデータでは同じファミリーID（FID）が与えられていないことである。*\n",
    "\n",
    "For each pair of 'related' individuals with a pihat > 0.2, we recommend to remove the individual with the lowest call rate.   *pihat > 0.2の「関連」個体の各ペアについて、コール率が最も低い個体を削除することを推奨する。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_11 --missing\n",
    "\n",
    "Use an UNIX text editor (e.g., vi(m) ) to check which individual has the highest call rate in the 'related pair'.  *UNIXのテキストエディタ(vi(m)など)を使って、「関連ペア」の中でどの個体のコール率が最も高いかを調べる。*\n",
    "\n",
    "Generate a list of FID and IID of the individual(s) with a Pihat above 0.2, to check who had the lower call rate of the pair.  *Pihatが0.2以上の個体のFIDとIIDのリストを作成し、そのペアの中でコール率が低いのは誰かを調べる。*\n",
    "\n",
    "In our dataset the individual 13291  NA07045 had the lower call rate. *私たちのデータセットでは、13291 NA07045という個体のコール率が低い。*\n",
    "\n",
    "    vi 0.2_low_call_rate_pihat.txt\n",
    "    i \n",
    "    13291  NA07045\n",
    "    # Press esc on keyboard!\n",
    "    :x\n",
    "    # Press enter on keyboard!\n",
    "\n",
    "In case of multiple 'related' pairs, the list generated above can be extended using the same method as for our lone 'related' pair.  *複数の「親族」ペアがある場合、上記で作成したリストは、単独の「親族」ペアと同じ方法で拡張できる。*\n",
    "\n",
    "Delete the individuals with the lowest call rate in 'related' pairs with a pihat > 0.2 　　*pihat>0.2の「親族」ペアの中で、コール率が最も低い個体を削除する。*\n",
    "\n",
    "    plink --bfile HapMap_3_r3_11 --remove 0.2_low_call_rate_pihat.txt --make-bed --out HapMap_3_r3_12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "2683e14a-26ed-407a-81e1-1be2bfd943c0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pihat_min0.2.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_10\n",
      "  --extract indepSNP.prune.in\n",
      "  --genome\n",
      "  --min 0.2\n",
      "  --out pihat_min0.2\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "162 people (78 males, 84 females) loaded from .fam.\n",
      "110 phenotype values loaded from .fam.\n",
      "--extract: 104144 variants remaining.\n",
      "Using up to 63 threads (change this with --threads).\n",
      "Before main variant filters, 110 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998042.\n",
      "104144 variants and 162 people pass filters and QC.\n",
      "Among remaining phenotypes, 55 are cases and 55 are controls.  (52 phenotypes\n",
      "are missing.)\n",
      "IBD calculations complete.  \n",
      "Finished writing pihat_min0.2.genome .\n"
     ]
    }
   ],
   "source": [
    "#pihat > 0.2の個体間の関係を調べます。\n",
    "plink --bfile HapMap_3_r3_10 --extract indepSNP.prune.in --genome --min 0.2 --out pihat_min0.2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "af8502d4-b6ce-4253-842d-6c5394a213a1",
   "metadata": {},
   "outputs": [],
   "source": [
    "#以下のコマンドは、これらの親子関係を z 値を使って具体的に可視化する。\n",
    "awk '{ if ($8 >0.9) print $0 }' pihat_min0.2.genome > zoom_pihat.genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "a273aae6-27e3-490d-b223-7c0d3b5c922c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pdf(\"relatedness.pdf\")\n",
      "relatedness = read.table(\"pihat_min0.2.genome\", header=T)\n",
      "par(pch=16, cex=1)\n",
      "with(relatedness,plot(Z0,Z1, xlim=c(0,1), ylim=c(0,1), type=\"n\"))\n",
      "with(subset(relatedness,RT==\"PO\") , points(Z0,Z1,col=4))\n",
      "with(subset(relatedness,RT==\"UN\") , points(Z0,Z1,col=3))\n",
      "legend(1,1, xjust=1, yjust=1, legend=levels(relatedness$RT), pch=16, col=c(4,3))\n",
      "\n",
      "pdf(\"zoom_relatedness.pdf\")\n",
      "relatedness_zoom = read.table(\"zoom_pihat.genome\", header=T)\n",
      "par(pch=16, cex=1)\n",
      "with(relatedness_zoom,plot(Z0,Z1, xlim=c(0,0.02), ylim=c(0.98,1), type=\"n\"))\n",
      "with(subset(relatedness_zoom,RT==\"PO\") , points(Z0,Z1,col=4))\n",
      "with(subset(relatedness_zoom,RT==\"UN\") , points(Z0,Z1,col=3))\n",
      "legend(0.02,1, xjust=1, yjust=1, legend=levels(relatedness$RT), pch=16, col=c(4,3))\n",
      "\n",
      "pdf(\"hist_relatedness.pdf\")\n",
      "relatedness = read.table(\"pihat_min0.2.genome\", header=T)\n",
      "hist(relatedness[,10],main=\"Histogram relatedness\", xlab= \"Pihat\")  \n",
      "dev.off() \n",
      "\n",
      "legend(1,1, xjust=1, yjust=1, legend=\u001b[01;31m\u001b[Klevels(\u001b[m\u001b[Krelatedness$RT), pch=16, col=c(4,3))\n",
      "legend(0.02,1, xjust=1, yjust=1, legend=\u001b[01;31m\u001b[Klevels(\u001b[m\u001b[Krelatedness$RT), pch=16, col=c(4,3))\n",
      "legend(1,1, xjust=1, yjust=1, legend=levels(relatedness$\u001b[01;31m\u001b[KRT)\u001b[m\u001b[K, pch=16, col=c(4,3))\n",
      "legend(0.02,1, xjust=1, yjust=1, legend=levels(relatedness$\u001b[01;31m\u001b[KRT)\u001b[m\u001b[K, pch=16, col=c(4,3))\n",
      "pdf(\"relatedness.pdf\")\n",
      "relatedness = read.table(\"pihat_min0.2.genome\", header=T)\n",
      "par(pch=16, cex=1)\n",
      "with(relatedness,plot(Z0,Z1, xlim=c(0,1), ylim=c(0,1), type=\"n\"))\n",
      "with(subset(relatedness,RT==\"PO\") , points(Z0,Z1,col=4))\n",
      "with(subset(relatedness,RT==\"UN\") , points(Z0,Z1,col=3))\n",
      "legend(1,1, xjust=1, yjust=1, legend=unique(relatedness[,\"RT\"]), pch=16, col=c(4,3))\n",
      "\n",
      "pdf(\"zoom_relatedness.pdf\")\n",
      "relatedness_zoom = read.table(\"zoom_pihat.genome\", header=T)\n",
      "par(pch=16, cex=1)\n",
      "with(relatedness_zoom,plot(Z0,Z1, xlim=c(0,0.02), ylim=c(0.98,1), type=\"n\"))\n",
      "with(subset(relatedness_zoom,RT==\"PO\") , points(Z0,Z1,col=4))\n",
      "with(subset(relatedness_zoom,RT==\"UN\") , points(Z0,Z1,col=3))\n",
      "legend(0.02,1, xjust=1, yjust=1, legend=unique(relatedness[,\"RT\"]), pch=16, col=c(4,3))\n",
      "\n",
      "pdf(\"hist_relatedness.pdf\")\n",
      "relatedness = read.table(\"pihat_min0.2.genome\", header=T)\n",
      "hist(relatedness[,10],main=\"Histogram relatedness\", xlab= \"Pihat\")  \n",
      "dev.off() \n",
      "\n",
      "pdf \n",
      "  2 \n"
     ]
    }
   ],
   "source": [
    "#血縁関係のタイプを評価するプロットを生成する。\n",
    "cat Relatedness.R\n",
    "#Rscript --no-save Relatedness.R\n",
    "cat Relatedness.R | grep \"levels(\"\n",
    "cat Relatedness.R | grep \"RT)\"\n",
    "cat Relatedness.R | sed -e 's/levels(relatedness$RT)/unique(relatedness[,\"RT\"])/g' \n",
    "cat Relatedness.R | sed -e 's/levels(relatedness$RT)/unique(relatedness[,\"RT\"])/g'  > Relatedness_HM.R\n",
    "Rscript --no-save Relatedness_HM.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "559d191f-b25e-49f7-96bb-0c9c37709b2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "filename=\"relatedness\"\n",
    "pdftoppm -png $filename.pdf $filename\n",
    "cp $filename-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "b29f7538-9fca-4f6a-b123-6729cfa4e490",
   "metadata": {},
   "outputs": [],
   "source": [
    "filename=\"zoom_relatedness\"\n",
    "pdftoppm -png $filename.pdf $filename\n",
    "cp $filename-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "7835fc08-4486-45ef-b492-197101b0fa1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "filename=\"hist_relatedness\"\n",
    "pdftoppm -png $filename.pdf $filename\n",
    "cp $filename-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e12dfb69-3e50-4a42-99ee-00e654c07024",
   "metadata": {},
   "source": [
    "<img src=\"pic/relatedness-1.png\" width=\"33%\"><img src=\"pic/zoom_relatedness-1.png\" width=\"33%\"><img src=\"pic/hist_relatedness-1.png\" width=\"33%\">\n",
    "生成されたプロットはHapmapデータにかなりの数の関連個体（説明プロット；PO = parent-offspring, UN = unrelated individuals）が含まれていることを示しているが、これはデータセットがそのように構築されていることから予想されることである。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "7ee40fe6-5580-4d44-a431-dd4c7410d3b3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_11.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_10\n",
      "  --filter-founders\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_11\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "162 people (78 males, 84 females) loaded from .fam.\n",
      "110 phenotype values loaded from .fam.\n",
      "52 people removed due to founder status (--filter-founders).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 110 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.998025.\n",
      "1073226 variants and 110 people pass filters and QC.\n",
      "Among remaining phenotypes, 55 are cases and 55 are controls.\n",
      "--make-bed to HapMap_3_r3_11.bed + HapMap_3_r3_11.bim + HapMap_3_r3_11.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to pihat_min0.2_in_founders.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_11\n",
      "  --extract indepSNP.prune.in\n",
      "  --genome\n",
      "  --min 0.2\n",
      "  --out pihat_min0.2_in_founders\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "110 people (55 males, 55 females) loaded from .fam.\n",
      "110 phenotype values loaded from .fam.\n",
      "--extract: 104144 variants remaining.\n",
      "Using up to 63 threads (change this with --threads).\n",
      "Before main variant filters, 110 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997978.\n",
      "104144 variants and 110 people pass filters and QC.\n",
      "Among remaining phenotypes, 55 are cases and 55 are controls.\n",
      "IBD calculations complete.  \n",
      "Finished writing pihat_min0.2_in_founders.genome .\n"
     ]
    }
   ],
   "source": [
    "#通常、ファミリー・ベースのデータは、特定のファミリー・ベースの手法を用いて解析されるべきです。このチュートリアルでは、デモンストレーションのために、ランダムな母集団サンプルにおける隠微な血縁関係を扱います。\n",
    "#このチュートリアルでは、データセットからすべての「血縁」を取り除くことを目的とする\n",
    "\n",
    "#血縁関係の大部分が親-子によるものであることを実証するために、ファウンダー（データセットに親のいない個体）のみを含める。\n",
    "plink --bfile HapMap_3_r3_10 --filter-founders --make-bed --out HapMap_3_r3_11\n",
    "\n",
    "#ここでもう一度、0.2以上のpihatを持つ個体について調べる。\n",
    "plink --bfile HapMap_3_r3_11 --extract indepSNP.prune.in --genome --min 0.2 --out pihat_min0.2_in_founders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "1dff2355-84f8-4870-883f-3c4161b2e0da",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   FID1     IID1   FID2     IID2 RT    EZ      Z0      Z1      Z2  PI_HAT PHE       DST     PPC   RATIO\n",
      "  13291  NA07045   1454  NA12813 UN    NA  0.2572  0.5007  0.2421  0.4924   0  0.839777  1.0000  9.7022\n"
     ]
    }
   ],
   "source": [
    "cat pihat_min0.2_in_founders.genome\n",
    "#これは、Z値から見て、完全な兄弟姉妹か二卵性双生児のペアである可能性が高い。注目すべきは、HapMapデータでは同じファミリーID（FID）が与えられていないことである"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "e8a7895c-ee7e-4b41-adab-6b70d4beb26f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plink.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_11\n",
      "  --missing\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "110 people (55 males, 55 females) loaded from .fam.\n",
      "110 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 110 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998025.\n",
      "--missing: Sample missing data report written to plink.imiss, and variant-based\n",
      "missing data report written to plink.lmiss.\n"
     ]
    }
   ],
   "source": [
    "#pihat > 0.2の「関連」個体の各ペアについて、コール率が最も低い個体を削除することを推奨する。\n",
    "plink --bfile HapMap_3_r3_11 --missing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "c9962727-7514-4a24-b772-99bf70fc2b50",
   "metadata": {},
   "outputs": [],
   "source": [
    "echo \"13291  NA07045\" > 0.2_low_call_rate_pihat.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "882542f0-c711-4219-8b78-53c2a4e5931a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_12.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_11\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_12\n",
      "  --remove 0.2_low_call_rate_pihat.txt\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "110 people (55 males, 55 females) loaded from .fam.\n",
      "110 phenotype values loaded from .fam.\n",
      "--remove: 109 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.998028.\n",
      "1073226 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "--make-bed to HapMap_3_r3_12.bed + HapMap_3_r3_12.bim + HapMap_3_r3_12.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#pihat>0.2の「親族」ペアの中で、コール率が最も低い個体を削除する。\n",
    "plink --bfile HapMap_3_r3_11 --remove 0.2_low_call_rate_pihat.txt --make-bed --out HapMap_3_r3_12"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ac8d7f24-b3fa-4840-b0a2-6828e0dd9c0c",
   "metadata": {},
   "source": [
    "### CONGRATULATIONS!! \n",
    "\n",
    "You've just succesfully completed the first tutorial! You are now able to conduct a proper genetic QC. \n",
    "\n",
    "For the next tutorial, using the script: 2_Main_script_MDS.txt, you need the following files:\n",
    "- The bfile HapMap_3_r3_12 (i.e., HapMap_3_r3_12.fam,HapMap_3_r3_12.bed, and HapMap_3_r3_12.bim\n",
    "- indepSNP.prune.in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "49a55af3-83d2-4b25-a9ba-b66ec01db05b",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/1_QC_GWAS\n",
    "cp HapMap_3_r3_12.fam ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/2_Population_stratification/\n",
    "cp HapMap_3_r3_12.bed ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/2_Population_stratification/\n",
    "cp HapMap_3_r3_12.bim ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/2_Population_stratification/\n",
    "cp indepSNP.prune.in ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/2_Population_stratification/\n",
    "\n",
    "cp HapMap_3_r3_12.fam ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS\n",
    "cp HapMap_3_r3_12.bed ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS\n",
    "cp HapMap_3_r3_12.bim ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS\n",
    "cp indepSNP.prune.in ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e1af854e-b8a5-4762-a138-8610a70f9be6",
   "metadata": {},
   "source": [
    "### ここまでのおおまかな流れをまとめるとこう\n",
    "\n",
    "##### Step 1：欠損データのチェック\n",
    "##### Step 2：性別チェック\n",
    "##### Step 3：常染色体SNP選択・レアアリル削除\n",
    "##### Step 4：ハーディーワインベルグ平衡(HWE)にないSNPを削除\n",
    "##### Step 5：ヘテロ接合度チェック\n",
    "##### Step 6：血縁関係チェック"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee1b58a0-976e-4fa9-8782-a43ce3fd7444",
   "metadata": {},
   "source": [
    "## 2.Population_stratification"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b0a4ddb6-9cab-4165-a3be-c22004d95591",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/2_Population_stratification/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b5876805-d844-4524-971b-c53a14bf3ccc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################ Explanation of the main script ##########################\n",
      "\n",
      "# This is the main script for second tutorial from our comprehensive tutorial on GWAS and PRS.\n",
      "# To run this script the following (b)files from the first tutorial are required: HapMap_3_r3_12 (this bfile contain: HapMap_3_r3_12.fam,HapMap_3_r3_12.bim, and HapMap_3_r3_12.bed; you need all three), and indepSNP.prune.in.\n",
      "# In this tutorial we are going to check for population stratification.\n",
      "# We will do this as follows, the bfile (HapMap_3_r3_12) generated at the end of the previous tutorial (1_QC_GWAS) is going to checked for population stratification using data from the 1000 Genomes Project. Individuals with a non-European ethnic background will be removed.\n",
      "# Furthermore, this tutorial will generate a covariate file which helps to adust for remaining population stratification within the European subjects.\n",
      "# In order to complete this tutorial it is necessary to have generated the bfile 'HapMap_3_r3_12' and the file 'indepSNP.prune.in' from the previous tutorial.\n",
      "\n",
      "\n",
      "##############################################################\n",
      "############### START ANALISIS ###############################\n",
      "##############################################################\n",
      "\n",
      "# Copy the (b)files from the previous tutorial to the current directory (see explanation of the main script).\n",
      "cp HOME/{user}/{path/1_QC_GWAS}/HapMap_3_r3_12.* HOME/{user}/{path/2_Population_stratification}\n",
      "cp HOME/{user}/{path/1_QC_GWAS}/indepSNP.prune.in HOME/{user}/{path/2_Population_stratification}\n",
      "\n",
      "\n",
      "## Download 1000 Genomes data ##\n",
      "# This file from the 1000 Genomes contains genetic data of 629 individuals from different ethnic backgrounds.\n",
      "# Note, this file is quite large (>60 gigabyte).  \n",
      "wget ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20100804/ALL.2of4intersection.20100804.genotypes.vcf.gz\n",
      "\n",
      "# Convert vcf to Plink format.\n",
      "plink --vcf ALL.2of4intersection.20100804.genotypes.vcf.gz --make-bed --out ALL.2of4intersection.20100804.genotypes\n",
      "# Noteworthy, the file 'ALL.2of4intersection.20100804.genotypes.bim' contains SNPs without an rs-identifier, these SNPs are indicated with \".\". This can also be observed in the file 'ALL.2of4intersection.20100804.genotypes.vcf.gz'. To check this file use this command: zmore ALL.2of4intersection.20100804.genotypes.vcf.gz .\n",
      "# The missing rs-identifiers in the 1000 Genomes data are not a problem for this tutorial.\n",
      "# However, for good practice, we will assign unique indentifiers to the SNPs with a missing rs-identifier (i.e., the SNPs with \".\").\n",
      "plink --bfile ALL.2of4intersection.20100804.genotypes --set-missing-var-ids @:#[b37]\\$1,\\$2 --make-bed --out ALL.2of4intersection.20100804.genotypes_no_missing_IDs\n",
      "\n",
      "## QC on 1000 Genomes data.\n",
      "# Remove variants based on missing genotype data.\n",
      "plink --bfile ALL.2of4intersection.20100804.genotypes_no_missing_IDs --geno 0.2 --allow-no-sex --make-bed --out 1kG_MDS\n",
      "\n",
      "# Remove individuals based on missing genotype data.\n",
      "plink --bfile 1kG_MDS --mind 0.2 --allow-no-sex --make-bed --out 1kG_MDS2\n",
      "\n",
      "# Remove variants based on missing genotype data.\n",
      "plink --bfile 1kG_MDS2 --geno 0.02 --allow-no-sex --make-bed --out 1kG_MDS3\n",
      "\n",
      "# Remove individuals based on missing genotype data.\n",
      "plink --bfile 1kG_MDS3 --mind 0.02 --allow-no-sex --make-bed --out 1kG_MDS4\n",
      "\n",
      "# Remove variants based on MAF.\n",
      "plink --bfile 1kG_MDS4 --maf 0.05 --allow-no-sex --make-bed --out 1kG_MDS5\n",
      "\n",
      "# Extract the variants present in HapMap dataset from the 1000 genomes dataset.\n",
      "awk '{print$2}' HapMap_3_r3_12.bim > HapMap_SNPs.txt\n",
      "plink --bfile 1kG_MDS5 --extract HapMap_SNPs.txt --make-bed --out 1kG_MDS6\n",
      "\n",
      "# Extract the variants present in 1000 Genomes dataset from the HapMap dataset.\n",
      "awk '{print$2}' 1kG_MDS6.bim > 1kG_MDS6_SNPs.txt\n",
      "plink --bfile HapMap_3_r3_12 --extract 1kG_MDS6_SNPs.txt --recode --make-bed --out HapMap_MDS\n",
      "# The datasets now contain the exact same variants.\n",
      "\n",
      "## The datasets must have the same build. Change the build 1000 Genomes data build.\n",
      "awk '{print$2,$4}' HapMap_MDS.map > buildhapmap.txt\n",
      "# buildhapmap.txt contains one SNP-id and physical position per line.\n",
      "\n",
      "plink --bfile 1kG_MDS6 --update-map buildhapmap.txt --make-bed --out 1kG_MDS7\n",
      "# 1kG_MDS7 and HapMap_MDS now have the same build.\n",
      "\n",
      "## Merge the HapMap and 1000 Genomes data sets\n",
      "\n",
      "# Prior to merging 1000 Genomes data with the HapMap data we want to make sure that the files are mergeable, for this we conduct 3 steps:\n",
      "# 1) Make sure the reference genome is similar in the HapMap and the 1000 Genomes Project datasets.\n",
      "# 2) Resolve strand issues.\n",
      "# 3) Remove the SNPs which after the previous two steps still differ between datasets.\n",
      "\n",
      "# The following steps are maybe quite technical in terms of commands, but we just compare the two data sets and make sure they correspond.\n",
      "\n",
      "# 1) set reference genome \n",
      "awk '{print$2,$5}' 1kG_MDS7.bim > 1kg_ref-list.txt\n",
      "plink --bfile HapMap_MDS --reference-allele 1kg_ref-list.txt --make-bed --out HapMap-adj\n",
      "# The 1kG_MDS7 and the HapMap-adj have the same reference genome for all SNPs.\n",
      "# This command will generate some warnings for impossible A1 allele assignment.\n",
      "\n",
      "# 2) Resolve strand issues.\n",
      "# Check for potential strand issues.\n",
      "awk '{print$2,$5,$6}' 1kG_MDS7.bim > 1kGMDS7_tmp\n",
      "awk '{print$2,$5,$6}' HapMap-adj.bim > HapMap-adj_tmp\n",
      "sort 1kGMDS7_tmp HapMap-adj_tmp |uniq -u > all_differences.txt\n",
      "# 1624 differences between the files, some of these might be due to strand issues.\n",
      "\n",
      "## Flip SNPs for resolving strand issues.\n",
      "# Print SNP-identifier and remove duplicates.\n",
      "awk '{print$1}' all_differences.txt | sort -u > flip_list.txt\n",
      "# Generates a file of 812 SNPs. These are the non-corresponding SNPs between the two files. \n",
      "# Flip the 812 non-corresponding SNPs. \n",
      "plink --bfile HapMap-adj --flip flip_list.txt --reference-allele 1kg_ref-list.txt --make-bed --out corrected_hapmap\n",
      "\n",
      "# Check for SNPs which are still problematic after they have been flipped.\n",
      "awk '{print$2,$5,$6}' corrected_hapmap.bim > corrected_hapmap_tmp\n",
      "sort 1kGMDS7_tmp corrected_hapmap_tmp |uniq -u  > uncorresponding_SNPs.txt\n",
      "# This file demonstrates that there are 84 differences between the files.\n",
      "\n",
      "# 3) Remove problematic SNPs from HapMap and 1000 Genomes.\n",
      "awk '{print$1}' uncorresponding_SNPs.txt | sort -u > SNPs_for_exlusion.txt\n",
      "# The command above generates a list of the 42 SNPs which caused the 84 differences between the HapMap and the 1000 Genomes data sets after flipping and setting of the reference genome.\n",
      "\n",
      "# Remove the 42 problematic SNPs from both datasets.\n",
      "plink --bfile corrected_hapmap --exclude SNPs_for_exlusion.txt --make-bed --out HapMap_MDS2\n",
      "plink --bfile 1kG_MDS7 --exclude SNPs_for_exlusion.txt --make-bed --out 1kG_MDS8\n",
      "\n",
      "# Merge HapMap with 1000 Genomes Data.\n",
      "plink --bfile HapMap_MDS2 --bmerge 1kG_MDS8.bed 1kG_MDS8.bim 1kG_MDS8.fam --allow-no-sex --make-bed --out MDS_merge2\n",
      "\n",
      "# Note, we are fully aware of the sample overlap between the HapMap and 1000 Genomes datasets. However, for the purpose of this tutorial this is not important.\n",
      "\n",
      "## Perform MDS on HapMap-CEU data anchored by 1000 Genomes data.\n",
      "# Using a set of pruned SNPs\n",
      "plink --bfile MDS_merge2 --extract indepSNP.prune.in --genome --out MDS_merge2\n",
      "plink --bfile MDS_merge2 --read-genome MDS_merge2.genome --cluster --mds-plot 10 --out MDS_merge2\n",
      "\n",
      "\n",
      "### MDS-plot\n",
      "\n",
      "# Download the file with population information of the 1000 genomes dataset.\n",
      "wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20100804/20100804.ALL.panel\n",
      "# The file 20100804.ALL.panel contains population codes of the individuals of 1000 genomes.\n",
      "\n",
      "# Convert population codes into superpopulation codes (i.e., AFR,AMR,ASN, and EUR).\n",
      "awk '{print$1,$1,$2}' 20100804.ALL.panel > race_1kG.txt\n",
      "sed 's/JPT/ASN/g' race_1kG.txt>race_1kG2.txt\n",
      "sed 's/ASW/AFR/g' race_1kG2.txt>race_1kG3.txt\n",
      "sed 's/CEU/EUR/g' race_1kG3.txt>race_1kG4.txt\n",
      "sed 's/CHB/ASN/g' race_1kG4.txt>race_1kG5.txt\n",
      "sed 's/CHD/ASN/g' race_1kG5.txt>race_1kG6.txt\n",
      "sed 's/YRI/AFR/g' race_1kG6.txt>race_1kG7.txt\n",
      "sed 's/LWK/AFR/g' race_1kG7.txt>race_1kG8.txt\n",
      "sed 's/TSI/EUR/g' race_1kG8.txt>race_1kG9.txt\n",
      "sed 's/MXL/AMR/g' race_1kG9.txt>race_1kG10.txt\n",
      "sed 's/GBR/EUR/g' race_1kG10.txt>race_1kG11.txt\n",
      "sed 's/FIN/EUR/g' race_1kG11.txt>race_1kG12.txt\n",
      "sed 's/CHS/ASN/g' race_1kG12.txt>race_1kG13.txt\n",
      "sed 's/PUR/AMR/g' race_1kG13.txt>race_1kG14.txt\n",
      "\n",
      "# Create a racefile of your own data.\n",
      "awk '{print$1,$2,\"OWN\"}' HapMap_MDS.fam>racefile_own.txt\n",
      "\n",
      "# Concatenate racefiles.\n",
      "cat race_1kG14.txt racefile_own.txt | sed -e '1i\\FID IID race' > racefile.txt\n",
      "\n",
      "# Generate population stratification plot.\n",
      "Rscript MDS_merged.R \n",
      "# The output file MDS.pdf demonstrates that our �own� data falls within the European group of the 1000 genomes data. Therefore, we do not have to remove subjects.\n",
      "# For educational purposes however, we give scripts below to filter out population stratification outliers. Please execute the script below in order to generate the appropriate files for the next tutorial.\n",
      "\n",
      "## Exclude ethnic outliers.\n",
      "# Select individuals in HapMap data below cut-off thresholds. The cut-off levels are not fixed thresholds but have to be determined based on the visualization of the first two dimensions. To exclude ethnic outliers, the thresholds need to be set around the cluster of population of interest.\n",
      "awk '{ if ($4 <-0.04 && $5 >0.03) print $1,$2 }' MDS_merge2.mds > EUR_MDS_merge2\n",
      "\n",
      "# Extract these individuals in HapMap data.\n",
      "plink --bfile HapMap_3_r3_12 --keep EUR_MDS_merge2 --make-bed --out HapMap_3_r3_13\n",
      "# Note, since our HapMap data did include any ethnic outliers, no individuls were removed at this step. However, if our data would have included individuals outside of the thresholds we set, then these individuals would have been removed.\n",
      "\n",
      "## Create covariates based on MDS.\n",
      "# Perform an MDS ONLY on HapMap data without ethnic outliers. The values of the 10 MDS dimensions are subsequently used as covariates in the association analysis in the third tutorial.\n",
      "plink --bfile HapMap_3_r3_13 --extract indepSNP.prune.in --genome --out HapMap_3_r3_13\n",
      "plink --bfile HapMap_3_r3_13 --read-genome HapMap_3_r3_13.genome --cluster --mds-plot 10 --out HapMap_3_r3_13_mds\n",
      "\n",
      "# Change the format of the .mds file into a plink covariate file.\n",
      "awk '{print$1, $2, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13}' HapMap_3_r3_13_mds.mds > covar_mds.txt\n",
      "\n",
      "# The values in covar_mds.txt will be used as covariates, to adjust for remaining population stratification, in the third tutorial where we will perform a genome-wide association analysis.\n",
      "\n",
      "##########################################################################################################################################################################\n",
      "\n",
      "## CONGRATULATIONS you have succesfully controlled your data for population stratification!\n",
      "\n",
      "# For the next tutorial you need the following files:\n",
      "# - HapMap_3_r3_13 (the bfile, i.e., HapMap_3_r3_13.bed,HapMap_3_r3_13.bim,and HapMap_3_r3_13.fam\n",
      "# - covar_mds.txt\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cat 2_Main_script_MDS.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be5f5ea3-cf3b-4513-9b1e-78b8a854dc61",
   "metadata": {},
   "source": [
    "### Explanation of the main script \n",
    "\n",
    "This is the main script for second tutorial from our comprehensive tutorial on GWAS and PRS.*これは、GWASとPRSに関する包括的なチュートリアルの2番目のチュートリアルのメインスクリプトです。*\n",
    "\n",
    "To run this script the following (b)files from the first tutorial are required: HapMap_3_r3_12 (this bfile contain: HapMap_3_r3_12.fam,HapMap_3_r3_12.bim, and HapMap_3_r3_12.bed; you need all three), and indepSNP.prune.in.*このスクリプトを実行するには、最初のチュートリアルの以下の(b)ファイルが必要です： HapMap_3_r3_12（このbファイルにはHapMap_3_r3_12.fam,HapMap_3_r3_12.bim,HapMap_3_r3_12.bedが含まれており、3つとも必要）、indepSNP.prune.in。*\n",
    "\n",
    "In this tutorial we are going to check for population stratification.*このチュートリアルでは、母集団の層別化をチェックします。*\n",
    "\n",
    "We will do this as follows, the bfile (HapMap_3_r3_12) generated at the end of the previous tutorial (1_QC_GWAS) is going to checked for population stratification using data from the 1000 Genomes Project. Individuals with a non-European ethnic background will be removed.*前回のチュートリアル（1_QC_GWAS）の最後に生成されたBファイル（HapMap_3_r3_12）を用いて、1000 Genomes Projectのデータを用いて集団階層性をチェックします。非ヨーロッパの民族的背景を持つ個体は取り除かれます。*\n",
    "\n",
    "Furthermore, this tutorial will generate a covariate file which helps to adust for remaining population stratification within the European subjects.*さらに、このチュートリアルでは、ヨーロッパ人対象者の中に残っている集団層別化を補正するための共変量ファイルを生成します。*\n",
    "\n",
    "In order to complete this tutorial it is necessary to have generated the bfile 'HapMap_3_r3_12' and the file 'indepSNP.prune.in' from the previous tutorial.*このチュートリアルを完了するには、前のチュートリアルで作成した bfile 'HapMap_3_r3_12' と file 'indepSNP.prune.in' が必要です。*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5c8ce32e-455b-4fa9-98af-ae5e7308c075",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p ~/hmiwa1/storage/hmiwa-lab\n",
    "mkdir -p ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38608483-72aa-496a-bbe1-5d280df25904",
   "metadata": {},
   "source": [
    "### Download 1000 Genomes data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ffc8c3a9-ef21-45f7-b3f2-7fddb4487d93",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2023-11-06 19:42:06--  ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20100804/ALL.2of4intersection.20100804.genotypes.vcf.gz\n",
      "           => ‘ALL.2of4intersection.20100804.genotypes.vcf.gz’\n",
      "Resolving ftp-trace.ncbi.nih.gov (ftp-trace.ncbi.nih.gov)... 130.14.250.11, 130.14.250.10, 2607:f220:41e:250::10, ...\n",
      "Connecting to ftp-trace.ncbi.nih.gov (ftp-trace.ncbi.nih.gov)|130.14.250.11|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /1000genomes/ftp/release/20100804 ... done.\n",
      "==> SIZE ALL.2of4intersection.20100804.genotypes.vcf.gz ... 65672092096\n",
      "==> PASV ... done.    ==> RETR ALL.2of4intersection.20100804.genotypes.vcf.gz ... done.\n",
      "Length: 65672092096 (61G) (unauthoritative)\n",
      "\n",
      "ALL.2of4intersectio 100%[===================>]  61.16G  17.7MB/s    in 60m 34s \n",
      "\n",
      "2023-11-06 20:42:42 (17.2 MB/s) - ‘ALL.2of4intersection.20100804.genotypes.vcf.gz’ saved [65672092096]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# This file from the 1000 Genomes contains genetic data of 629 individuals from different ethnic backgrounds.\n",
    "# Note, this file is quite large (>60 gigabyte).  \n",
    "cd ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS\n",
    "wget ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20100804/ALL.2of4intersection.20100804.genotypes.vcf.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f7bf3c82-bc36-481b-a9e5-8e1a96b6e26c",
   "metadata": {},
   "source": [
    "### Convert vcf to Plink format."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6b8d6677-456f-49ef-8273-eeed90942f08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.2of4intersection.20100804.genotypes.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.2of4intersection.20100804.genotypes\n",
      "  --vcf ALL.2of4intersection.20100804.genotypes.vcf.gz\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "--vcf: ALL.2of4intersection.20100804.genotypes-temporary.bed +\n",
      "ALL.2of4intersection.20100804.genotypes-temporary.bim +\n",
      "ALL.2of4intersection.20100804.genotypes-temporary.fam written.\n",
      "25488488 variants loaded from .bim file.\n",
      "629 people (0 males, 0 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.2of4intersection.20100804.genotypes.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 629 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.615305.\n",
      "25488488 variants and 629 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.2of4intersection.20100804.genotypes.bed +\n",
      "ALL.2of4intersection.20100804.genotypes.bim +\n",
      "ALL.2of4intersection.20100804.genotypes.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "cd ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS\n",
    "# Convert vcf to Plink format.\n",
    "plink --vcf ALL.2of4intersection.20100804.genotypes.vcf.gz --make-bed --out ALL.2of4intersection.20100804.genotypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "14bfaf8e-d980-4510-9c38-d9c705a66843",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.2of4intersection.20100804.genotypes_no_missing_IDs.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.2of4intersection.20100804.genotypes\n",
      "  --make-bed\n",
      "  --out ALL.2of4intersection.20100804.genotypes_no_missing_IDs\n",
      "  --set-missing-var-ids @:#[b37]$1,$2\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "25488488 variants loaded from .bim file.\n",
      "10375501 missing IDs set.\n",
      "629 people (0 males, 0 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "ALL.2of4intersection.20100804.genotypes_no_missing_IDs.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 629 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.615305.\n",
      "25488488 variants and 629 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.2of4intersection.20100804.genotypes_no_missing_IDs.bed +\n",
      "ALL.2of4intersection.20100804.genotypes_no_missing_IDs.bim +\n",
      "ALL.2of4intersection.20100804.genotypes_no_missing_IDs.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# 注目すべきは、ファイル'ALL.2of4intersection.20100804.genotypes.bim'にはrs-identifierを持たないSNPが含まれていることである。これらのSNPは\". \"で示されている。\n",
    "# これはファイル'ALL.2of4intersection.20100804.genotypes.vcf.gz'でも確認できる。\n",
    "# このファイルを確認するには、次のコマンドを使用します: zmore ALL.2of4intersection.20100804.genotypes.vcf.gz .\n",
    "# このチュートリアルでは、1000 Genomesデータのrs-identifierの欠落は問題にならない。\n",
    "# しかし、良い練習のために、rs-identifierが欠損しているSNP（つまり、\". \"が付いているSNP）にユニークなindentifierを割り当てます。\n",
    "plink --bfile ALL.2of4intersection.20100804.genotypes --set-missing-var-ids @:#[b37]\\$1,\\$2 --make-bed --out ALL.2of4intersection.20100804.genotypes_no_missing_IDs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5e17a04e-c948-4390-b831-ae7dbe557125",
   "metadata": {},
   "source": [
    "### QC on 1000 Genomes data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8599e9e3-9a3b-44f8-a2ae-ff2095ef30e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 1kG_MDS.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile ALL.2of4intersection.20100804.genotypes_no_missing_IDs\n",
      "  --geno 0.2\n",
      "  --make-bed\n",
      "  --out 1kG_MDS\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "25488488 variants loaded from .bim file.\n",
      "629 people (0 males, 0 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to 1kG_MDS.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 629 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.615305.\n",
      "16481066 variants removed due to missing genotype data (--geno).\n",
      "9007422 variants and 629 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to 1kG_MDS.bed + 1kG_MDS.bim + 1kG_MDS.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "## QC on 1000 Genomes data.\n",
    "cd ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS\n",
    "# Remove variants based on missing genotype data.\n",
    "plink --bfile ALL.2of4intersection.20100804.genotypes_no_missing_IDs --geno 0.2 --allow-no-sex --make-bed --out 1kG_MDS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ff7de94c-b803-4377-8f24-0e1823e4a2b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 1kG_MDS2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile 1kG_MDS\n",
      "  --make-bed\n",
      "  --mind 0.2\n",
      "  --out 1kG_MDS2\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "9007422 variants loaded from .bim file.\n",
      "629 people (0 males, 0 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to 1kG_MDS2.nosex .\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 629 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.995911.\n",
      "9007422 variants and 629 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to 1kG_MDS2.bed + 1kG_MDS2.bim + 1kG_MDS2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# Remove individuals based on missing genotype data.\n",
    "plink --bfile 1kG_MDS --mind 0.2 --allow-no-sex --make-bed --out 1kG_MDS2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5983fb4e-21b0-4e6e-9f5f-0ed9f3711135",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 1kG_MDS3.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile 1kG_MDS2\n",
      "  --geno 0.02\n",
      "  --make-bed\n",
      "  --out 1kG_MDS3\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "9007422 variants loaded from .bim file.\n",
      "629 people (0 males, 0 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to 1kG_MDS3.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 629 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.995911.\n",
      "766677 variants removed due to missing genotype data (--geno).\n",
      "8240745 variants and 629 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to 1kG_MDS3.bed + 1kG_MDS3.bim + 1kG_MDS3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# Remove variants based on missing genotype data.\n",
    "plink --bfile 1kG_MDS2 --geno 0.02 --allow-no-sex --make-bed --out 1kG_MDS3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "27d8bee3-4385-4e38-a13e-cba43f115e06",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 1kG_MDS4.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile 1kG_MDS3\n",
      "  --make-bed\n",
      "  --mind 0.02\n",
      "  --out 1kG_MDS4\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "8240745 variants loaded from .bim file.\n",
      "629 people (0 males, 0 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to 1kG_MDS4.nosex .\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 629 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999477.\n",
      "8240745 variants and 629 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to 1kG_MDS4.bed + 1kG_MDS4.bim + 1kG_MDS4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# Remove individuals based on missing genotype data.\n",
    "plink --bfile 1kG_MDS3 --mind 0.02 --allow-no-sex --make-bed --out 1kG_MDS4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "35091502-be1c-4fad-8d1b-e530dfa4a255",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 1kG_MDS5.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile 1kG_MDS4\n",
      "  --maf 0.05\n",
      "  --make-bed\n",
      "  --out 1kG_MDS5\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "8240745 variants loaded from .bim file.\n",
      "629 people (0 males, 0 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to 1kG_MDS5.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 629 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999477.\n",
      "2432435 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "5808310 variants and 629 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to 1kG_MDS5.bed + 1kG_MDS5.bim + 1kG_MDS5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# Remove variants based on MAF.\n",
    "plink --bfile 1kG_MDS4 --maf 0.05 --allow-no-sex --make-bed --out 1kG_MDS5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "2817ace0-50bf-4dfb-8fb0-a18452b64911",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 1kG_MDS6.log.\n",
      "Options in effect:\n",
      "  --bfile 1kG_MDS5\n",
      "  --extract HapMap_SNPs.txt\n",
      "  --make-bed\n",
      "  --out 1kG_MDS6\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "5808310 variants loaded from .bim file.\n",
      "629 people (0 males, 0 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to 1kG_MDS6.nosex .\n",
      "--extract: 1000993 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 629 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999411.\n",
      "1000993 variants and 629 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to 1kG_MDS6.bed + 1kG_MDS6.bim + 1kG_MDS6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# Extract the variants present in HapMap dataset from the 1000 genomes dataset.\n",
    "awk '{print$2}' HapMap_3_r3_12.bim > HapMap_SNPs.txt\n",
    "plink --bfile 1kG_MDS5 --extract HapMap_SNPs.txt --make-bed --out 1kG_MDS6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "3fa619dc-971b-421f-ba53-204ab5c3e221",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_MDS.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_12\n",
      "  --extract 1kG_MDS6_SNPs.txt\n",
      "  --make-bed\n",
      "  --out HapMap_MDS\n",
      "  --recode\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "--extract: 1000993 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998036.\n",
      "1000993 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "--make-bed to HapMap_MDS.bed + HapMap_MDS.bim + HapMap_MDS.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--recode ped to HapMap_MDS.ped + HapMap_MDS.map ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# Extract the variants present in 1000 Genomes dataset from the HapMap dataset.\n",
    "awk '{print$2}' 1kG_MDS6.bim > 1kG_MDS6_SNPs.txt\n",
    "plink --bfile HapMap_3_r3_12 --extract 1kG_MDS6_SNPs.txt --recode --make-bed --out HapMap_MDS\n",
    "# The datasets now contain the exact same variants."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "159c61dd-26d6-47f6-95ee-0fb7a6aed6e3",
   "metadata": {},
   "source": [
    "### The datasets must have the same build. Change the build 1000 Genomes data build."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f41723e1-1bed-458e-b141-01d6ddfb886a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 1kG_MDS7.log.\n",
      "Options in effect:\n",
      "  --bfile 1kG_MDS6\n",
      "  --make-bed\n",
      "  --out 1kG_MDS7\n",
      "  --update-map buildhapmap.txt\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1000993 variants loaded from .bim file.\n",
      "629 people (0 males, 0 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to 1kG_MDS7.nosex .\n",
      "--update-map: 1000993 values updated.\n",
      "Warning: Base-pair positions are now unsorted!\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 629 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999411.\n",
      "1000993 variants and 629 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to 1kG_MDS7.bed + 1kG_MDS7.bim + 1kG_MDS7.fam ... 1010111112121313141415151616171718181919202021212222232324242525262627272828292930303131323233333434353536363737383839394040414142424343444445454646474748484949505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n"
     ]
    }
   ],
   "source": [
    "## The datasets must have the same build. Change the build 1000 Genomes data build.\n",
    "awk '{print$2,$4}' HapMap_MDS.map > buildhapmap.txt\n",
    "# buildhapmap.txt contains one SNP-id and physical position per line.\n",
    "\n",
    "plink --bfile 1kG_MDS6 --update-map buildhapmap.txt --make-bed --out 1kG_MDS7\n",
    "# 1kG_MDS7 and HapMap_MDS now have the same build."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a763a161-64f4-49b8-ada0-99ff81001d7c",
   "metadata": {},
   "source": [
    "### Merge the HapMap and 1000 Genomes data sets\n",
    "\n",
    "Prior to merging 1000 Genomes data with the HapMap data we want to make sure that the files are mergeable, for this we conduct 3 steps:\n",
    "1. Make sure the reference genome is similar in the HapMap and the 1000 Genomes Project datasets.\n",
    "2. Resolve strand issues.\n",
    "3. Remove the SNPs which after the previous two steps still differ between datasets.\n",
    "\n",
    "The following steps are maybe quite technical in terms of commands, but we just compare the two data sets and make sure they correspond.\n",
    "\n",
    "*1000ゲノムのデータとHapMapのデータをマージする前に、ファイルがマージ可能であることを確認したい*\n",
    "1. *参照ゲノムがHapMapと1000ゲノムプロジェクトのデータセットで類似していることを確認する。*\n",
    "2. *鎖の問題を解決する。*\n",
    "3. *前の2つのステップの後でもデータセット間で異なるSNPを削除する。*\n",
    "\n",
    "*以下のステップはコマンド的にはかなり専門的かもしれないが、2つのデータセットを比較し、対応していることを確認するだけである。*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "115e14f9-50c0-4c73-848d-aa158d6367ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap-adj.log.\n",
      "Options in effect:\n",
      "  --a1-allele 1kg_ref-list.txt\n",
      "  --bfile HapMap_MDS\n",
      "  --make-bed\n",
      "  --out HapMap-adj\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1000993 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998036.\n",
      "Warning: Impossible A1 allele assignment for variant rs11488462.\n",
      "Warning: Impossible A1 allele assignment for variant rs4648786.\n",
      "Warning: Impossible A1 allele assignment for variant rs28635343.\n",
      "Warning: Impossible A1 allele assignment for variant rs28456011.\n",
      "Warning: Impossible A1 allele assignment for variant rs4074196.\n",
      "Warning: Impossible A1 allele assignment for variant rs4074197.\n",
      "Warning: Impossible A1 allele assignment for variant rs7520934.\n",
      "Warning: Impossible A1 allele assignment for variant rs28487995.\n",
      "Warning: Impossible A1 allele assignment for variant rs760925.\n",
      "Warning: Impossible A1 allele assignment for variant rs4648764.\n",
      "Warning: Impossible A1 allele assignment for variant rs2495365.\n",
      "Warning: Impossible A1 allele assignment for variant rs2234167.\n",
      "Warning: Impossible A1 allele assignment for variant rs2281852.\n",
      "Warning: Impossible A1 allele assignment for variant rs2257763.\n",
      "Warning: Impossible A1 allele assignment for variant rs1886730.\n",
      "Warning: Impossible A1 allele assignment for variant rs35383955.\n",
      "Warning: Impossible A1 allele assignment for variant rs34283457.\n",
      "Warning: Impossible A1 allele assignment for variant rs35357981.\n",
      "Warning: Impossible A1 allele assignment for variant rs35025175.\n",
      "Warning: Impossible A1 allele assignment for variant rs35974482.\n",
      "Warning: Impossible A1 allele assignment for variant rs34835780.\n",
      "Warning: Impossible A1 allele assignment for variant rs10753374.\n",
      "Warning: Impossible A1 allele assignment for variant rs11260657.\n",
      "Warning: Impossible A1 allele assignment for variant rs4661727.\n",
      "Warning: Impossible A1 allele assignment for variant rs1048977.\n",
      "Warning: Impossible A1 allele assignment for variant rs6699113.\n",
      "Warning: Impossible A1 allele assignment for variant rs35614701.\n",
      "Warning: Impossible A1 allele assignment for variant rs35431118.\n",
      "Warning: Impossible A1 allele assignment for variant rs35589882.\n",
      "Warning: Impossible A1 allele assignment for variant rs35543461.\n",
      "Warning: Impossible A1 allele assignment for variant rs35886763.\n",
      "Warning: Impossible A1 allele assignment for variant rs35941602.\n",
      "Warning: Impossible A1 allele assignment for variant rs34131388.\n",
      "Warning: Impossible A1 allele assignment for variant rs35260034.\n",
      "Warning: Impossible A1 allele assignment for variant rs34201264.\n",
      "Warning: Impossible A1 allele assignment for variant rs34734086.\n",
      "Warning: Impossible A1 allele assignment for variant rs35593799.\n",
      "Warning: Impossible A1 allele assignment for variant rs34288882.\n",
      "Warning: Impossible A1 allele assignment for variant rs34697248.\n",
      "Warning: Impossible A1 allele assignment for variant rs35107626.\n",
      "Warning: Impossible A1 allele assignment for variant rs35047308.\n",
      "Warning: Impossible A1 allele assignment for variant rs36053581.\n",
      "Warning: Impossible A1 allele assignment for variant rs34405972.\n",
      "Warning: Impossible A1 allele assignment for variant rs35294772.\n",
      "Warning: Impossible A1 allele assignment for variant rs616546.\n",
      "Warning: Impossible A1 allele assignment for variant rs12118489.\n",
      "Warning: Impossible A1 allele assignment for variant rs1023098.\n",
      "Warning: Impossible A1 allele assignment for variant rs2182022.\n",
      "Warning: Impossible A1 allele assignment for variant rs209603.\n",
      "Warning: Impossible A1 allele assignment for variant rs10524.\n",
      "Warning: Impossible A1 allele assignment for variant rs12026359.\n",
      "Warning: Impossible A1 allele assignment for variant rs4926805.\n",
      "Warning: Impossible A1 allele assignment for variant rs2294515.\n",
      "Warning: Impossible A1 allele assignment for variant rs697594.\n",
      "Warning: Impossible A1 allele assignment for variant rs2275859.\n",
      "Warning: Impossible A1 allele assignment for variant rs3762317.\n",
      "Warning: Impossible A1 allele assignment for variant rs3766265.\n",
      "Warning: Impossible A1 allele assignment for variant rs1889603.\n",
      "Warning: Impossible A1 allele assignment for variant rs10493486.\n",
      "Warning: Impossible A1 allele assignment for variant rs12145114.\n",
      "Warning: Impossible A1 allele assignment for variant rs2391154.\n",
      "Warning: Impossible A1 allele assignment for variant rs718873.\n",
      "Warning: Impossible A1 allele assignment for variant rs17123142.\n",
      "Warning: Impossible A1 allele assignment for variant rs1688597.\n",
      "Warning: Impossible A1 allele assignment for variant rs12125356.\n",
      "Warning: Impossible A1 allele assignment for variant rs4914959.\n",
      "Warning: Impossible A1 allele assignment for variant rs13303136.\n",
      "Warning: Impossible A1 allele assignment for variant rs17663802.\n",
      "Warning: Impossible A1 allele assignment for variant rs3790604.\n",
      "Warning: Impossible A1 allele assignment for variant rs12753188.\n",
      "Warning: Impossible A1 allele assignment for variant rs699714.\n",
      "Warning: Impossible A1 allele assignment for variant rs3765501.\n",
      "Warning: Impossible A1 allele assignment for variant rs1011297.\n",
      "Warning: Impossible A1 allele assignment for variant rs17723260.\n",
      "Warning: Impossible A1 allele assignment for variant rs34444588.\n",
      "Warning: Impossible A1 allele assignment for variant rs6656168.\n",
      "Warning: Impossible A1 allele assignment for variant rs11591079.\n",
      "Warning: Impossible A1 allele assignment for variant rs3813622.\n",
      "Warning: Impossible A1 allele assignment for variant rs509587.\n",
      "Warning: Impossible A1 allele assignment for variant rs12097632.\n",
      "Warning: Impossible A1 allele assignment for variant rs3117838.\n",
      "Warning: Impossible A1 allele assignment for variant rs28927681.\n",
      "Warning: Impossible A1 allele assignment for variant rs3001156.\n",
      "Warning: Impossible A1 allele assignment for variant rs12038043.\n",
      "Warning: Impossible A1 allele assignment for variant rs1934663.\n",
      "Warning: Impossible A1 allele assignment for variant rs2293335.\n",
      "Warning: Impossible A1 allele assignment for variant rs875563.\n",
      "Warning: Impossible A1 allele assignment for variant rs34229647.\n",
      "Warning: Impossible A1 allele assignment for variant rs6680730.\n",
      "Warning: Impossible A1 allele assignment for variant rs4846792.\n",
      "Warning: Impossible A1 allele assignment for variant rs28675156.\n",
      "Warning: Impossible A1 allele assignment for variant rs4846796.\n",
      "Warning: Impossible A1 allele assignment for variant rs17701857.\n",
      "Warning: Impossible A1 allele assignment for variant rs4134452.\n",
      "Warning: Impossible A1 allele assignment for variant rs6604997.\n",
      "Warning: Impossible A1 allele assignment for variant rs3795743.\n",
      "Warning: Impossible A1 allele assignment for variant rs34158997.\n",
      "Warning: Impossible A1 allele assignment for variant rs35808016.\n",
      "Warning: Impossible A1 allele assignment for variant rs35164209.\n",
      "Warning: Impossible A1 allele assignment for variant rs34900680.\n",
      "Warning: Impossible A1 allele assignment for variant rs34187970.\n",
      "Warning: Impossible A1 allele assignment for variant rs1417766.\n",
      "Warning: Impossible A1 allele assignment for variant rs3820624.\n",
      "Warning: Impossible A1 allele assignment for variant rs7535031.\n",
      "Warning: Impossible A1 allele assignment for variant rs34578818.\n",
      "Warning: Impossible A1 allele assignment for variant rs2897454.\n",
      "Warning: Impossible A1 allele assignment for variant rs1127975.\n",
      "Warning: Impossible A1 allele assignment for variant rs35729322.\n",
      "Warning: Impossible A1 allele assignment for variant rs35565142.\n",
      "Warning: Impossible A1 allele assignment for variant rs34360045.\n",
      "Warning: Impossible A1 allele assignment for variant rs34099250.\n",
      "Warning: Impossible A1 allele assignment for variant rs4659822.\n",
      "Warning: Impossible A1 allele assignment for variant rs4329572.\n",
      "Warning: Impossible A1 allele assignment for variant rs34003373.\n",
      "Warning: Impossible A1 allele assignment for variant rs2273741.\n",
      "Warning: Impossible A1 allele assignment for variant rs1252166.\n",
      "Warning: Impossible A1 allele assignment for variant rs16840285.\n",
      "Warning: Impossible A1 allele assignment for variant rs3007405.\n",
      "Warning: Impossible A1 allele assignment for variant rs11553746.\n",
      "Warning: Impossible A1 allele assignment for variant rs10495530.\n",
      "Warning: Impossible A1 allele assignment for variant rs2303917.\n",
      "Warning: Impossible A1 allele assignment for variant rs1384491.\n",
      "Warning: Impossible A1 allele assignment for variant rs17045393.\n",
      "Warning: Impossible A1 allele assignment for variant rs6545238.\n",
      "Warning: Impossible A1 allele assignment for variant rs17792521.\n",
      "Warning: Impossible A1 allele assignment for variant rs989586.\n",
      "Warning: Impossible A1 allele assignment for variant rs7421835.\n",
      "Warning: Impossible A1 allele assignment for variant rs28687.\n",
      "Warning: Impossible A1 allele assignment for variant rs1989614.\n",
      "Warning: Impossible A1 allele assignment for variant rs11680149.\n",
      "Warning: Impossible A1 allele assignment for variant rs7558635.\n",
      "Warning: Impossible A1 allele assignment for variant rs2276654.\n",
      "Warning: Impossible A1 allele assignment for variant rs1139829.\n",
      "Warning: Impossible A1 allele assignment for variant rs35061433.\n",
      "Warning: Impossible A1 allele assignment for variant rs4075701.\n",
      "Warning: Impossible A1 allele assignment for variant rs976297.\n",
      "Warning: Impossible A1 allele assignment for variant rs838069.\n",
      "Warning: Impossible A1 allele assignment for variant rs3980213.\n",
      "Warning: Impossible A1 allele assignment for variant rs11542416.\n",
      "Warning: Impossible A1 allele assignment for variant rs16826050.\n",
      "Warning: Impossible A1 allele assignment for variant rs11883933.\n",
      "Warning: Impossible A1 allele assignment for variant rs16823475.\n",
      "Warning: Impossible A1 allele assignment for variant rs967381.\n",
      "Warning: Impossible A1 allele assignment for variant rs4561615.\n",
      "Warning: Impossible A1 allele assignment for variant rs17255817.\n",
      "Warning: Impossible A1 allele assignment for variant rs3755490.\n",
      "Warning: Impossible A1 allele assignment for variant rs7557843.\n",
      "Warning: Impossible A1 allele assignment for variant rs13393016.\n",
      "Warning: Impossible A1 allele assignment for variant rs13003362.\n",
      "Warning: Impossible A1 allele assignment for variant rs7565312.\n",
      "Warning: Impossible A1 allele assignment for variant rs17180544.\n",
      "Warning: Impossible A1 allele assignment for variant rs13417895.\n",
      "Warning: Impossible A1 allele assignment for variant rs7419666.\n",
      "Warning: Impossible A1 allele assignment for variant rs12467878.\n",
      "Warning: Impossible A1 allele assignment for variant rs6726184.\n",
      "Warning: Impossible A1 allele assignment for variant rs17862867.\n",
      "Warning: Impossible A1 allele assignment for variant rs3771341.\n",
      "Warning: Impossible A1 allele assignment for variant rs4129104.\n",
      "Warning: Impossible A1 allele assignment for variant rs3791580.\n",
      "Warning: Impossible A1 allele assignment for variant rs34980723.\n",
      "Warning: Impossible A1 allele assignment for variant rs35399295.\n",
      "Warning: Impossible A1 allele assignment for variant rs4072221.\n",
      "Warning: Impossible A1 allele assignment for variant rs4973697.\n",
      "Warning: Impossible A1 allele assignment for variant rs6605267.\n",
      "Warning: Impossible A1 allele assignment for variant rs7425592.\n",
      "Warning: Impossible A1 allele assignment for variant rs7421596.\n",
      "Warning: Impossible A1 allele assignment for variant rs6605275.\n",
      "Warning: Impossible A1 allele assignment for variant rs167463.\n",
      "Warning: Impossible A1 allele assignment for variant rs3792491.\n",
      "Warning: Impossible A1 allele assignment for variant rs2121906.\n",
      "Warning: Impossible A1 allele assignment for variant rs1689581.\n",
      "Warning: Impossible A1 allele assignment for variant rs1021937.\n",
      "Warning: Impossible A1 allele assignment for variant rs711730.\n",
      "Warning: Impossible A1 allele assignment for variant rs6778759.\n",
      "Warning: Impossible A1 allele assignment for variant rs13075520.\n",
      "Warning: Impossible A1 allele assignment for variant rs11130263.\n",
      "Warning: Impossible A1 allele assignment for variant rs13093798.\n",
      "Warning: Impossible A1 allele assignment for variant rs11928389.\n",
      "Warning: Impossible A1 allele assignment for variant rs7652667.\n",
      "Warning: Impossible A1 allele assignment for variant rs9864701.\n",
      "Warning: Impossible A1 allele assignment for variant rs1078849.\n",
      "Warning: Impossible A1 allele assignment for variant rs12489749.\n",
      "Warning: Impossible A1 allele assignment for variant rs17634797.\n",
      "Warning: Impossible A1 allele assignment for variant rs1498840.\n",
      "Warning: Impossible A1 allele assignment for variant rs17280613.\n",
      "Warning: Impossible A1 allele assignment for variant rs277646.\n",
      "Warning: Impossible A1 allele assignment for variant rs1165919.\n",
      "Warning: Impossible A1 allele assignment for variant rs1163448.\n",
      "Warning: Impossible A1 allele assignment for variant rs1259321.\n",
      "Warning: Impossible A1 allele assignment for variant rs1259320.\n",
      "Warning: Impossible A1 allele assignment for variant rs1136995.\n",
      "Warning: Impossible A1 allele assignment for variant rs6765489.\n",
      "Warning: Impossible A1 allele assignment for variant rs2084400.\n",
      "Warning: Impossible A1 allele assignment for variant rs1113277.\n",
      "Warning: Impossible A1 allele assignment for variant rs9880098.\n",
      "Warning: Impossible A1 allele assignment for variant rs11919230.\n",
      "Warning: Impossible A1 allele assignment for variant rs17372345.\n",
      "Warning: Impossible A1 allele assignment for variant rs7611483.\n",
      "Warning: Impossible A1 allele assignment for variant rs7615304.\n",
      "Warning: Impossible A1 allele assignment for variant rs10936388.\n",
      "Warning: Impossible A1 allele assignment for variant rs7624210.\n",
      "Warning: Impossible A1 allele assignment for variant rs906721.\n",
      "Warning: Impossible A1 allele assignment for variant rs11546878.\n",
      "Warning: Impossible A1 allele assignment for variant rs3774288.\n",
      "Warning: Impossible A1 allele assignment for variant rs4686566.\n",
      "Warning: Impossible A1 allele assignment for variant rs884309.\n",
      "Warning: Impossible A1 allele assignment for variant rs3806620.\n",
      "Warning: Impossible A1 allele assignment for variant rs6801044.\n",
      "Warning: Impossible A1 allele assignment for variant rs9886989.\n",
      "Warning: Impossible A1 allele assignment for variant rs4677689.\n",
      "Warning: Impossible A1 allele assignment for variant rs6605338.\n",
      "Warning: Impossible A1 allele assignment for variant rs7627129.\n",
      "Warning: Impossible A1 allele assignment for variant rs4677695.\n",
      "Warning: Impossible A1 allele assignment for variant rs4677617.\n",
      "Warning: Impossible A1 allele assignment for variant rs9820715.\n",
      "Warning: Impossible A1 allele assignment for variant rs403820.\n",
      "Warning: Impossible A1 allele assignment for variant rs36056866.\n",
      "Warning: Impossible A1 allele assignment for variant rs35840880.\n",
      "Warning: Impossible A1 allele assignment for variant rs35792778.\n",
      "Warning: Impossible A1 allele assignment for variant rs6847677.\n",
      "Warning: Impossible A1 allele assignment for variant rs7682902.\n",
      "Warning: Impossible A1 allele assignment for variant rs16854250.\n",
      "Warning: Impossible A1 allele assignment for variant rs12646999.\n",
      "Warning: Impossible A1 allele assignment for variant rs13143552.\n",
      "Warning: Impossible A1 allele assignment for variant rs4860941.\n",
      "Warning: Impossible A1 allele assignment for variant rs4860942.\n",
      "Warning: Impossible A1 allele assignment for variant rs13128530.\n",
      "Warning: Impossible A1 allele assignment for variant rs17148495.\n",
      "Warning: Impossible A1 allele assignment for variant rs34789936.\n",
      "Warning: Impossible A1 allele assignment for variant rs2949614.\n",
      "Warning: Impossible A1 allele assignment for variant rs17007758.\n",
      "Warning: Impossible A1 allele assignment for variant rs10006274.\n",
      "Warning: Impossible A1 allele assignment for variant rs7673834.\n",
      "Warning: Impossible A1 allele assignment for variant rs17047901.\n",
      "Warning: Impossible A1 allele assignment for variant rs7667092.\n",
      "Warning: Impossible A1 allele assignment for variant rs11737164.\n",
      "Warning: Impossible A1 allele assignment for variant rs4691482.\n",
      "Warning: Impossible A1 allele assignment for variant rs17249941.\n",
      "Warning: Impossible A1 allele assignment for variant rs3187025.\n",
      "Warning: Impossible A1 allele assignment for variant rs10060593.\n",
      "Warning: Impossible A1 allele assignment for variant rs1215429.\n",
      "Warning: Impossible A1 allele assignment for variant rs1395093.\n",
      "Warning: Impossible A1 allele assignment for variant rs11744397.\n",
      "Warning: Impossible A1 allele assignment for variant rs1848990.\n",
      "Warning: Impossible A1 allele assignment for variant rs770308.\n",
      "Warning: Impossible A1 allele assignment for variant rs809987.\n",
      "Warning: Impossible A1 allele assignment for variant rs2972819.\n",
      "Warning: Impossible A1 allele assignment for variant rs199361.\n",
      "Warning: Impossible A1 allele assignment for variant rs4403186.\n",
      "Warning: Impossible A1 allele assignment for variant rs1574436.\n",
      "Warning: Impossible A1 allele assignment for variant rs4481332.\n",
      "Warning: Impossible A1 allele assignment for variant rs1048944.\n",
      "Warning: Impossible A1 allele assignment for variant rs1605683.\n",
      "Warning: Impossible A1 allele assignment for variant rs16869031.\n",
      "Warning: Impossible A1 allele assignment for variant rs17397864.\n",
      "Warning: Impossible A1 allele assignment for variant rs12153556.\n",
      "Warning: Impossible A1 allele assignment for variant rs831859.\n",
      "Warning: Impossible A1 allele assignment for variant rs153378.\n",
      "Warning: Impossible A1 allele assignment for variant rs10514932.\n",
      "Warning: Impossible A1 allele assignment for variant rs17830886.\n",
      "Warning: Impossible A1 allele assignment for variant rs160980.\n",
      "Warning: Impossible A1 allele assignment for variant rs4704197.\n",
      "Warning: Impossible A1 allele assignment for variant rs16872521.\n",
      "Warning: Impossible A1 allele assignment for variant rs3797418.\n",
      "Warning: Impossible A1 allele assignment for variant rs173545.\n",
      "Warning: Impossible A1 allele assignment for variant rs10746.\n",
      "Warning: Impossible A1 allele assignment for variant rs2060424.\n",
      "Warning: Impossible A1 allele assignment for variant rs2218903.\n",
      "Warning: Impossible A1 allele assignment for variant rs7710010.\n",
      "Warning: Impossible A1 allele assignment for variant rs35540.\n",
      "Warning: Impossible A1 allele assignment for variant rs17285400.\n",
      "Warning: Impossible A1 allele assignment for variant rs449359.\n",
      "Warning: Impossible A1 allele assignment for variant rs11746705.\n",
      "Warning: Impossible A1 allele assignment for variant rs2278261.\n",
      "Warning: Impossible A1 allele assignment for variant rs7448017.\n",
      "Warning: Impossible A1 allele assignment for variant rs2073839.\n",
      "Warning: Impossible A1 allele assignment for variant rs17839786.\n",
      "Warning: Impossible A1 allele assignment for variant rs702380.\n",
      "Warning: Impossible A1 allele assignment for variant rs2636096.\n",
      "Warning: Impossible A1 allele assignment for variant rs13180237.\n",
      "Warning: Impossible A1 allele assignment for variant rs17104437.\n",
      "Warning: Impossible A1 allele assignment for variant rs17776554.\n",
      "Warning: Impossible A1 allele assignment for variant rs3733666.\n",
      "Warning: Impossible A1 allele assignment for variant rs190302.\n",
      "Warning: Impossible A1 allele assignment for variant rs265995.\n",
      "Warning: Impossible A1 allele assignment for variant rs265999.\n",
      "Warning: Impossible A1 allele assignment for variant rs266000.\n",
      "Warning: Impossible A1 allele assignment for variant rs4868581.\n",
      "Warning: Impossible A1 allele assignment for variant rs413948.\n",
      "Warning: Impossible A1 allele assignment for variant rs4454070.\n",
      "Warning: Impossible A1 allele assignment for variant rs723063.\n",
      "Warning: Impossible A1 allele assignment for variant rs4710897.\n",
      "Warning: Impossible A1 allele assignment for variant rs3799369.\n",
      "Warning: Impossible A1 allele assignment for variant rs9461653.\n",
      "Warning: Impossible A1 allele assignment for variant rs9268326.\n",
      "Warning: Impossible A1 allele assignment for variant rs3129945.\n",
      "Warning: Impossible A1 allele assignment for variant rs1129765.\n",
      "Warning: Impossible A1 allele assignment for variant rs12207915.\n",
      "Warning: Impossible A1 allele assignment for variant rs4565302.\n",
      "Warning: Impossible A1 allele assignment for variant rs970392.\n",
      "Warning: Impossible A1 allele assignment for variant rs1938204.\n",
      "Warning: Impossible A1 allele assignment for variant rs2075967.\n",
      "Warning: Impossible A1 allele assignment for variant rs573684.\n",
      "Warning: Impossible A1 allele assignment for variant rs9404042.\n",
      "Warning: Impossible A1 allele assignment for variant rs9373596.\n",
      "Warning: Impossible A1 allele assignment for variant rs10872644.\n",
      "Warning: Impossible A1 allele assignment for variant rs3816304.\n",
      "Warning: Impossible A1 allele assignment for variant rs1099663.\n",
      "Warning: Impossible A1 allele assignment for variant rs3924019.\n",
      "Warning: Impossible A1 allele assignment for variant rs9690447.\n",
      "Warning: Impossible A1 allele assignment for variant rs9800958.\n",
      "Warning: Impossible A1 allele assignment for variant rs9719226.\n",
      "Warning: Impossible A1 allele assignment for variant rs9690480.\n",
      "Warning: Impossible A1 allele assignment for variant rs28617446.\n",
      "Warning: Impossible A1 allele assignment for variant rs4721782.\n",
      "Warning: Impossible A1 allele assignment for variant rs4397302.\n",
      "Warning: Impossible A1 allele assignment for variant rs11559146.\n",
      "Warning: Impossible A1 allele assignment for variant rs217600.\n",
      "Warning: Impossible A1 allele assignment for variant rs2961253.\n",
      "Warning: Impossible A1 allele assignment for variant rs2428430.\n",
      "Warning: Impossible A1 allele assignment for variant rs765133.\n",
      "Warning: Impossible A1 allele assignment for variant rs2247995.\n",
      "Warning: Impossible A1 allele assignment for variant rs2268307.\n",
      "Warning: Impossible A1 allele assignment for variant rs35735865.\n",
      "Warning: Impossible A1 allele assignment for variant rs3793345.\n",
      "Warning: Impossible A1 allele assignment for variant rs35416119.\n",
      "Warning: Impossible A1 allele assignment for variant rs1059309.\n",
      "Warning: Impossible A1 allele assignment for variant rs4870666.\n",
      "Warning: Impossible A1 allele assignment for variant rs799509.\n",
      "Warning: Impossible A1 allele assignment for variant rs4718270.\n",
      "Warning: Impossible A1 allele assignment for variant rs4149468.\n",
      "Warning: Impossible A1 allele assignment for variant rs13221920.\n",
      "Warning: Impossible A1 allele assignment for variant rs7807263.\n",
      "Warning: Impossible A1 allele assignment for variant rs1043987.\n",
      "Warning: Impossible A1 allele assignment for variant rs7786339.\n",
      "Warning: Impossible A1 allele assignment for variant rs2105511.\n",
      "Warning: Impossible A1 allele assignment for variant rs7050.\n",
      "Warning: Impossible A1 allele assignment for variant rs2068338.\n",
      "Warning: Impossible A1 allele assignment for variant rs3823899.\n",
      "Warning: Impossible A1 allele assignment for variant rs173720.\n",
      "Warning: Impossible A1 allele assignment for variant rs17166357.\n",
      "Warning: Impossible A1 allele assignment for variant rs17305795.\n",
      "Warning: Impossible A1 allele assignment for variant rs4729249.\n",
      "Warning: Impossible A1 allele assignment for variant rs1640558.\n",
      "Warning: Impossible A1 allele assignment for variant rs2283017.\n",
      "Warning: Impossible A1 allele assignment for variant rs4463351.\n",
      "Warning: Impossible A1 allele assignment for variant rs10487171.\n",
      "Warning: Impossible A1 allele assignment for variant rs11984204.\n",
      "Warning: Impossible A1 allele assignment for variant rs2402634.\n",
      "Warning: Impossible A1 allele assignment for variant rs2566879.\n",
      "Warning: Impossible A1 allele assignment for variant rs3778799.\n",
      "Warning: Impossible A1 allele assignment for variant rs949425.\n",
      "Warning: Impossible A1 allele assignment for variant rs2854536.\n",
      "Warning: Impossible A1 allele assignment for variant rs2854540.\n",
      "Warning: Impossible A1 allele assignment for variant rs2854541.\n",
      "Warning: Impossible A1 allele assignment for variant rs361487.\n",
      "Warning: Impossible A1 allele assignment for variant rs361488.\n",
      "Warning: Impossible A1 allele assignment for variant rs361489.\n",
      "Warning: Impossible A1 allele assignment for variant rs2855882.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734060.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734062.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734064.\n",
      "Warning: Impossible A1 allele assignment for variant rs2855896.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734068.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734072.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734078.\n",
      "Warning: Impossible A1 allele assignment for variant rs361440.\n",
      "Warning: Impossible A1 allele assignment for variant rs2244520.\n",
      "Warning: Impossible A1 allele assignment for variant rs2855911.\n",
      "Warning: Impossible A1 allele assignment for variant rs1573618.\n",
      "Warning: Impossible A1 allele assignment for variant rs2855914.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734112.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734113.\n",
      "Warning: Impossible A1 allele assignment for variant rs17209.\n",
      "Warning: Impossible A1 allele assignment for variant rs11768398.\n",
      "Warning: Impossible A1 allele assignment for variant rs2367191.\n",
      "Warning: Impossible A1 allele assignment for variant rs2855919.\n",
      "Warning: Impossible A1 allele assignment for variant rs2855920.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734119.\n",
      "Warning: Impossible A1 allele assignment for variant rs17248.\n",
      "Warning: Impossible A1 allele assignment for variant rs7798099.\n",
      "Warning: Impossible A1 allele assignment for variant rs2855929.\n",
      "Warning: Impossible A1 allele assignment for variant rs2855931.\n",
      "Warning: Impossible A1 allele assignment for variant rs6464492.\n",
      "Warning: Impossible A1 allele assignment for variant rs6950275.\n",
      "Warning: Impossible A1 allele assignment for variant rs361465.\n",
      "Warning: Impossible A1 allele assignment for variant rs361466.\n",
      "Warning: Impossible A1 allele assignment for variant rs975494.\n",
      "Warning: Impossible A1 allele assignment for variant rs6961143.\n",
      "Warning: Impossible A1 allele assignment for variant rs4439035.\n",
      "Warning: Impossible A1 allele assignment for variant rs4352756.\n",
      "Warning: Impossible A1 allele assignment for variant rs17231.\n",
      "Warning: Impossible A1 allele assignment for variant rs6979421.\n",
      "Warning: Impossible A1 allele assignment for variant rs17163237.\n",
      "Warning: Impossible A1 allele assignment for variant rs1008660.\n",
      "Warning: Impossible A1 allele assignment for variant rs17250.\n",
      "Warning: Impossible A1 allele assignment for variant rs2156940.\n",
      "Warning: Impossible A1 allele assignment for variant rs11768792.\n",
      "Warning: Impossible A1 allele assignment for variant rs4725581.\n",
      "Warning: Impossible A1 allele assignment for variant rs4725583.\n",
      "Warning: Impossible A1 allele assignment for variant rs6976636.\n",
      "Warning: Impossible A1 allele assignment for variant rs10216024.\n",
      "Warning: Impossible A1 allele assignment for variant rs4726527.\n",
      "Warning: Impossible A1 allele assignment for variant rs17275.\n",
      "Warning: Impossible A1 allele assignment for variant rs17277.\n",
      "Warning: Impossible A1 allele assignment for variant rs17279.\n",
      "Warning: Impossible A1 allele assignment for variant rs17281.\n",
      "Warning: Impossible A1 allele assignment for variant rs1114694.\n",
      "Warning: Impossible A1 allele assignment for variant rs1114695.\n",
      "Warning: Impossible A1 allele assignment for variant rs10260565.\n",
      "Warning: Impossible A1 allele assignment for variant rs10276363.\n",
      "Warning: Impossible A1 allele assignment for variant rs17232.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734207.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734171.\n",
      "Warning: Impossible A1 allele assignment for variant rs6979469.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734174.\n",
      "Warning: Impossible A1 allele assignment for variant rs2734211.\n",
      "Warning: Impossible A1 allele assignment for variant rs1882723.\n",
      "Warning: Impossible A1 allele assignment for variant rs2011726.\n",
      "Warning: Impossible A1 allele assignment for variant rs1987475.\n",
      "Warning: Impossible A1 allele assignment for variant rs10487531.\n",
      "Warning: Impossible A1 allele assignment for variant rs361445.\n",
      "Warning: Impossible A1 allele assignment for variant rs2156956.\n",
      "Warning: Impossible A1 allele assignment for variant rs2011310.\n",
      "Warning: Impossible A1 allele assignment for variant rs6943682.\n",
      "Warning: Impossible A1 allele assignment for variant rs7793482.\n",
      "Warning: Impossible A1 allele assignment for variant rs361374.\n",
      "Warning: Impossible A1 allele assignment for variant rs926044.\n",
      "Warning: Impossible A1 allele assignment for variant rs11982016.\n",
      "Warning: Impossible A1 allele assignment for variant rs2226967.\n",
      "Warning: Impossible A1 allele assignment for variant rs6971657.\n",
      "Warning: Impossible A1 allele assignment for variant rs6942393.\n",
      "Warning: Impossible A1 allele assignment for variant rs4726529.\n",
      "Warning: Impossible A1 allele assignment for variant rs2015603.\n",
      "Warning: Impossible A1 allele assignment for variant rs1134309.\n",
      "Warning: Impossible A1 allele assignment for variant rs740952.\n",
      "Warning: Impossible A1 allele assignment for variant rs17835147.\n",
      "Warning: Impossible A1 allele assignment for variant rs1114856.\n",
      "Warning: Impossible A1 allele assignment for variant rs10088098.\n",
      "Warning: Impossible A1 allele assignment for variant rs2272000.\n",
      "Warning: Impossible A1 allele assignment for variant rs1047030.\n",
      "Warning: Impossible A1 allele assignment for variant rs12155980.\n",
      "Warning: Impossible A1 allele assignment for variant rs11780139.\n",
      "Warning: Impossible A1 allele assignment for variant rs4737417.\n",
      "Warning: Impossible A1 allele assignment for variant rs1058196.\n",
      "Warning: Impossible A1 allele assignment for variant rs39767.\n",
      "Warning: Impossible A1 allele assignment for variant rs4626644.\n",
      "Warning: Impossible A1 allele assignment for variant rs7017931.\n",
      "Warning: Impossible A1 allele assignment for variant rs7387080.\n",
      "Warning: Impossible A1 allele assignment for variant rs1027623.\n",
      "Warning: Impossible A1 allele assignment for variant rs10116901.\n",
      "Warning: Impossible A1 allele assignment for variant rs1545909.\n",
      "Warning: Impossible A1 allele assignment for variant rs7026127.\n",
      "Warning: Impossible A1 allele assignment for variant rs3800829.\n",
      "Warning: Impossible A1 allele assignment for variant rs7856222.\n",
      "Warning: Impossible A1 allele assignment for variant rs4369069.\n",
      "Warning: Impossible A1 allele assignment for variant rs3758280.\n",
      "Warning: Impossible A1 allele assignment for variant rs3739880.\n",
      "Warning: Impossible A1 allele assignment for variant rs10973198.\n",
      "Warning: Impossible A1 allele assignment for variant rs2351466.\n",
      "Warning: Impossible A1 allele assignment for variant rs3808902.\n",
      "Warning: Impossible A1 allele assignment for variant rs17269854.\n",
      "Warning: Impossible A1 allele assignment for variant rs534721.\n",
      "Warning: Impossible A1 allele assignment for variant rs13290374.\n",
      "Warning: Impossible A1 allele assignment for variant rs11139569.\n",
      "Warning: Impossible A1 allele assignment for variant rs1187334.\n",
      "Warning: Impossible A1 allele assignment for variant rs7467820.\n",
      "Warning: Impossible A1 allele assignment for variant rs9695578.\n",
      "Warning: Impossible A1 allele assignment for variant rs16908089.\n",
      "Warning: Impossible A1 allele assignment for variant rs2245389.\n",
      "Warning: Impossible A1 allele assignment for variant rs1133750.\n",
      "Warning: Impossible A1 allele assignment for variant rs34312136.\n",
      "Warning: Impossible A1 allele assignment for variant rs35079168.\n",
      "Warning: Impossible A1 allele assignment for variant rs35244336.\n",
      "Warning: Impossible A1 allele assignment for variant rs34322762.\n",
      "Warning: Impossible A1 allele assignment for variant rs4078122.\n",
      "Warning: Impossible A1 allele assignment for variant rs35680508.\n",
      "Warning: Impossible A1 allele assignment for variant rs35667158.\n",
      "Warning: Impossible A1 allele assignment for variant rs34984083.\n",
      "Warning: Impossible A1 allele assignment for variant rs8281.\n",
      "Warning: Impossible A1 allele assignment for variant rs3207775.\n",
      "Warning: Impossible A1 allele assignment for variant rs752534.\n",
      "Warning: Impossible A1 allele assignment for variant rs17139206.\n",
      "Warning: Impossible A1 allele assignment for variant rs561415.\n",
      "Warning: Impossible A1 allele assignment for variant rs1802669.\n",
      "Warning: Impossible A1 allele assignment for variant rs4385792.\n",
      "Warning: Impossible A1 allele assignment for variant rs7013.\n",
      "Warning: Impossible A1 allele assignment for variant rs3780898.\n",
      "Warning: Impossible A1 allele assignment for variant rs1059696.\n",
      "Warning: Impossible A1 allele assignment for variant rs3781226.\n",
      "Warning: Impossible A1 allele assignment for variant rs3213025.\n",
      "Warning: Impossible A1 allele assignment for variant rs11813861.\n",
      "Warning: Impossible A1 allele assignment for variant rs11593312.\n",
      "Warning: Impossible A1 allele assignment for variant rs10509262.\n",
      "Warning: Impossible A1 allele assignment for variant rs1650166.\n",
      "Warning: Impossible A1 allele assignment for variant rs4039305.\n",
      "Warning: Impossible A1 allele assignment for variant rs11196005.\n",
      "Warning: Impossible A1 allele assignment for variant rs12243523.\n",
      "Warning: Impossible A1 allele assignment for variant rs11599632.\n",
      "Warning: Impossible A1 allele assignment for variant rs4934332.\n",
      "Warning: Impossible A1 allele assignment for variant rs7897521.\n",
      "Warning: Impossible A1 allele assignment for variant rs7100377.\n",
      "Warning: Impossible A1 allele assignment for variant rs12415539.\n",
      "Warning: Impossible A1 allele assignment for variant rs3740148.\n",
      "Warning: Impossible A1 allele assignment for variant rs242955.\n",
      "Warning: Impossible A1 allele assignment for variant rs2273748.\n",
      "Warning: Impossible A1 allele assignment for variant rs1105126.\n",
      "Warning: Impossible A1 allele assignment for variant rs7947.\n",
      "Warning: Impossible A1 allele assignment for variant rs5030779.\n",
      "Warning: Impossible A1 allele assignment for variant rs7396046.\n",
      "Warning: Impossible A1 allele assignment for variant rs516761.\n",
      "Warning: Impossible A1 allele assignment for variant rs234872.\n",
      "Warning: Impossible A1 allele assignment for variant rs6578399.\n",
      "Warning: Impossible A1 allele assignment for variant rs3850508.\n",
      "Warning: Impossible A1 allele assignment for variant rs7107290.\n",
      "Warning: Impossible A1 allele assignment for variant rs7940346.\n",
      "Warning: Impossible A1 allele assignment for variant rs2297781.\n",
      "Warning: Impossible A1 allele assignment for variant rs4287314.\n",
      "Warning: Impossible A1 allele assignment for variant rs10501259.\n",
      "Warning: Impossible A1 allele assignment for variant rs10160472.\n",
      "Warning: Impossible A1 allele assignment for variant rs4756057.\n",
      "Warning: Impossible A1 allele assignment for variant rs1044796.\n",
      "Warning: Impossible A1 allele assignment for variant rs20542.\n",
      "Warning: Impossible A1 allele assignment for variant rs7109372.\n",
      "Warning: Impossible A1 allele assignment for variant rs28395880.\n",
      "Warning: Impossible A1 allele assignment for variant rs4930395.\n",
      "Warning: Impossible A1 allele assignment for variant rs12799530.\n",
      "Warning: Impossible A1 allele assignment for variant rs12361490.\n",
      "Warning: Impossible A1 allele assignment for variant rs650171.\n",
      "Warning: Impossible A1 allele assignment for variant rs10501467.\n",
      "Warning: Impossible A1 allele assignment for variant rs17507049.\n",
      "Warning: Impossible A1 allele assignment for variant rs632149.\n",
      "Warning: Impossible A1 allele assignment for variant rs661124.\n",
      "Warning: Impossible A1 allele assignment for variant rs683601.\n",
      "Warning: Impossible A1 allele assignment for variant rs1894080.\n",
      "Warning: Impossible A1 allele assignment for variant rs7104532.\n",
      "Warning: Impossible A1 allele assignment for variant rs2324509.\n",
      "Warning: Impossible A1 allele assignment for variant rs610829.\n",
      "Warning: Impossible A1 allele assignment for variant rs4936260.\n",
      "Warning: Impossible A1 allele assignment for variant rs2283312.\n",
      "Warning: Impossible A1 allele assignment for variant rs11524965.\n",
      "Warning: Impossible A1 allele assignment for variant rs2159347.\n",
      "Warning: Impossible A1 allele assignment for variant rs3782598.\n",
      "Warning: Impossible A1 allele assignment for variant rs933552.\n",
      "Warning: Impossible A1 allele assignment for variant rs1551191.\n",
      "Warning: Impossible A1 allele assignment for variant rs1451772.\n",
      "Warning: Impossible A1 allele assignment for variant rs1669413.\n",
      "Warning: Impossible A1 allele assignment for variant rs10505915.\n",
      "Warning: Impossible A1 allele assignment for variant rs3782514.\n",
      "Warning: Impossible A1 allele assignment for variant rs33174.\n",
      "Warning: Impossible A1 allele assignment for variant rs10772153.\n",
      "Warning: Impossible A1 allele assignment for variant rs6581275.\n",
      "Warning: Impossible A1 allele assignment for variant rs3751175.\n",
      "Warning: Impossible A1 allele assignment for variant rs2446991.\n",
      "Warning: Impossible A1 allele assignment for variant rs2277329.\n",
      "Warning: Impossible A1 allele assignment for variant rs7176.\n",
      "Warning: Impossible A1 allele assignment for variant rs2372379.\n",
      "Warning: Impossible A1 allele assignment for variant rs7970175.\n",
      "Warning: Impossible A1 allele assignment for variant rs35400189.\n",
      "Warning: Impossible A1 allele assignment for variant rs34699916.\n",
      "Warning: Impossible A1 allele assignment for variant rs1980061.\n",
      "Warning: Impossible A1 allele assignment for variant rs3794256.\n",
      "Warning: Impossible A1 allele assignment for variant rs7295441.\n",
      "Warning: Impossible A1 allele assignment for variant rs7976768.\n",
      "Warning: Impossible A1 allele assignment for variant rs803569.\n",
      "Warning: Impossible A1 allele assignment for variant rs12302069.\n",
      "Warning: Impossible A1 allele assignment for variant rs7306167.\n",
      "Warning: Impossible A1 allele assignment for variant rs2384608.\n",
      "Warning: Impossible A1 allele assignment for variant rs2279405.\n",
      "Warning: Impossible A1 allele assignment for variant rs3815210.\n",
      "Warning: Impossible A1 allele assignment for variant rs6633.\n",
      "Warning: Impossible A1 allele assignment for variant rs7957839.\n",
      "Warning: Impossible A1 allele assignment for variant rs9552364.\n",
      "Warning: Impossible A1 allele assignment for variant rs1018661.\n",
      "Warning: Impossible A1 allele assignment for variant rs9576606.\n",
      "Warning: Impossible A1 allele assignment for variant rs4643180.\n",
      "Warning: Impossible A1 allele assignment for variant rs3886077.\n",
      "Warning: Impossible A1 allele assignment for variant rs7339131.\n",
      "Warning: Impossible A1 allele assignment for variant rs17058290.\n",
      "Warning: Impossible A1 allele assignment for variant rs1413155.\n",
      "Warning: Impossible A1 allele assignment for variant rs1175967.\n",
      "Warning: Impossible A1 allele assignment for variant rs9583789.\n",
      "Warning: Impossible A1 allele assignment for variant rs1414719.\n",
      "Warning: Impossible A1 allele assignment for variant rs9300901.\n",
      "Warning: Impossible A1 allele assignment for variant rs34857803.\n",
      "Warning: Impossible A1 allele assignment for variant rs12016920.\n",
      "Warning: Impossible A1 allele assignment for variant rs7338868.\n",
      "Warning: Impossible A1 allele assignment for variant rs7399591.\n",
      "Warning: Impossible A1 allele assignment for variant rs9604511.\n",
      "Warning: Impossible A1 allele assignment for variant rs7996853.\n",
      "Warning: Impossible A1 allele assignment for variant rs7399982.\n",
      "Warning: Impossible A1 allele assignment for variant rs9604529.\n",
      "Warning: Impossible A1 allele assignment for variant rs11259844.\n",
      "Warning: Impossible A1 allele assignment for variant rs7489888.\n",
      "Warning: Impossible A1 allele assignment for variant rs9604480.\n",
      "Warning: Impossible A1 allele assignment for variant rs7329468.\n",
      "Warning: Impossible A1 allele assignment for variant rs9550266.\n",
      "Warning: Impossible A1 allele assignment for variant rs11618091.\n",
      "Warning: Impossible A1 allele assignment for variant rs7490083.\n",
      "Warning: Impossible A1 allele assignment for variant rs6602901.\n",
      "Warning: Impossible A1 allele assignment for variant rs9577914.\n",
      "Warning: Impossible A1 allele assignment for variant rs9604566.\n",
      "Warning: Impossible A1 allele assignment for variant rs7323426.\n",
      "Warning: Impossible A1 allele assignment for variant rs6602905.\n",
      "Warning: Impossible A1 allele assignment for variant rs8000868.\n",
      "Warning: Impossible A1 allele assignment for variant rs7997328.\n",
      "Warning: Impossible A1 allele assignment for variant rs9577922.\n",
      "Warning: Impossible A1 allele assignment for variant rs9604488.\n",
      "Warning: Impossible A1 allele assignment for variant rs7400121.\n",
      "Warning: Impossible A1 allele assignment for variant rs7994900.\n",
      "Warning: Impossible A1 allele assignment for variant rs6602910.\n",
      "Warning: Impossible A1 allele assignment for variant rs9604573.\n",
      "Warning: Impossible A1 allele assignment for variant rs7323932.\n",
      "Warning: Impossible A1 allele assignment for variant rs9604466.\n",
      "Warning: Impossible A1 allele assignment for variant rs9577873.\n",
      "Warning: Impossible A1 allele assignment for variant rs7399637.\n",
      "Warning: Impossible A1 allele assignment for variant rs11843762.\n",
      "Warning: Impossible A1 allele assignment for variant rs7996145.\n",
      "Warning: Impossible A1 allele assignment for variant rs9577874.\n",
      "Warning: Impossible A1 allele assignment for variant rs9577876.\n",
      "Warning: Impossible A1 allele assignment for variant rs3088277.\n",
      "Warning: Impossible A1 allele assignment for variant rs7332546.\n",
      "Warning: Impossible A1 allele assignment for variant rs9604512.\n",
      "Warning: Impossible A1 allele assignment for variant rs9577886.\n",
      "Warning: Impossible A1 allele assignment for variant rs9329348.\n",
      "Warning: Impossible A1 allele assignment for variant rs9329350.\n",
      "Warning: Impossible A1 allele assignment for variant rs7400029.\n",
      "Warning: Impossible A1 allele assignment for variant rs9577895.\n",
      "Warning: Impossible A1 allele assignment for variant rs7326506.\n",
      "Warning: Impossible A1 allele assignment for variant rs7491716.\n",
      "Warning: Impossible A1 allele assignment for variant rs7491764.\n",
      "Warning: Impossible A1 allele assignment for variant rs7492267.\n",
      "Warning: Impossible A1 allele assignment for variant rs7490106.\n",
      "Warning: Impossible A1 allele assignment for variant rs9550238.\n",
      "Warning: Impossible A1 allele assignment for variant rs9635144.\n",
      "Warning: Impossible A1 allele assignment for variant rs3936212.\n",
      "Warning: Impossible A1 allele assignment for variant rs7400267.\n",
      "Warning: Impossible A1 allele assignment for variant rs7489746.\n",
      "Warning: Impossible A1 allele assignment for variant rs9652268.\n",
      "Warning: Impossible A1 allele assignment for variant rs6602894.\n",
      "Warning: Impossible A1 allele assignment for variant rs9604526.\n",
      "Warning: Impossible A1 allele assignment for variant rs7489420.\n",
      "Warning: Impossible A1 allele assignment for variant rs7399469.\n",
      "Warning: Impossible A1 allele assignment for variant rs6602895.\n",
      "Warning: Impossible A1 allele assignment for variant rs6422414.\n",
      "Warning: Impossible A1 allele assignment for variant rs9577907.\n",
      "Warning: Impossible A1 allele assignment for variant rs7490802.\n",
      "Warning: Impossible A1 allele assignment for variant rs7335819.\n",
      "Warning: Impossible A1 allele assignment for variant rs7333911.\n",
      "Warning: Impossible A1 allele assignment for variant rs11619575.\n",
      "Warning: Impossible A1 allele assignment for variant rs13379029.\n",
      "Warning: Impossible A1 allele assignment for variant rs7330442.\n",
      "Warning: Impossible A1 allele assignment for variant rs2144827.\n",
      "Warning: Impossible A1 allele assignment for variant rs4147557.\n",
      "Warning: Impossible A1 allele assignment for variant rs10483432.\n",
      "Warning: Impossible A1 allele assignment for variant rs12894724.\n",
      "Warning: Impossible A1 allele assignment for variant rs2184631.\n",
      "Warning: Impossible A1 allele assignment for variant rs17224589.\n",
      "Warning: Impossible A1 allele assignment for variant rs11627089.\n",
      "Warning: Impossible A1 allele assignment for variant rs17597295.\n",
      "Warning: Impossible A1 allele assignment for variant rs12793.\n",
      "Warning: Impossible A1 allele assignment for variant rs17699400.\n",
      "Warning: Impossible A1 allele assignment for variant rs1744296.\n",
      "Warning: Impossible A1 allele assignment for variant rs4983517.\n",
      "Warning: Impossible A1 allele assignment for variant rs2239669.\n",
      "Warning: Impossible A1 allele assignment for variant rs17116135.\n",
      "Warning: Impossible A1 allele assignment for variant rs8034978.\n",
      "Warning: Impossible A1 allele assignment for variant rs8042945.\n",
      "Warning: Impossible A1 allele assignment for variant rs3809581.\n",
      "Warning: Impossible A1 allele assignment for variant rs12900251.\n",
      "Warning: Impossible A1 allele assignment for variant rs10519208.\n",
      "Warning: Impossible A1 allele assignment for variant rs2278166.\n",
      "Warning: Impossible A1 allele assignment for variant rs10083559.\n",
      "Warning: Impossible A1 allele assignment for variant rs3087567.\n",
      "Warning: Impossible A1 allele assignment for variant rs17190650.\n",
      "Warning: Impossible A1 allele assignment for variant rs754643.\n",
      "Warning: Impossible A1 allele assignment for variant rs289095.\n",
      "Warning: Impossible A1 allele assignment for variant rs10519228.\n",
      "Warning: Impossible A1 allele assignment for variant rs11556218.\n",
      "Warning: Impossible A1 allele assignment for variant rs10520774.\n",
      "Warning: Impossible A1 allele assignment for variant rs3803424.\n",
      "Warning: Impossible A1 allele assignment for variant rs1128550.\n",
      "Warning: Impossible A1 allele assignment for variant rs3817833.\n",
      "Warning: Impossible A1 allele assignment for variant rs17135503.\n",
      "Warning: Impossible A1 allele assignment for variant rs1050134.\n",
      "Warning: Impossible A1 allele assignment for variant rs129969.\n",
      "Warning: Impossible A1 allele assignment for variant rs17721144.\n",
      "Warning: Impossible A1 allele assignment for variant rs4357929.\n",
      "Warning: Impossible A1 allele assignment for variant rs4781860.\n",
      "Warning: Impossible A1 allele assignment for variant rs2107234.\n",
      "Warning: Impossible A1 allele assignment for variant rs3785083.\n",
      "Warning: Impossible A1 allele assignment for variant rs17776086.\n",
      "Warning: Impossible A1 allele assignment for variant rs16976568.\n",
      "Warning: Impossible A1 allele assignment for variant rs6498041.\n",
      "Warning: Impossible A1 allele assignment for variant rs252304.\n",
      "Warning: Impossible A1 allele assignment for variant rs11861565.\n",
      "Warning: Impossible A1 allele assignment for variant rs11859733.\n",
      "Warning: Impossible A1 allele assignment for variant rs11645522.\n",
      "Warning: Impossible A1 allele assignment for variant rs8061401.\n",
      "Warning: Impossible A1 allele assignment for variant rs13337562.\n",
      "Warning: Impossible A1 allele assignment for variant rs28437095.\n",
      "Warning: Impossible A1 allele assignment for variant rs9928892.\n",
      "Warning: Impossible A1 allele assignment for variant rs1048149.\n",
      "Warning: Impossible A1 allele assignment for variant rs2273454.\n",
      "Warning: Impossible A1 allele assignment for variant rs28930668.\n",
      "Warning: Impossible A1 allele assignment for variant rs2306270.\n",
      "Warning: Impossible A1 allele assignment for variant rs1642800.\n",
      "Warning: Impossible A1 allele assignment for variant rs764688.\n",
      "Warning: Impossible A1 allele assignment for variant rs12948905.\n",
      "Warning: Impossible A1 allele assignment for variant rs141108.\n",
      "Warning: Impossible A1 allele assignment for variant rs3764420.\n",
      "Warning: Impossible A1 allele assignment for variant rs1557731.\n",
      "Warning: Impossible A1 allele assignment for variant rs17248832.\n",
      "Warning: Impossible A1 allele assignment for variant rs3744357.\n",
      "Warning: Impossible A1 allele assignment for variant rs1136897.\n",
      "Warning: Impossible A1 allele assignment for variant rs7210126.\n",
      "Warning: Impossible A1 allele assignment for variant rs35621842.\n",
      "Warning: Impossible A1 allele assignment for variant rs17673149.\n",
      "Warning: Impossible A1 allele assignment for variant rs181246.\n",
      "Warning: Impossible A1 allele assignment for variant rs11868321.\n",
      "Warning: Impossible A1 allele assignment for variant rs2668622.\n",
      "Warning: Impossible A1 allele assignment for variant rs1042678.\n",
      "Warning: Impossible A1 allele assignment for variant rs12449734.\n",
      "Warning: Impossible A1 allele assignment for variant rs14422.\n",
      "Warning: Impossible A1 allele assignment for variant rs3744155.\n",
      "Warning: Impossible A1 allele assignment for variant rs16942082.\n",
      "Warning: Impossible A1 allele assignment for variant rs9973085.\n",
      "Warning: Impossible A1 allele assignment for variant rs6506389.\n",
      "Warning: Impossible A1 allele assignment for variant rs4121959.\n",
      "Warning: Impossible A1 allele assignment for variant rs9989546.\n",
      "Warning: Impossible A1 allele assignment for variant rs11080748.\n",
      "Warning: Impossible A1 allele assignment for variant rs4527110.\n",
      "Warning: Impossible A1 allele assignment for variant rs11081866.\n",
      "Warning: Impossible A1 allele assignment for variant rs355308.\n",
      "Warning: Impossible A1 allele assignment for variant rs355311.\n",
      "Warning: Impossible A1 allele assignment for variant rs3826608.\n",
      "Warning: Impossible A1 allele assignment for variant rs17565918.\n",
      "Warning: Impossible A1 allele assignment for variant rs10502668.\n",
      "Warning: Impossible A1 allele assignment for variant rs8097124.\n",
      "Warning: Impossible A1 allele assignment for variant rs4919838.\n",
      "Warning: Impossible A1 allele assignment for variant rs4523.\n",
      "Warning: Impossible A1 allele assignment for variant rs11085099.\n",
      "Warning: Impossible A1 allele assignment for variant rs2436514.\n",
      "Warning: Impossible A1 allele assignment for variant rs10417523.\n",
      "Warning: Impossible A1 allele assignment for variant rs2240299.\n",
      "Warning: Impossible A1 allele assignment for variant rs11666779.\n",
      "Warning: Impossible A1 allele assignment for variant rs3093088.\n",
      "Warning: Impossible A1 allele assignment for variant rs14129.\n",
      "Warning: Impossible A1 allele assignment for variant rs8100232.\n",
      "Warning: Impossible A1 allele assignment for variant rs1862273.\n",
      "Warning: Impossible A1 allele assignment for variant rs2306964.\n",
      "Warning: Impossible A1 allele assignment for variant rs8109833.\n",
      "Warning: Impossible A1 allele assignment for variant rs2278996.\n",
      "Warning: Impossible A1 allele assignment for variant rs3745847.\n",
      "Warning: Impossible A1 allele assignment for variant rs565819.\n",
      "Warning: Impossible A1 allele assignment for variant rs12977246.\n",
      "Warning: Impossible A1 allele assignment for variant rs430989.\n",
      "Warning: Impossible A1 allele assignment for variant rs3760667.\n",
      "Warning: Impossible A1 allele assignment for variant rs3187346.\n",
      "Warning: Impossible A1 allele assignment for variant rs4803127.\n",
      "Warning: Impossible A1 allele assignment for variant rs1627096.\n",
      "Warning: Impossible A1 allele assignment for variant rs3810354.\n",
      "Warning: Impossible A1 allele assignment for variant rs2286750.\n",
      "Warning: Impossible A1 allele assignment for variant rs2269817.\n",
      "Warning: Impossible A1 allele assignment for variant rs2076553.\n",
      "Warning: Impossible A1 allele assignment for variant rs3761248.\n",
      "Warning: Impossible A1 allele assignment for variant rs6110212.\n",
      "Warning: Impossible A1 allele assignment for variant rs6076364.\n",
      "Warning: Impossible A1 allele assignment for variant rs3746600.\n",
      "Warning: Impossible A1 allele assignment for variant rs2295034.\n",
      "Warning: Impossible A1 allele assignment for variant rs2070327.\n",
      "Warning: Impossible A1 allele assignment for variant rs6093431.\n",
      "Warning: Impossible A1 allele assignment for variant rs8356.\n",
      "Warning: Impossible A1 allele assignment for variant rs9074.\n",
      "Warning: Impossible A1 allele assignment for variant rs6018699.\n",
      "Warning: Impossible A1 allele assignment for variant rs3761210.\n",
      "Warning: Impossible A1 allele assignment for variant rs16981575.\n",
      "Warning: Impossible A1 allele assignment for variant rs6062145.\n",
      "Warning: Impossible A1 allele assignment for variant rs3810458.\n",
      "Warning: Impossible A1 allele assignment for variant rs17001274.\n",
      "Warning: Impossible A1 allele assignment for variant rs3787659.\n",
      "Warning: Impossible A1 allele assignment for variant rs17114359.\n",
      "Warning: Impossible A1 allele assignment for variant rs3788014.\n",
      "Warning: Impossible A1 allele assignment for variant rs11550376.\n",
      "Warning: Impossible A1 allele assignment for variant rs2329893.\n",
      "Warning: Impossible A1 allele assignment for variant rs35829851.\n",
      "Warning: Impossible A1 allele assignment for variant rs4819880.\n",
      "Warning: Impossible A1 allele assignment for variant rs2268780.\n",
      "Warning: Impossible A1 allele assignment for variant rs7675.\n",
      "Warning: Impossible A1 allele assignment for variant rs2238774.\n",
      "Warning: Impossible A1 allele assignment for variant rs2058556.\n",
      "Warning: Impossible A1 allele assignment for variant rs464805.\n",
      "Warning: Impossible A1 allele assignment for variant rs131393.\n",
      "Warning: Impossible A1 allele assignment for variant rs4820280.\n",
      "Warning: Impossible A1 allele assignment for variant rs9611591.\n",
      "Warning: Impossible A1 allele assignment for variant rs34810367.\n",
      "Warning: Impossible A1 allele assignment for variant rs34420568.\n",
      "Warning: Impossible A1 allele assignment for variant rs35086751.\n",
      "Warning: Impossible A1 allele assignment for variant rs9614750.\n",
      "Warning: Impossible A1 allele assignment for variant rs34315830.\n",
      "Warning: Impossible A1 allele assignment for variant rs34088184.\n",
      "Warning: Impossible A1 allele assignment for variant rs16995472.\n",
      "Warning: Impossible A1 allele assignment for variant rs5767487.\n",
      "Warning: Impossible A1 allele assignment for variant rs34155012.\n",
      "Warning: Impossible A1 allele assignment for variant rs34597827.\n",
      "Warning: Impossible A1 allele assignment for variant rs35662666.\n",
      "Warning: Impossible A1 allele assignment for variant rs35197053.\n",
      "Warning: Impossible A1 allele assignment for variant rs35812349.\n",
      "Warning: Impossible A1 allele assignment for variant rs4074135.\n",
      "--a1-allele: 1000993 assignments attempted, 1000184 made.\n",
      "1000993 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "--make-bed to HapMap-adj.bed + HapMap-adj.bim + HapMap-adj.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# 1) set reference genome \n",
    "awk '{print$2,$5}' 1kG_MDS7.bim > 1kg_ref-list.txt\n",
    "plink --bfile HapMap_MDS --reference-allele 1kg_ref-list.txt --make-bed --out HapMap-adj\n",
    "# The 1kG_MDS7 and the HapMap-adj have the same reference genome for all SNPs.\n",
    "# This command will generate some warnings for impossible A1 allele assignment."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "516db1ff-68e8-46ce-a7a8-a11e9c88c557",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to corrected_hapmap.log.\n",
      "Options in effect:\n",
      "  --a1-allele 1kg_ref-list.txt\n",
      "  --bfile HapMap-adj\n",
      "  --flip flip_list.txt\n",
      "  --make-bed\n",
      "  --out corrected_hapmap\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1000993 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "--flip: 812 SNPs flipped.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998036.\n",
      "Warning: Impossible A1 allele assignment for variant rs34030886.\n",
      "Warning: Impossible A1 allele assignment for variant rs11687477.\n",
      "Warning: Impossible A1 allele assignment for variant rs2581195.\n",
      "Warning: Impossible A1 allele assignment for variant rs1498840.\n",
      "Warning: Impossible A1 allele assignment for variant rs9614750.\n",
      "--a1-allele: 1000993 assignments attempted, 1000988 made.\n",
      "1000993 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "--make-bed to corrected_hapmap.bed + corrected_hapmap.bim +\n",
      "corrected_hapmap.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# 2) Resolve strand issues.\n",
    "# Check for potential strand issues.\n",
    "awk '{print$2,$5,$6}' 1kG_MDS7.bim > 1kGMDS7_tmp\n",
    "awk '{print$2,$5,$6}' HapMap-adj.bim > HapMap-adj_tmp\n",
    "sort 1kGMDS7_tmp HapMap-adj_tmp |uniq -u > all_differences.txt\n",
    "# 1624 differences between the files, some of these might be due to strand issues.\n",
    "\n",
    "## Flip SNPs for resolving strand issues.\n",
    "# Print SNP-identifier and remove duplicates.\n",
    "awk '{print$1}' all_differences.txt | sort -u > flip_list.txt\n",
    "# Generates a file of 812 SNPs. These are the non-corresponding SNPs between the two files. \n",
    "# Flip the 812 non-corresponding SNPs. \n",
    "plink --bfile HapMap-adj --flip flip_list.txt --reference-allele 1kg_ref-list.txt --make-bed --out corrected_hapmap\n",
    "\n",
    "# Check for SNPs which are still problematic after they have been flipped.\n",
    "awk '{print$2,$5,$6}' corrected_hapmap.bim > corrected_hapmap_tmp\n",
    "sort 1kGMDS7_tmp corrected_hapmap_tmp |uniq -u  > uncorresponding_SNPs.txt\n",
    "# This file demonstrates that there are 84 differences between the files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "bddd4854-69fd-47c3-a455-572799986682",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_MDS2.log.\n",
      "Options in effect:\n",
      "  --bfile corrected_hapmap\n",
      "  --exclude SNPs_for_exlusion.txt\n",
      "  --make-bed\n",
      "  --out HapMap_MDS2\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1000993 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "--exclude: 1000951 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998036.\n",
      "1000951 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "--make-bed to HapMap_MDS2.bed + HapMap_MDS2.bim + HapMap_MDS2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to 1kG_MDS8.log.\n",
      "Options in effect:\n",
      "  --bfile 1kG_MDS7\n",
      "  --exclude SNPs_for_exlusion.txt\n",
      "  --make-bed\n",
      "  --out 1kG_MDS8\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1000993 variants loaded from .bim file.\n",
      "629 people (0 males, 0 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to 1kG_MDS8.nosex .\n",
      "--exclude: 1000951 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 629 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999411.\n",
      "1000951 variants and 629 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to 1kG_MDS8.bed + 1kG_MDS8.bim + 1kG_MDS8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to MDS_merge2.log.\n",
      "Options in effect:\n",
      "  --allow-no-sex\n",
      "  --bfile HapMap_MDS2\n",
      "  --bmerge 1kG_MDS8.bed 1kG_MDS8.bim 1kG_MDS8.fam\n",
      "  --make-bed\n",
      "  --out MDS_merge2\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "109 people loaded from HapMap_MDS2.fam.\n",
      "629 people to be merged from 1kG_MDS8.fam.\n",
      "Of these, 629 are new, while 0 are present in the base dataset.\n",
      "Warning: Multiple chromosomes seen for variant 'rs1640558'.\n",
      "1000951 markers loaded from HapMap_MDS2.bim.\n",
      "1000951 markers to be merged from 1kG_MDS8.bim.\n",
      "Of these, 0 are new, while 1000951 are present in the base dataset.\n",
      "Performing single-pass merge (738 people, 1000951 variants).\n",
      "Merged fileset written to MDS_merge2-merge.bed + MDS_merge2-merge.bim +\n",
      "MDS_merge2-merge.fam .\n",
      "1000951 variants loaded from .bim file.\n",
      "738 people (55 males, 54 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to MDS_merge2.nosex .\n",
      "109 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 738 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999208.\n",
      "1000951 variants and 738 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.  (629 phenotypes\n",
      "are missing.)\n",
      "--make-bed to MDS_merge2.bed + MDS_merge2.bim + MDS_merge2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# 3) Remove problematic SNPs from HapMap and 1000 Genomes.【問題のあるSNPの除去】\n",
    "awk '{print$1}' uncorresponding_SNPs.txt | sort -u > SNPs_for_exlusion.txt\n",
    "# The command above generates a list of the 42 SNPs which caused the 84 differences between the HapMap and the 1000 Genomes data sets after flipping and setting of the reference genome.\n",
    "\n",
    "# Remove the 42 problematic SNPs from both datasets.\n",
    "plink --bfile corrected_hapmap --exclude SNPs_for_exlusion.txt --make-bed --out HapMap_MDS2\n",
    "plink --bfile 1kG_MDS7 --exclude SNPs_for_exlusion.txt --make-bed --out 1kG_MDS8\n",
    "\n",
    "# Merge HapMap with 1000 Genomes Data.\n",
    "plink --bfile HapMap_MDS2 --bmerge 1kG_MDS8.bed 1kG_MDS8.bim 1kG_MDS8.fam --allow-no-sex --make-bed --out MDS_merge2\n",
    "\n",
    "# Note, we are fully aware of the sample overlap between the HapMap and 1000 Genomes datasets. However, for the purpose of this tutorial this is not important."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "530a3e99-ad09-4db8-8f92-da401eb63a05",
   "metadata": {},
   "source": [
    "Note, we are fully aware of the sample overlap between the HapMap and 1000 Genomes datasets. However, for the purpose of this tutorial this is not important."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bb6076d6-d576-4584-b24c-b698588bb83d",
   "metadata": {},
   "source": [
    "### Perform MDS on HapMap-CEU data anchored by 1000 Genomes data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "54463695-e808-43ed-af77-4c78d2be791b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to MDS_merge2.log.\n",
      "Options in effect:\n",
      "  --bfile MDS_merge2\n",
      "  --extract indepSNP.prune.in\n",
      "  --genome\n",
      "  --out MDS_merge2\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1000951 variants loaded from .bim file.\n",
      "738 people (55 males, 54 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to MDS_merge2.nosex .\n",
      "109 phenotype values loaded from .fam.\n",
      "--extract: 93331 variants remaining.\n",
      "Using up to 63 threads (change this with --threads).\n",
      "Before main variant filters, 738 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99925.\n",
      "93331 variants and 738 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.  (629 phenotypes\n",
      "are missing.)\n",
      "IBD calculations complete.  \n",
      "Finished writing MDS_merge2.genome .\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to MDS_merge2.log.\n",
      "Options in effect:\n",
      "  --bfile MDS_merge2\n",
      "  --cluster\n",
      "  --mds-plot 10\n",
      "  --out MDS_merge2\n",
      "  --read-genome MDS_merge2.genome\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1000951 variants loaded from .bim file.\n",
      "738 people (55 males, 54 females, 629 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to MDS_merge2.nosex .\n",
      "109 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 738 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999208.\n",
      "1000951 variants and 738 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.  (629 phenotypes\n",
      "are missing.)\n",
      "Clustering... done.                        \n",
      "Cluster solution written to MDS_merge2.cluster1 , MDS_merge2.cluster2 , and3343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899100%\n",
      "MDS_merge2.cluster3 .\n",
      "Performing multidimensional scaling analysis (SVD algorithm, 10\n",
      "dimensions)... done.\n",
      "MDS solution written to MDS_merge2.mds .\n"
     ]
    }
   ],
   "source": [
    "# Using a set of pruned SNPs\n",
    "plink --bfile MDS_merge2 --extract indepSNP.prune.in --genome --out MDS_merge2\n",
    "plink --bfile MDS_merge2 --read-genome MDS_merge2.genome --cluster --mds-plot 10 --out MDS_merge2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01cbf641-0619-45c2-8517-afb2ed462ef8",
   "metadata": {},
   "source": [
    "#### MDS-plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "0d0bf14d-0f6e-4460-8b78-80938ca198e8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2023-11-06 23:28:38--  ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20100804/20100804.ALL.panel\n",
      "           => ‘20100804.ALL.panel.1’\n",
      "Resolving ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)... 193.62.193.167\n",
      "Connecting to ftp.1000genomes.ebi.ac.uk (ftp.1000genomes.ebi.ac.uk)|193.62.193.167|:21... connected.\n",
      "Logging in as anonymous ... Logged in!\n",
      "==> SYST ... done.    ==> PWD ... done.\n",
      "==> TYPE I ... done.  ==> CWD (1) /vol1/ftp/release/20100804 ... done.\n",
      "==> SIZE 20100804.ALL.panel ... 13499\n",
      "==> PASV ... done.    ==> RETR 20100804.ALL.panel ... done.\n",
      "Length: 13499 (13K) (unauthoritative)\n",
      "\n",
      "20100804.ALL.panel. 100%[===================>]  13.18K  --.-KB/s    in 0s      \n",
      "\n",
      "2023-11-06 23:28:40 (136 MB/s) - ‘20100804.ALL.panel.1’ saved [13499]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Download the file with population information of the 1000 genomes dataset.　【集団情報を取得】\n",
    "wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20100804/20100804.ALL.panel\n",
    "# The file 20100804.ALL.panel contains population codes of the individuals of 1000 genomes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "17b61cde-3861-4877-83a1-384207524504",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Convert population codes into superpopulation codes (i.e., AFR,AMR,ASN, and EUR).\n",
    "awk '{print$1,$1,$2}' 20100804.ALL.panel > race_1kG.txt\n",
    "sed 's/JPT/ASN/g' race_1kG.txt>race_1kG2.txt\n",
    "sed 's/ASW/AFR/g' race_1kG2.txt>race_1kG3.txt\n",
    "sed 's/CEU/EUR/g' race_1kG3.txt>race_1kG4.txt\n",
    "sed 's/CHB/ASN/g' race_1kG4.txt>race_1kG5.txt\n",
    "sed 's/CHD/ASN/g' race_1kG5.txt>race_1kG6.txt\n",
    "sed 's/YRI/AFR/g' race_1kG6.txt>race_1kG7.txt\n",
    "sed 's/LWK/AFR/g' race_1kG7.txt>race_1kG8.txt\n",
    "sed 's/TSI/EUR/g' race_1kG8.txt>race_1kG9.txt\n",
    "sed 's/MXL/AMR/g' race_1kG9.txt>race_1kG10.txt\n",
    "sed 's/GBR/EUR/g' race_1kG10.txt>race_1kG11.txt\n",
    "sed 's/FIN/EUR/g' race_1kG11.txt>race_1kG12.txt\n",
    "sed 's/CHS/ASN/g' race_1kG12.txt>race_1kG13.txt\n",
    "sed 's/PUR/AMR/g' race_1kG13.txt>race_1kG14.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "4633ec9c-7a3e-440e-8ce1-8e0174a7ee9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a racefile of your own data.\n",
    "awk '{print$1,$2,\"OWN\"}' HapMap_MDS.fam>racefile_own.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f0e40ddb-35ba-4bfb-a6cc-c40e879764f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Concatenate racefiles.\n",
    "cat race_1kG14.txt racefile_own.txt | sed -e '1i\\FID IID race' > racefile.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "eeb16c3d-b7c2-4c3b-822a-89329d50bb42",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      IID     FID SOL         C1        C2         C3           C4           C5\n",
      "1 HG00098 HG00098   0 -0.0581226 0.0469293 0.00666338 -7.37038e-04 -0.002499530\n",
      "2 HG00100 HG00100   0 -0.0593471 0.0477212 0.00572784  3.78147e-04 -0.004967440\n",
      "3 HG00106 HG00106   0 -0.0587480 0.0468448 0.00662642  2.14437e-05 -0.004974110\n",
      "4 HG00112 HG00112   0 -0.0608584 0.0478612 0.01088190 -3.53118e-03 -0.009468360\n",
      "5 HG00114 HG00114   0 -0.0568818 0.0461408 0.00124838 -1.73026e-03 -0.001305660\n",
      "6 HG00116 HG00116   0 -0.0572653 0.0484633 0.00627335 -9.37135e-04  0.000392984\n",
      "           C6           C7           C8           C9          C10 race\n",
      "1 0.004471640 -0.000404856 -0.001668910 -0.000194086  0.002558240  EUR\n",
      "2 0.006016500 -0.000144140 -0.001573140 -0.003624610  0.000148238  EUR\n",
      "3 0.000383189  0.005917300  0.001485840 -0.000901324  0.000639414  EUR\n",
      "4 0.007731860  0.002299020  0.001092530  0.002964120 -0.000514571  EUR\n",
      "5 0.006965720  0.001562360  0.000930131 -0.002414360 -0.001229670  EUR\n",
      "6 0.003875330  0.003900560 -0.000775670  0.001354420 -0.000778259  EUR\n",
      "data<- read.table(file=\"MDS_merge2.mds\",header=TRUE)\n",
      "race<- read.table(file=\"racefile.txt\",header=TRUE)\n",
      "datafile<- merge(data,race,by=c(\"IID\",\"FID\"))\n",
      "head(datafile)\n",
      "\n",
      "pdf(\"MDS.pdf\",width=7,height=7)\n",
      "for (i in 1:nrow(datafile))\n",
      "{\n",
      "if (datafile[i,14]==\"EUR\") {plot(datafile[i,4],datafile[i,5],type=\"p\",xlim=c(-0.1,0.2),ylim=c(-0.15,0.1),xlab=\"MDS Component 1\",ylab=\"MDS Component 2\",pch=1,cex=0.5,col=\"green\")}\n",
      "par(new=T)\n",
      "if (datafile[i,14]==\"ASN\") {plot(datafile[i,4],datafile[i,5],type=\"p\",xlim=c(-0.1,0.2),ylim=c(-0.15,0.1),xlab=\"MDS Component 1\",ylab=\"MDS Component 2\",pch=1,cex=0.5,col=\"red\")}\n",
      "par(new=T)\n",
      "if (datafile[i,14]==\"AMR\") {plot(datafile[i,4],datafile[i,5],type=\"p\",xlim=c(-0.1,0.2),ylim=c(-0.15,0.1),xlab=\"MDS Component 1\",ylab=\"MDS Component 2\",pch=1,cex=0.5,col=470)}\n",
      "par(new=T)\n",
      "if (datafile[i,14]==\"AFR\") {plot(datafile[i,4],datafile[i,5],type=\"p\",xlim=c(-0.1,0.2),ylim=c(-0.15,0.1),xlab=\"MDS Component 1\",ylab=\"MDS Component 2\",pch=1,cex=0.5,col=\"blue\")}\n",
      "par(new=T)\n",
      "if (datafile[i,14]==\"OWN\") {plot(datafile[i,4],datafile[i,5],type=\"p\",xlim=c(-0.1,0.2),ylim=c(-0.15,0.1),xlab=\"MDS Component 1\",ylab=\"MDS Component 2\",pch=3,cex=0.7,col=\"black\")}\n",
      "par(new=T)\n",
      "}\n",
      "\n",
      "abline(v=-0.035,lty=3)\n",
      "abline(h=0.035,lty=3)\n",
      "legend(\"topright\", pch=c(1,1,1,1,3),c(\"EUR\",\"ASN\",\"AMR\",\"AFR\",\"OWN\"),col=c(\"green\",\"red\",470,\"blue\",\"black\"),bty=\"o\",cex=1)\n"
     ]
    }
   ],
   "source": [
    "# Generate population stratification plot.\n",
    "Rscript ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/2_Population_stratification/MDS_merged.R \n",
    "# The output file MDS.pdf demonstrates that our �own� data falls within the European group of the 1000 genomes data. \n",
    "# Therefore, we do not have to remove subjects.\n",
    "# For educational purposes however, we give scripts below to filter out population stratification outliers. \n",
    "# Please execute the script below in order to generate the appropriate files for the next tutorial.\n",
    "cat ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/2_Population_stratification/MDS_merged.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "77f0e150-b183-4518-adab-379e1e48b75f",
   "metadata": {},
   "outputs": [],
   "source": [
    "filename=\"MDS\"\n",
    "cp $filename.pdf ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/2_Population_stratification/\n",
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/2_Population_stratification/\n",
    "pdftoppm -png $filename.pdf $filename\n",
    "cp $filename-1.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "116746aa-e706-4872-b814-31f56caa7ce4",
   "metadata": {},
   "source": [
    "<img src=\"pic/MDS-1.png\" width=\"50%\">"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2c65c357-10e4-40c9-9310-57238158efd8",
   "metadata": {},
   "source": [
    "### Exclude ethnic outliers."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "ab5f178a-6fb4-401d-ab9c-0ef677f14b97",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_13.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_12\n",
      "  --keep EUR_MDS_merge2\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_13\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "--keep: 109 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998028.\n",
      "1073226 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "--make-bed to HapMap_3_r3_13.bed + HapMap_3_r3_13.bim + HapMap_3_r3_13.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "cd ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS\n",
    "# Select individuals in HapMap data below cut-off thresholds. \n",
    "#The cut-off levels are not fixed thresholds but have to be determined based on the visualization of the first two dimensions. \n",
    "#To exclude ethnic outliers, the thresholds need to be set around the cluster of population of interest.\n",
    "awk '{ if ($4 <-0.04 && $5 >0.03) print $1,$2 }' MDS_merge2.mds > EUR_MDS_merge2\n",
    "\n",
    "# Extract these individuals in HapMap data.\n",
    "plink --bfile HapMap_3_r3_12 --keep EUR_MDS_merge2 --make-bed --out HapMap_3_r3_13\n",
    "# Note, since our HapMap data did include any ethnic outliers, no individuls were removed at this step. \n",
    "#However, if our data would have included individuals outside of the thresholds we set, then these individuals would have been removed."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51d2867c-e963-41b2-b2ee-a6914d4f4532",
   "metadata": {},
   "source": [
    "### Create covariates based on MDS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "d5365702-618e-4ac2-bad9-ac16ff8d23a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_13.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_13\n",
      "  --extract indepSNP.prune.in\n",
      "  --genome\n",
      "  --out HapMap_3_r3_13\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "--extract: 104144 variants remaining.\n",
      "Using up to 63 threads (change this with --threads).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99798.\n",
      "104144 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "IBD calculations complete.  \n",
      "Finished writing HapMap_3_r3_13.genome .\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_13_mds.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_13\n",
      "  --cluster\n",
      "  --mds-plot 10\n",
      "  --out HapMap_3_r3_13_mds\n",
      "  --read-genome HapMap_3_r3_13.genome\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998028.\n",
      "1073226 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "Clustering... done.                        \n",
      "Cluster solution written to HapMap_3_r3_13_mds.cluster1 ,4252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899100%\n",
      "HapMap_3_r3_13_mds.cluster2 , and HapMap_3_r3_13_mds.cluster3 .\n",
      "Performing multidimensional scaling analysis (SVD algorithm, 10\n",
      "dimensions)... done.\n",
      "MDS solution written to HapMap_3_r3_13_mds.mds .\n"
     ]
    }
   ],
   "source": [
    "# Perform an MDS ONLY on HapMap data without ethnic outliers. \n",
    "# 民族的外れ値を含まないHapMapデータに対して、MDSのみを実行する。\n",
    "# The values of the 10 MDS dimensions are subsequently used as covariates in the association analysis in the third tutorial.\n",
    "# 10次元のMDSの値は、3つ目のチュートリアルの関連解析で共変量として使用されます。\n",
    "plink --bfile HapMap_3_r3_13 --extract indepSNP.prune.in --genome --out HapMap_3_r3_13\n",
    "plink --bfile HapMap_3_r3_13 --read-genome HapMap_3_r3_13.genome --cluster --mds-plot 10 --out HapMap_3_r3_13_mds\n",
    "\n",
    "# Change the format of the .mds file into a plink covariate file.【PLINKフォーマットへ変換】\n",
    "awk '{print$1, $2, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13}' HapMap_3_r3_13_mds.mds > covar_mds.txt\n",
    "\n",
    "# The values in covar_mds.txt will be used as covariates, to adjust for remaining population stratification, in the third tutorial where we will perform a genome-wide association analysis.\n",
    "# covar_mds.txtの値は、ゲノムワイド関連解析を行う3番目のチュートリアルで、残りの集団層別化を調整するための共変量として使用される。"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7baaa117-869d-499a-9c3b-2c67eb81c05d",
   "metadata": {},
   "source": [
    "### CONGRATULATIONS \n",
    "\n",
    "you have succesfully controlled your data for population stratification!\n",
    "\n",
    "For the next tutorial you need the following files:\n",
    "- HapMap_3_r3_13 (the bfile, i.e., HapMap_3_r3_13.bed,HapMap_3_r3_13.bim,and HapMap_3_r3_13.fam\n",
    "- covar_mds.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "294549a7-3de8-4786-af3f-d23f060982be",
   "metadata": {},
   "source": [
    "## 3.Association_GWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "52a9f439-02c6-4aed-9f4f-09f507021972",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3_Main_script_association_GWAS.txt  Manhattan_plot.R  QQ_plot.R\n"
     ]
    }
   ],
   "source": [
    "cd ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS\n",
    "ls ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/3_Association_GWAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "d1a50e11-21ad-42ca-8ac0-baeb93c6d184",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "################ Explanation of the main script ##########################\n",
      "\n",
      "# Just as with the previous tutorials, this tutorial can be completed simply by copy-and-pasting all commands from this �main script� into the Unix terminal.\n",
      "# For a theoretical background on these method we refer to the accompanying article entitled �A tutorial on conducting Genome-Wide-Association Studies: Quality control and statistical analysis� (https://www.ncbi.nlm.nih.gov/pubmed/29484742).\n",
      "# In order to run this script you need the following files from the previous tutorial: covar_mds.txt and HapMap_3_r3_13 (bfile, i.e., HapMap_3_r3_13.bed, HapMap_3_r3_13.bim, HapMap_3_r3_13.fam).\n",
      "\n",
      "###########################################################\n",
      "### Association analyses ###\n",
      "\n",
      "# For the association analyses we use the files generated in the previous tutorial (population stratification), named: HapMap_3_r3_13 (with .bed, .bim, and .fam. extensions) and covar_mds.txt\n",
      "\n",
      "# Copy the bfile HapMap_3_r3_13 from the previous tutorial to the current directory.\n",
      "cp HOME/{user}/{path/ 2_Population_stratification}/HapMap_3_r3_13.* HOME/{user}/{path/3_Main_script_association_GWAS.txt}\n",
      "# Copy the covar_mds.txt from the previous tutorial in the current directory.\n",
      "cp HOME/{user}/{path/ 2_Population_stratification}/covar_mds.txt HOME/{user}/{path/3_Main_script_association_GWAS.txt}\n",
      "\n",
      "# For binary traits.\n",
      "\n",
      "# assoc\n",
      "plink --bfile HapMap_3_r3_13 --assoc --out assoc_results\n",
      "# Note, the --assoc option does not allow to correct covariates such as principal components (PC's)/ MDS components, which makes it less suited for association analyses.\n",
      "\n",
      "# logistic \n",
      "# We will be using 10 principal components as covariates in this logistic analysis. We use the MDS components calculated from the previous tutorial: covar_mds.txt.\n",
      "plink --bfile HapMap_3_r3_13 --covar covar_mds.txt --logistic --hide-covar --out logistic_results\n",
      "# Note, we use the option -�hide-covar to only show the additive results of the SNPs in the output file.\n",
      "\n",
      "# Remove NA values, those might give problems generating plots in later steps.\n",
      "awk '!/'NA'/' logistic_results.assoc.logistic > logistic_results.assoc_2.logistic\n",
      "\n",
      "# The results obtained from these GWAS analyses will be visualized in the last step. This will also show if the data set contains any genome-wide significant SNPs.\n",
      "\n",
      "# Note, in case of a quantitative outcome measure the option --logistic should be replaced by --linear. The use of the --assoc option is also possible for quantitative outcome measures (as metioned previously, this option does not allow the use of covariates).\n",
      "\n",
      "#################################################################\n",
      "\n",
      "# Multiple testing\n",
      "# There are various way to deal with multiple testing outside of the conventional genome-wide significance threshold of 5.0E-8, below we present a couple. \n",
      "\n",
      "#adjust\n",
      "plink --bfile HapMap_3_r3_13 -assoc --adjust --out adjusted_assoc_results\n",
      "# This file gives a Bonferroni corrected p-value, along with FDR and others.\n",
      "\n",
      "\n",
      "## Permutation\n",
      "# This is a computational intensive step. Further pros and cons of this method, which can be used for association and dealing with multiple testing, are described in our article corresponding to this tutorial (https://www.ncbi.nlm.nih.gov/pubmed/29484742).\n",
      "# The reduce computational time we only perform this test on a subset of the SNPs from chromosome 22.\n",
      "# The EMP2 collumn provides the for multiple testing corrected p-value.\n",
      "\n",
      "# Generate subset of SNPs\n",
      "awk '{ if ($4 >= 21595000 && $4 <= 21605000) print $2 }' HapMap_3_r3_13.bim > subset_snp_chr_22.txt\n",
      "# Filter your bfile based on the subset of SNPs generated in the step above.\n",
      "plink --bfile HapMap_3_r3_13 --extract subset_snp_chr_22.txt --make-bed --out HapMap_subset_for_perm\n",
      "# Perform 1000000 perrmutations.\n",
      "plink --bfile HapMap_subset_for_perm --assoc --mperm 1000000 --out subset_1M_perm_result\n",
      "\n",
      "# Order your data, from lowest to highest p-value.\n",
      "sort -gk 4 subset_1M_perm_result.assoc.mperm > sorted_subset.txt\n",
      "# Check ordered permutation results\n",
      "head sorted_subset.txt\n",
      "\n",
      "#####################################################################\n",
      "\n",
      "# Generate Manhattan and QQ plots.\n",
      "\n",
      "# These scripts assume R >= 3.0.0.\n",
      "# If you changed the name of the .assoc file or to the assoc.logistic file, please assign those names also to the Rscripts for the Manhattan and QQ plot, otherwise the scripts will not run.\n",
      "\n",
      "# The following Rscripts require the R package qqman, the scripts provided will automatically download this R package and install it in /home/{user}/ . Additionally, the scripts load the qqman library and can therefore, similar to all other Rscript on this GitHub page, be executed from the command line.\n",
      "# This location can be changed to your desired directory\n",
      "\n",
      "Rscript --no-save Manhattan_plot.R\n",
      "Rscript --no-save QQ_plot.R\n",
      "\n",
      "# Please read below when you encountered an error:\n",
      "# Note, the mirror used to download the package qqman can no longer by active, which will result in an error (so below for specific error).\n",
      "# If you encounter this error, please contact me at a.t.marees@amc.uva.nl.\n",
      "# This error can simply be resolved by changing the addresses in the scripts: Manhattan_plot.R and QQ_plot.R.\n",
      "# Simply change the address (http://cran...) in line below, in both Rscripts (Manhattan_plot.R and QQ_plot.R) with (for example) https://cran.univ-paris1.fr/ .\n",
      "# install.packages(\"qqman\",repos=\"http://cran.cnr.berkeley.edu/\",lib=\"~\" ) # location of installation can be changed but has to correspond with the library location\n",
      "\n",
      "# Example error:\n",
      "# \"Error in library(\"qqman\", lib.loc = \"-\") :\n",
      "#\tthere is no package called 'qqman'\n",
      "#Execution halted\"\n",
      "\n",
      "#######################################################\n",
      "\n",
      "## CONGRATULATIONS you have succesfully conducted a GWAS analyses!!\n",
      "\n",
      "# If you are also interested in learning how to conduct a polygenic risk score (PRS) analysis please see our fourth tutorial.\n",
      "# The tutorial explaining PRS is independent from the previous tutorials.\n"
     ]
    }
   ],
   "source": [
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/3_Association_GWAS\n",
    "cat 3_Main_script_association_GWAS.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "779f61b8-4d23-48dc-a437-7851d789ac40",
   "metadata": {},
   "source": [
    "### Explanation of the main script\n",
    "\n",
    "Just as with the previous tutorials, this tutorial can be completed simply by copy-and-pasting all commands from this �main script� into the Unix terminal.\n",
    "\n",
    "For a theoretical background on these method we refer to the accompanying article entitled �A tutorial on conducting Genome-Wide-Association Studies: Quality control and statistical analysis� (https://www.ncbi.nlm.nih.gov/pubmed/29484742).\n",
    "\n",
    "In order to run this script you need the following files from the previous tutorial: covar_mds.txt and HapMap_3_r3_13 (bfile, i.e., HapMap_3_r3_13.bed, HapMap_3_r3_13.bim, HapMap_3_r3_13.fam)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85fd4b4f-ec9c-4631-a4ec-7dfdae0247e7",
   "metadata": {},
   "source": [
    "### Association analyses\n",
    "\n",
    "For the association analyses we use the files generated in the previous tutorial (population stratification), named: HapMap_3_r3_13 (with .bed, .bim, and .fam. extensions) and covar_mds.txt\n",
    "\n",
    "#### For binary traits.\n",
    "\n",
    "* assoc\n",
    "\n",
    "    Note, the --assoc option does not allow to correct covariates such as principal components (PC's)/ MDS components, which makes it less suited for association analyses. *--assocオプションは、主成分（PC）/MDS成分などの共変量を補正することができないので、関連解析にはあまり適していない。*\n",
    "\n",
    "* logistic \n",
    "\n",
    "    We will be using 10 principal components as covariates in this logistic analysis. We use the MDS components calculated from the previous tutorial: covar_mds.txt.*このロジスティック分析では、10個の主成分を共変量として使用します。前のチュートリアルで計算したMDS成分を使用します： covar_mds.txt.*\n",
    "  \n",
    "    Note, we use the option -�hide-covar to only show the additive results of the SNPs in the output file.*hide-covarオプションを使用して、SNPの加法結果のみを出力ファイルに表示します。*\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "23d69969-8263-45cb-811f-2bee13f08947",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to assoc_results.log.\n",
      "Options in effect:\n",
      "  --assoc\n",
      "  --bfile HapMap_3_r3_13\n",
      "  --out assoc_results\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998028.\n",
      "1073226 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "Writing C/C --assoc report to assoc_results.assoc ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to logistic_results.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_13\n",
      "  --covar covar_mds.txt\n",
      "  --hide-covar\n",
      "  --logistic\n",
      "  --out logistic_results\n",
      "\n",
      "Note: --hide-covar flag deprecated.  Use e.g. \"--linear hide-covar\".\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "--covar: 10 covariates loaded.\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998028.\n",
      "1073226 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "Writing logistic model association results to logistic_results.assoc.logistic\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "cd ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS\n",
    "# assoc\n",
    "plink --bfile HapMap_3_r3_13 --assoc --out assoc_results\n",
    "# Note, the --assoc option does not allow to correct covariates such as principal components (PC's)/ MDS components, which makes it less suited for association analyses.\n",
    "\n",
    "# logistic \n",
    "# We will be using 10 principal components as covariates in this logistic analysis. We use the MDS components calculated from the previous tutorial: covar_mds.txt.\n",
    "plink --bfile HapMap_3_r3_13 --covar covar_mds.txt --logistic --hide-covar --out logistic_results\n",
    "# Note, we use the option -�hide-covar to only show the additive results of the SNPs in the output file.\n",
    "\n",
    "# Remove NA values, those might give problems generating plots in later steps.\n",
    "awk '!/'NA'/' logistic_results.assoc.logistic > logistic_results.assoc_2.logistic"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c211774-cb90-453b-ac08-997f812079d6",
   "metadata": {},
   "source": [
    "The results obtained from these GWAS analyses will be visualized in the last step. \n",
    "\n",
    "This will also show if the data set contains any genome-wide significant SNPs.\n",
    "\n",
    "Note, in case of a quantitative outcome measure the option --logistic should be replaced by --linear. \n",
    "\n",
    "The use of the --assoc option is also possible for quantitative outcome measures (as metioned previously, this option does not allow the use of covariates)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2314f8ce-4c5b-4d4f-997d-d064c2a6ce28",
   "metadata": {},
   "source": [
    "#### Multiple testing\n",
    "\n",
    "There are various way to deal with multiple testing outside of the conventional genome-wide significance threshold of 5.0E-8, below we present a couple. \n",
    "\n",
    "*5.0E-8という従来のゲノムワイドの有意性閾値以外の多重検定に対処する方法はいろいろあるが、以下にいくつか紹介する。*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "8e128295-1d80-485f-9efe-d2770b28dab7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to adjusted_assoc_results.log.\n",
      "Options in effect:\n",
      "  --adjust\n",
      "  --assoc\n",
      "  --bfile HapMap_3_r3_13\n",
      "  --out adjusted_assoc_results\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998028.\n",
      "1073226 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "Writing C/C --assoc report to adjusted_assoc_results.assoc ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "--adjust: Genomic inflation est. lambda (based on median chisq) = 1.00965.\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899--adjust values (1073226 variants) written to\n",
      "adjusted_assoc_results.assoc.adjusted .\n"
     ]
    }
   ],
   "source": [
    "#adjust\n",
    "plink --bfile HapMap_3_r3_13 -assoc --adjust --out adjusted_assoc_results\n",
    "# This file gives a Bonferroni corrected p-value, along with FDR and others."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6321dc47-0cca-4a26-afc5-f08c5eb5a457",
   "metadata": {},
   "source": [
    "#### Permutation\n",
    "\n",
    "This is a computational intensive step. Further pros and cons of this method, which can be used for association and dealing with multiple testing, are described in our article corresponding to this tutorial (https://www.ncbi.nlm.nih.gov/pubmed/29484742).\n",
    "\n",
    "The reduce computational time we only perform this test on a subset of the SNPs from chromosome 22.\n",
    "\n",
    "The EMP2 collumn provides the for multiple testing corrected p-value.\n",
    "\n",
    "*これは計算集約的なステップである。この方法の長所と短所については、このチュートリアルに対応する記事（ https://www.ncbi.nlm.nih.gov/pubmed/29484742 ）に記載されています。*\n",
    "\n",
    "*計算時間を短縮するために、22番染色体のSNPのサブセットに対してのみこの検定を行う。*\n",
    "\n",
    "*EMP2列は、多重検定補正p値を提供します。*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "36a3c955-dd91-48cc-855a-462ea5883c45",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_subset_for_perm.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_13\n",
      "  --extract subset_snp_chr_22.txt\n",
      "  --make-bed\n",
      "  --out HapMap_subset_for_perm\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1073226 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "--extract: 84 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99727.\n",
      "84 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "--make-bed to HapMap_subset_for_perm.bed + HapMap_subset_for_perm.bim +\n",
      "HapMap_subset_for_perm.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to subset_1M_perm_result.log.\n",
      "Options in effect:\n",
      "  --assoc\n",
      "  --bfile HapMap_subset_for_perm\n",
      "  --mperm 1000000\n",
      "  --out subset_1M_perm_result\n",
      "\n",
      "Note: --mperm flag deprecated.  Use e.g. \"--model mperm=<value>\".\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "84 variants loaded from .bim file.\n",
      "109 people (55 males, 54 females) loaded from .fam.\n",
      "109 phenotype values loaded from .fam.\n",
      "Using up to 63 threads (change this with --threads).\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99727.\n",
      "84 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 54 are cases and 55 are controls.\n",
      "Writing C/C --assoc report to subset_1M_perm_result.assoc ...\n",
      "[generating permutations                        1417202530343640536163738285868994done.\n",
      "1000000 max(T) permutations complete.\n",
      "Permutation test report written to subset_1M_perm_result.assoc.mperm .\n",
      " CHR          SNP         EMP1         EMP2 \n",
      "  22    rs8135996      0.06832       0.9071 \n",
      "   3    rs1457588      0.03209       0.9142 \n",
      "   5   rs16874721       0.1003       0.9734 \n",
      "  16   rs11640609      0.07229       0.9928 \n",
      "  16    rs7194128      0.07229       0.9928 \n",
      "  22     rs374167        0.107       0.9949 \n",
      "  14    rs2204979       0.1178       0.9987 \n",
      "  11   rs17233395       0.1774       0.9996 \n",
      "  16   rs12445216       0.1592       0.9998 \n"
     ]
    }
   ],
   "source": [
    "# Generate subset of SNPs\n",
    "awk '{ if ($4 >= 21595000 && $4 <= 21605000) print $2 }' HapMap_3_r3_13.bim > subset_snp_chr_22.txt\n",
    "# Filter your bfile based on the subset of SNPs generated in the step above.\n",
    "plink --bfile HapMap_3_r3_13 --extract subset_snp_chr_22.txt --make-bed --out HapMap_subset_for_perm\n",
    "# Perform 1000000 perrmutations.\n",
    "plink --bfile HapMap_subset_for_perm --assoc --mperm 1000000 --out subset_1M_perm_result\n",
    "\n",
    "# Order your data, from lowest to highest p-value.\n",
    "sort -gk 4 subset_1M_perm_result.assoc.mperm > sorted_subset.txt\n",
    "# Check ordered permutation results\n",
    "head sorted_subset.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4304f28b-600b-4a34-b0ed-ab1b9860fc18",
   "metadata": {},
   "source": [
    "### Generate Manhattan and QQ plots.\n",
    "\n",
    "These scripts assume R >= 3.0.0.\n",
    "\n",
    "If you changed the name of the .assoc file or to the assoc.logistic file, please assign those names also to the Rscripts for the Manhattan and QQ plot, otherwise the scripts will not run.\n",
    "\n",
    "The following Rscripts require the R package qqman, the scripts provided will automatically download this R package and install it in /home/{user}/ . \n",
    "\n",
    "Additionally, the scripts load the qqman library and can therefore, similar to all other Rscript on this GitHub page, be executed from the command line.\n",
    "\n",
    "This location can be changed to your desired directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "f2313c9d-d561-4d0a-8881-7535aae97edb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "install.packages(\"qqman\",repos=\"http://cran.cnr.berkeley.edu/\",lib=\"~\" ) # location of installation can be changed but has to correspond with the library location \n",
      "library(\"qqman\",lib.loc=\"~\")  \n",
      "results_log <- read.table(\"logistic_results.assoc_2.logistic\", head=TRUE)\n",
      "jpeg(\"Logistic_manhattan.jpeg\")\n",
      "manhattan(results_log,chr=\"CHR\",bp=\"BP\",p=\"P\",snp=\"SNP\", main = \"Manhattan plot: logistic\")\n",
      "dev.off()\n",
      "\n",
      "results_as <- read.table(\"assoc_results.assoc\", head=TRUE)\n",
      "jpeg(\"assoc_manhattan.jpeg\")\n",
      "manhattan(results_as,chr=\"CHR\",bp=\"BP\",p=\"P\",snp=\"SNP\", main = \"Manhattan plot: assoc\")\n",
      "dev.off()  \n",
      "\n",
      "\n",
      "\n",
      "\n",
      "install.packages(\"qqman\",repos=\"http://cran.cnr.berkeley.edu/\",lib=\"~\" ) # location of installation can be changed but has to correspond with the library location \n",
      "library(\"qqman\",lib.loc=\"~\") \n",
      "results_log <- read.table(\"logistic_results.assoc_2.logistic\", head=TRUE)\n",
      "jpeg(\"QQ-Plot_logistic.jpeg\")\n",
      "qq(results_log$P, main = \"Q-Q plot of GWAS p-values : log\")\n",
      "dev.off()\n",
      "\n",
      "results_as <- read.table(\"assoc_results.assoc\", head=TRUE)\n",
      "jpeg(\"QQ-Plot_assoc.jpeg\")\n",
      "qq(results_as$P, main = \"Q-Q plot of GWAS p-values : log\")\n",
      "dev.off()\n",
      "\n",
      "\n",
      "For example usage please run: vignette('qqman')\n",
      "\n",
      "Citation appreciated but not required:\n",
      "Turner, (2018). qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots. Journal of Open Source Software, 3(25), 731, https://doi.org/10.21105/joss.00731.\n",
      "\n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n",
      "\n",
      "For example usage please run: vignette('qqman')\n",
      "\n",
      "Citation appreciated but not required:\n",
      "Turner, (2018). qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots. Journal of Open Source Software, 3(25), 731, https://doi.org/10.21105/joss.00731.\n",
      "\n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n"
     ]
    }
   ],
   "source": [
    "cat ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/3_Association_GWAS/Manhattan_plot.R\n",
    "cat ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/3_Association_GWAS/QQ_plot.R\n",
    "\n",
    "cat ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/3_Association_GWAS/Manhattan_plot.R | sed -e \"s/install/#install/g\" -e 's/,lib.loc=\"~\")/)/g' > ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/3_Association_GWAS/Manhattan_plot_HM.R\n",
    "cat ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/3_Association_GWAS/QQ_plot.R | sed -e \"s/install/#install/g\" -e 's/,lib.loc=\"~\")/)/g' > ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/3_Association_GWAS/QQ_plot_HM.R\n",
    "\n",
    "#Rscript -e 'install.packages(\"qqman\")'\n",
    "Rscript --no-save ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/3_Association_GWAS/Manhattan_plot_HM.R\n",
    "Rscript --no-save ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial/3_Association_GWAS/QQ_plot_HM.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "7cee1126-450a-4b97-a270-d23cbe14a8bf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;35m/home/hmiwa/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic/assoc_manhattan.jpeg\u001b[0m\u001b[K\n",
      "\u001b[01;35m/home/hmiwa/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic/Logistic_manhattan.jpeg\u001b[0m\u001b[K\n",
      "\u001b[01;35m/home/hmiwa/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic/QQ-Plot_assoc.jpeg\u001b[0m\u001b[K\n",
      "\u001b[01;35m/home/hmiwa/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic/QQ-Plot_logistic.jpeg\u001b[0m\u001b[K\n"
     ]
    }
   ],
   "source": [
    "cp *.jpeg ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic\n",
    "ls ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic/*.jpeg"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "afbeaf8b-e105-41a4-b6bc-33860073fe8a",
   "metadata": {},
   "source": [
    "<img src=\"pic/assoc_manhattan.jpeg\" width=\"49%\"> <img src=\"pic/Logistic_manhattan.jpeg\" width=\"49%\">\n",
    "<img src=\"pic/QQ-Plot_assoc.jpeg\" width=\"49%\"> <img src=\"pic/QQ-Plot_logistic.jpeg\" width=\"49%\">"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d31375d2-5514-4ad6-8238-b2fccc9db6ec",
   "metadata": {},
   "source": [
    "#### Please read below when you encountered an error:\n",
    "\n",
    "Note, the mirror used to download the package qqman can no longer by active, which will result in an error (so below for specific error).\n",
    "\n",
    "If you encounter this error, please contact me at a.t.marees@amc.uva.nl.\n",
    "\n",
    "This error can simply be resolved by changing the addresses in the scripts: Manhattan_plot.R and QQ_plot.R.\n",
    "\n",
    "Simply change the address (http://cran...) in line below, in both Rscripts (Manhattan_plot.R and QQ_plot.R) with (for example) https://cran.univ-paris1.fr/ .\n",
    "\n",
    "    install.packages(\"qqman\",repos=\"http://cran.cnr.berkeley.edu/\",lib=\"~\" ) \n",
    "    # location of installation can be changed but has to correspond with the library location\n",
    "\n",
    "    Example error:\n",
    "        \"Error in library(\"qqman\", lib.loc = \"-\") :\n",
    "        there is no package called 'qqman'\n",
    "        Execution halted\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b4b0aa2-8e5c-45b8-ad69-623105a41691",
   "metadata": {},
   "source": [
    "### CONGRATULATIONS \n",
    "\n",
    "you have succesfully conducted a GWAS analyses!!\n",
    "\n",
    "If you are also interested in learning how to conduct a polygenic risk score (PRS) analysis please see our fourth tutorial.\n",
    "\n",
    "The tutorial explaining PRS is independent from the previous tutorials."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "0de15b92-4084-4d4a-bb96-a7a4b7c077b7",
   "metadata": {},
   "source": [
    "## 4.PRS \n",
    "\n",
    "To perform polygenic risk score analysis we suggest to use PRSice (http://prsice.info). In this tutorial we provide a step-by-step guide to perform a simple polygenic risk score analysis using PRSice and explain how to interpret the results. The current tutorial is based on Linux. If you use Windows of Mac, or require additional information refer to the PRSice user manual.\n",
    "\n",
    "### Installation of PRSice-2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "2ea3124c-6018-4e9b-8bf7-13f528932d75",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2023-11-07 00:28:44--  https://github.com/choishingwan/PRSice/releases/download/2.2.11/PRSice_linux.nightly.zip\n",
      "Resolving github.com (github.com)... 20.27.177.113\n",
      "Connecting to github.com (github.com)|20.27.177.113|:443... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/64860041/03c11900-f005-11e9-9b4f-39f2de45709e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20231106%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20231106T152844Z&X-Amz-Expires=300&X-Amz-Signature=546a94b36d781d94301fd8b87d32bbb7175b8fc2b511e0c450a74f8a4e1f19b2&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=64860041&response-content-disposition=attachment%3B%20filename%3DPRSice_linux.nightly.zip&response-content-type=application%2Foctet-stream [following]\n",
      "--2023-11-07 00:28:44--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/64860041/03c11900-f005-11e9-9b4f-39f2de45709e?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20231106%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20231106T152844Z&X-Amz-Expires=300&X-Amz-Signature=546a94b36d781d94301fd8b87d32bbb7175b8fc2b511e0c450a74f8a4e1f19b2&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=64860041&response-content-disposition=attachment%3B%20filename%3DPRSice_linux.nightly.zip&response-content-type=application%2Foctet-stream\n",
      "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.110.133, 185.199.111.133, 185.199.109.133, ...\n",
      "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.110.133|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 35137723 (34M) [application/octet-stream]\n",
      "Saving to: ‘PRSice_linux.nightly.zip’\n",
      "\n",
      "PRSice_linux.nightl 100%[===================>]  33.51M   110MB/s    in 0.3s    \n",
      "\n",
      "2023-11-07 00:28:45 (110 MB/s) - ‘PRSice_linux.nightly.zip’ saved [35137723/35137723]\n",
      "\n",
      "Archive:  PRSice_linux.nightly.zip\n",
      "  inflating: PRSice.R                \n",
      "  inflating: TOY_BASE_GWAS.assoc     \n",
      "  inflating: TOY_TARGET_DATA.bed     \n",
      "  inflating: TOY_TARGET_DATA.bim     \n",
      "  inflating: TOY_TARGET_DATA.fam     \n",
      "  inflating: PRSice_linux            \n"
     ]
    }
   ],
   "source": [
    "cd ~/local\n",
    "mkdir -p PRSice\n",
    "cd PRSice\n",
    "wget https://github.com/choishingwan/PRSice/releases/download/2.2.11/PRSice_linux.nightly.zip\n",
    "unzip PRSice_linux.nightly.zip"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "990af103-d714-44ff-b40e-b55fabc312fe",
   "metadata": {},
   "source": [
    "### Installing required R packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "d0a13cc3-7e96-456f-8aa3-883216249c3f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;32m/home/hmiwa/local/PRSice/PRSice_linux\u001b[0m\n",
      "\u001b[01;31m/home/hmiwa/local/PRSice/PRSice_linux.nightly.zip\u001b[0m\n",
      "\u001b[01;32m/home/hmiwa/local/PRSice/PRSice.R\u001b[0m\n",
      "/home/hmiwa/local/PRSice/TOY_BASE_GWAS.assoc\n",
      "/home/hmiwa/local/PRSice/TOY_TARGET_DATA.bed\n",
      "/home/hmiwa/local/PRSice/TOY_TARGET_DATA.bim\n",
      "/home/hmiwa/local/PRSice/TOY_TARGET_DATA.fam\n",
      "\n",
      "/home/hmiwa/local/PRSice/lib:\n",
      "\u001b[01;34mdata.table\u001b[0m  \u001b[01;34mgetopt\u001b[0m  \u001b[01;34moptparse\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls ~/local/PRSice/*"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "58bad1fa-f303-486c-a5b8-c1f154f47dde",
   "metadata": {},
   "source": [
    "PRSice can automatically download all required packages, even without administrative right. You can specify the install directory using --dir. For example:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "75da9a4e-3c9d-476b-9c03-aa58056278ca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25hError: Cannot run PRSice without the PRSice binary file\n",
      "Execution halted\n",
      "\u001b[?25h\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "Rscript PRSice.R --dir ."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "c98da3d6-2db3-4e3f-9f3f-2ef7066cee8d",
   "metadata": {},
   "source": [
    "will install all required packages under the local directory.\n",
    "\n",
    "*なぜだか上記だと動かないらしい*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "82e0891f-2146-4ca8-8ea6-23e83ed70165",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: PRSice [options] <-b base_file> <-t target_file>\n",
      "\n",
      "Base File:\n",
      "    --A1                    Column header containing allele 1 (effective allele)\n",
      "                            Default: A1\n",
      "    --A2                    Column header containing allele 2 (non-effective allele)\n",
      "                            Default: A2\n",
      "    --base          | -b    Base association file\n",
      "    --base-info             Base INFO score filtering. Format should be\n",
      "                            <Column name>,<Threshold>. SNPs with info \n",
      "                            score less than <Threshold> will be ignored\n",
      "                            Column name default: INFO\n",
      "                            Threshold default: 0.9\n",
      "    --base-maf             Base MAF filtering. Format should be\n",
      "                            <Column name>,<Threshold>. SNPs with maf\n",
      "                            less than <Threshold> will be ignored. An\n",
      "                            additional column can also be added (e.g.\n",
      "                            also filter MAF for cases), using the\n",
      "                            following format:\n",
      "                            <Column name>,<Threshold>:<Column name>,<Threshold>\n",
      "    --beta                  Whether the test statistic is in the form of \n",
      "                            BETA or OR. If set, test statistic is assume\n",
      "                            to be in the form of BETA. Mutually exclusive\n",
      "                            from --or \n",
      "    --bp                    Column header containing the SNP coordinate\n",
      "                            Default: BP\n",
      "    --chr                   Column header containing the chromosome\n",
      "                            Default: CHR\n",
      "    --index                 If set, assume the INDEX instead of NAME  for\n",
      "                            the corresponding columns are provided. Index\n",
      "                            should be 0-based (start counting from 0)\n",
      "    --no-default            Remove all default options. If set, PRSice\n",
      "                            will not set any default column name and you\n",
      "                            will have to ensure all required columns are\n",
      "                            provided. (--snp, --stat, --A1, --pvalue)\n",
      "    --or                    Whether the test statistic is in the form of \n",
      "                            BETA or OR. If set, test statistic is assume\n",
      "                            to be in the form of OR. Mutually exclusive \n",
      "                            from --beta \n",
      "    --pvalue        | -p    Column header containing the p-value\n",
      "                            Default: P\n",
      "    --snp                   Column header containing the SNP ID\n",
      "                            Default: SNP\n",
      "    --stat                  Column header containing the summary statistic\n",
      "                            If --beta is set, default as BETA. Otherwise,\n",
      "                            will search for OR or BETA from the header\n",
      "                            of the base file\n",
      "\n",
      "Target File:\n",
      "    --binary-target         Indicate whether the target phenotype\n",
      "                            is binary or not. Either T or F should be\n",
      "                            provided where T represent a binary phenotype.\n",
      "                            For multiple phenotypes, the input should be\n",
      "                            separated by comma without space. \n",
      "                            Default: T if --beta and F if --beta is not\n",
      "    --geno                  Filter SNPs based on gentype missingness\n",
      "    --info                  Filter SNPs based on info score. Only used\n",
      "                            for imputed target\n",
      "    --keep                  File containing the sample(s) to be extracted from\n",
      "                            the target file. First column should be FID and\n",
      "                            the second column should be IID. If --ignore-fid is\n",
      "                            set, first column should be IID\n",
      "                            Mutually exclusive from --remove\n",
      "    --maf                   Filter SNPs based on minor allele frequency (MAF)\n",
      "    --nonfounders           Keep the nonfounders in the analysis\n",
      "                            Note: They will still be excluded from LD calculation\n",
      "    --pheno         | -f    Phenotype file containing the phenotype(s).\n",
      "                            First column must be FID of the samples and\n",
      "                            the second column must be IID of the samples.\n",
      "                            When --ignore-fid is set, first column must\n",
      "                            be the IID of the samples.\n",
      "                            Must contain a header if --pheno-col is\n",
      "                            specified\n",
      "    --pheno-col     | -F    Headers of phenotypes to be included from the\n",
      "                            phenotype file\n",
      "    --prevalence    | -k    Prevalence of all binary trait. If provided\n",
      "                            will adjust the ascertainment bias of the R2.\n",
      "                            Note that when multiple binary trait is found,\n",
      "                            prevalence information must be provided for\n",
      "                            all of them (Either adjust all binary traits,\n",
      "                            or don't adjust at all)\n",
      "    --remove                File containing the sample(s) to be removed from\n",
      "                            the target file. First column should be FID and\n",
      "                            the second column should be IID. If --ignore-fid is\n",
      "                            set, first column should be IID\n",
      "                            Mutually exclusive from --keep\n",
      "    --target        | -t    Target genotype file. Currently support\n",
      "                            both BGEN and binary PLINK format. For \n",
      "                            multiple chromosome input, simply substitute\n",
      "                            the chromosome number with #. PRSice will\n",
      "                            automatically replace # with 1-22\n",
      "                            For binary plink format, you can also specify\n",
      "                            a seperate fam file by <prefix>,<fam file>\n",
      "    --target-list           File containing prefix of target genotype\n",
      "                            files. Similar to --target but allow more \n",
      "                            flexibility. Do not support external fam file\n",
      "                            at the moment\n",
      "    --type                  File type of the target file. Support bed \n",
      "                            (binary plink) and bgen format. Default: bed\n",
      "\n",
      "Dosage:\n",
      "    --allow-inter           Allow the generate of intermediate file. This will\n",
      "                            speed up PRSice when using dosage data as clumping\n",
      "                            reference and for hard coding PRS calculation\n",
      "    --dose-thres            Translate any SNPs with highest genotype probability\n",
      "                            less than this threshold to missing call\n",
      "    --hard-thres            A hardcall is saved when the distance to the nearest\n",
      "                            hardcall is less than the hardcall threshold.\n",
      "                            Otherwise a missing code is saved\n",
      "                            Default is: 0.1\n",
      "    --hard                  Use hard coding instead of dosage for PRS construction.\n",
      "                            Default is to use dosage instead of hard coding\n",
      "\n",
      "Clumping:\n",
      "    --clump-kb              The distance for clumping in kb\n",
      "                            Default: 250\n",
      "    --clump-r2              The R2 threshold for clumping\n",
      "                            Default: 0.1\n",
      "    --clump-p               The p-value threshold use for clumping.\n",
      "                            Default: 1\n",
      "    --ld            | -L    LD reference file. Use for LD calculation. If not\n",
      "                            provided, will use the post-filtered target genotype\n",
      "                            for LD calculation. Support multiple chromosome input\n",
      "                            Please see --target for more information\n",
      "    --ld-dose-thres         Translate any SNPs with highest genotype probability\n",
      "                            less than this threshold to missing call\n",
      "    --ld-geno               Filter SNPs based on genotype missingness\n",
      "    --ld-hard-thres         A hardcall is saved when the distance to the nearest\n",
      "                            hardcall is less than the hardcall threshold.\n",
      "                            Otherwise a missing code is saved\n",
      "                            Default is: 0.1\n",
      "    --ld-info               Filter SNPs based on info score. Only used\n",
      "                            for imputed LD reference\n",
      "    --ld-keep               File containing the sample(s) to be extracted from\n",
      "                            the LD reference file. First column should be FID and\n",
      "                            the second column should be IID. If --ignore-fid is\n",
      "                            set, first column should be IID\n",
      "                            Mutually exclusive from --ld-remove\n",
      "                            No effect if --ld was not provided\n",
      "    --ld-list               File containing prefix of LD reference files.\n",
      "                            Similar to --ld but allow more \n",
      "                            flexibility. Do not support external fam file\n",
      "                            at the moment\n",
      "    --ld-maf                Filter SNPs based on minor allele frequency\n",
      "    --ld-remove             File containing the sample(s) to be removed from\n",
      "                            the LD reference file. First column should be FID and\n",
      "                            the second column should be IID. If --ignore-fid is\n",
      "                            set, first column should be IID\n",
      "                            Mutually exclusive from --ld-keep\n",
      "    --ld-type               File type of the LD file. Support bed (binary plink)\n",
      "                            and bgen format. Default: bed\n",
      "    --no-clump              Stop PRSice from performing clumping\n",
      "    --proxy                 Proxy threshold for index SNP to be considered\n",
      "                            as part of the region represented by the clumped\n",
      "                            SNP(s). e.g. --proxy 0.8 means the index SNP will\n",
      "                            represent region of any clumped SNP(s) that has a\n",
      "                            R2>=0.8 even if the index SNP does not physically\n",
      "                            locate within the region\n",
      "\n",
      "Covariate:\n",
      "    --cov           | -C    Covariate file. First column should be FID and \n",
      "                            the second column should be IID. If --ignore-fid\n",
      "                            is set, first column should be IID\n",
      "    --cov-col       | -c    Header of covariates. If not provided, will use\n",
      "                            all variables in the covariate file. By adding\n",
      "                            @ in front of the string, any numbers within [\n",
      "                            and ] will be parsed. E.g. @PC[1-3] will be\n",
      "                            read as PC1,PC2,PC3. Discontinuous input are also\n",
      "                            supported: @cov[1.3-5] will be parsed as \n",
      "                            cov1,cov3,cov4,cov5\n",
      "    --cov-factor            Header of categorical covariate(s). Dummy variable\n",
      "                            will be automatically generated. Any items in\n",
      "                            --cov-factor must also be found in --cov-col\n",
      "                            Also accept continuous input (start with @).\n",
      "\n",
      "P-value Thresholding:\n",
      "    --bar-levels            Level of barchart to be plotted. When --fastscore\n",
      "                            is set, PRSice will only calculate the PRS for \n",
      "                            threshold within the bar level. Levels should be\n",
      "                            comma separated without space\n",
      "    --fastscore             Only calculate threshold stated in --bar-levels\n",
      "    --no-full               By default, PRSice will include the full model, \n",
      "                            i.e. p-value threshold = 1. Setting this flag will\n",
      "                            disable that behaviour\n",
      "    --interval      | -i    The step size of the threshold. Default: 5e-05\n",
      "    --lower         | -l    The starting p-value threshold. Default: 5e-08\n",
      "    --model                 Genetic model use for regression. The genetic\n",
      "                            encoding is based on the base data where the\n",
      "                            encoding represent number of the coding allele\n",
      "                            Available models include:\n",
      "                            add - Additive model, code as 0/1/2 (default)\n",
      "                            dom - Dominant model, code as 0/1/1\n",
      "                            rec - Recessive model, code as 0/0/1\n",
      "                            het - Heterozygous only model, code as 0/1/0\n",
      "    --missing               Method to handle missing genotypes. By default, \n",
      "                            final scores are averages of valid per-allele \n",
      "                            scores with missing genotypes contribute an amount\n",
      "                            proportional to imputed allele frequency. To throw\n",
      "                            out missing observations instead (decreasing the\n",
      "                            denominator in the final average when this happens),\n",
      "                            use the 'SET_ZERO' modifier. Alternatively,\n",
      "                            you can use the 'CENTER' modifier to shift all scores\n",
      "                            to mean zero. \n",
      "    --no-regress            Do not perform the regression analysis and simply\n",
      "                            output all PRS.\n",
      "    --score                 Method to calculate the polygenic score.\n",
      "                            Available methods include:\n",
      "                            avg - Take the average effect size (default)\n",
      "                            std - Standardize the effect size \n",
      "                            sum - Direct summation of the effect size \n",
      "    --upper         | -u    The final p-value threshold. Default: 0.5\n",
      "\n",
      "PRSet:\n",
      "    --background            String to indicate a background file. This string\n",
      "                            should have the format of Name:Type where type can be\n",
      "                            bed   - 0-based range with 3 column. Chr Start End\n",
      "                            range - 1-based range with 3 column. Chr Start End\n",
      "                            gene  - A file contain a column of gene name\n",
      "    --bed           | -B    Bed file containing the selected regions.\n",
      "                            Name of bed file will be used as the region\n",
      "                            identifier. WARNING: Bed file is 0-based\n",
      "    --feature               Feature(s) to be included from the gtf file.\n",
      "                            Default: exon,CDS,gene,protein_coding.\n",
      "    --full-back             Use the whole genome as background for competitive\n",
      "                            p-value calculation\n",
      "    --gtf           | -g    GTF file containing gene boundaries. Required\n",
      "                            when --msigdb is used\n",
      "    --msigdb        | -m    MSIGDB file containing the pathway information.\n",
      "                            Require the gtf file\n",
      "    --snp-set               Provide a SNP set file containing the snp set(s).\n",
      "                            Two different file format is allowed:\n",
      "                            SNP list format - A file containing a single\n",
      "                                              column of SNP ID. Name of the\n",
      "                                              set will be the file name or\n",
      "                                              can be provided using \n",
      "                                              --snp-set File:Name\n",
      "                            MSigDB format   - Each row represent a single SNP \n",
      "                                              set with the first column \n",
      "                                              containing the name of the SNP\n",
      "                                              set.\n",
      "    --wind-3                Add N base(s) to the 3' region of each feature(s) \n",
      "    --wind-5                Add N base(s) to the 5' region of each feature(s) \n",
      "\n",
      "Misc:\n",
      "    --all-score             Output PRS for ALL threshold. WARNING: This\n",
      "                            will generate a huge file\n",
      "    --enable-mmap           Enable memory mapping. This will provide a\n",
      "                            small speed boost if large amount of memory\n",
      "                            is available and if all genotypes were stored\n",
      "                            in the same file\n",
      "    --exclude               File contains SNPs to be excluded from the\n",
      "                            analysis\n",
      "    --extract               File contains SNPs to be included in the \n",
      "                            analysis\n",
      "    --id-delim              This parameter causes sample IDs to be parsed as\n",
      "                            <FID><delimiter><IID>; the default delimiter\n",
      "                            is '_'. \n",
      "    --ignore-fid            Ignore FID for all input. When this is set,\n",
      "                            first column of all file will be assume to\n",
      "                            be IID instead of FID\n",
      "    --keep-ambig            Keep ambiguous SNPs. Only use this option\n",
      "                            if you are certain that the base and target\n",
      "                            has the same A1 and A2 alleles\n",
      "    --logit-perm            When performing permutation, still use logistic\n",
      "                            regression instead of linear regression. This\n",
      "                            will substantially slow down PRSice\n",
      "    --memory                Maximum memory usage allowed. PRSice will try\n",
      "                            its best to honor this setting\n",
      "    --non-cumulate          Calculate non-cumulative PRS. PRS will be reset\n",
      "                            to 0 for each new P-value threshold instead of\n",
      "                            adding up\n",
      "    --out           | -o    Prefix for all file output\n",
      "    --pearson               Use Pearson Correlation for LD calculation\n",
      "                            instead of the maximum likelihood haplotype\n",
      "                            frequency estimates. This will slightly \n",
      "                            decrease the accuracy of LD estimates, but\n",
      "                            should increase the speed of clumping\n",
      "    --perm                  Number of permutation to perform. This swill\n",
      "                            generate the empirical p-value. Recommend to\n",
      "                            use value larger than 10,000\n",
      "    --print-snp             Print all SNPs that remains in the analysis \n",
      "                            after clumping is performed. For PRSet, Y \n",
      "                            indicate the SNPs falls within the gene set \n",
      "                            of interest and N otherwise. If only PRSice \n",
      "                            is performed, a single \"gene set\" called \n",
      "                            \"Base\" will be presented with all entries\n",
      "                            marked as Y\n",
      "    --seed          | -s    Seed used for permutation. If not provided,\n",
      "                            system time will be used as seed. When same\n",
      "                            seed and same input is provided, same result\n",
      "                            can be generated\n",
      "    --thread        | -n    Number of thread use\n",
      "    --use-ref-maf           When specified, missingness imputation will be\n",
      "                            performed based on the reference samples\n",
      "    --x-range               Range of SNPs to be excluded from the whole\n",
      "                            analysis. It can either be a single bed file\n",
      "                            or a comma seperated list of range. Range must\n",
      "                            be in the format of chr:start-end or chr:coordinate\n",
      "    --help          | -h    Display this help message\n",
      "\n"
     ]
    },
    {
     "ename": "",
     "evalue": "255",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "/home/hmiwa/local/PRSice/PRSice_linux"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c0eea34e-a999-4c5a-bf44-998424e18dbb",
   "metadata": {},
   "source": [
    "上記のバイナリは動くようなのでこれで良いのではないか？"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "id": "a246c6f1-bc99-41a1-8117-f281beedc42b",
   "metadata": {},
   "source": [
    "### Running a polygenic risk score with PRSice\n",
    "\n",
    "To run a polygenic risk score analysis on the toy data provided with PRSice run the following code, which is tested in R version 3.2.1 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "0408c9aa-8177-4ab0-a2cd-0bd88a522ca7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25h\u001b[?25hPRSice 2.2.11.b (2019-10-16) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2019 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-11-07 00:37:03\n",
      "./PRSice_linux \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base TOY_BASE_GWAS.assoc \\\n",
      "    --binary-target T \\\n",
      "    --clump-kb 250 \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --or  \\\n",
      "    --out PRSice \\\n",
      "    --seed 4080440511 \\\n",
      "    --stat OR \\\n",
      "    --target TOY_TARGET_DATA \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: TOY_TARGET_DATA (bed) \n",
      "\n",
      "Start processing TOY_BASE_GWAS \n",
      "================================================== \n",
      "\n",
      "Reading 31.74%\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reading 95.27%\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Clumping Progress: 90.13%\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The Jupyter server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--ServerApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "ServerApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "ServerApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?25h\u001b[?25h\n"
     ]
    }
   ],
   "source": [
    "#For binary traits:\n",
    "Rscript PRSice.R --dir . \\\n",
    "--prsice PRSice_linux \\\n",
    "--base TOY_BASE_GWAS.assoc  \\\n",
    "--target TOY_TARGET_DATA \\\n",
    "--thread 1  \\\n",
    "--stat OR \\\n",
    "--binary-target T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "58559a89-8498-45cc-953d-e14ecd9b055e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mlib\u001b[0m                                  PRSice.prsice\n",
      "\u001b[01;35mPRSice_BARPLOT_2023-11-07.png\u001b[0m        \u001b[01;32mPRSice.R\u001b[0m\n",
      "PRSice.best                          PRSice.summary\n",
      "\u001b[01;35mPRSice_HIGH-RES_PLOT_2023-11-07.png\u001b[0m  TOY_BASE_GWAS.assoc\n",
      "\u001b[01;32mPRSice_linux\u001b[0m                         TOY_TARGET_DATA.bed\n",
      "\u001b[01;31mPRSice_linux.nightly.zip\u001b[0m             TOY_TARGET_DATA.bim\n",
      "PRSice.log                           TOY_TARGET_DATA.fam\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "21ae809c-aff7-4a03-81fd-a07a658705e1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SNP CHR BP A1 A2 P OR\n",
      "SNP_22857 4 103593179 1 2 0.2852 13.29\n",
      "SNP_13879 2 237416793 1 2 0.8784 21.624\n",
      "SNP_20771 4 16957461 1 2 0.1994 91.265\n",
      "SNP_13787 2 235355721 1 2 0.7234 3.178\n",
      "SNP_25383 4 189927377 1 2 0.3309 3.167\n",
      "SNP_25290 4 187995996 1 2 0.6327 0.427\n",
      "SNP_21478 4 40161304 1 2 0.06454 5.066\n",
      "SNP_12129 2 176643771 1 2 0.9378 1.276\n",
      "SNP_22809 4 101441465 1 2 0.8111 0.004\n"
     ]
    }
   ],
   "source": [
    "head TOY_BASE_GWAS.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "7c11f479-3d76-4347-9b4d-79c8fc238801",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Phenotype\tSet\tThreshold\tPRS.R2\tFull.R2\tNull.R2\tPrevalence\tCoefficient\tStandard.Error\tP\tNum_SNP\n",
      "-\tBase\t0.4463\t0.0520082\t0.0520082\t0\t-\t86.288\t9.96334\t4.69493e-18\t36759\n"
     ]
    }
   ],
   "source": [
    "cat PRSice.summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "d591752e-9eb0-431b-b78b-a97a994bc397",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PRSice 2.2.11.b (2019-10-16) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2019 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-11-07 00:37:03\n",
      "./PRSice_linux \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base TOY_BASE_GWAS.assoc \\\n",
      "    --binary-target T \\\n",
      "    --clump-kb 250 \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --interval 5e-05 \\\n",
      "    --lower 5e-08 \\\n",
      "    --or  \\\n",
      "    --out PRSice \\\n",
      "    --seed 4080440511 \\\n",
      "    --stat OR \\\n",
      "    --target TOY_TARGET_DATA \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.5\n",
      "\n",
      "Initializing Genotype file: TOY_TARGET_DATA (bed) \n",
      "\n",
      "Start processing TOY_BASE_GWAS \n",
      "================================================== \n",
      "\n",
      "Base file: TOY_BASE_GWAS.assoc \n",
      "91062 variant(s) observed in base file, with: \n",
      "88836 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "2000 people (1024 male(s), 976 female(s)) observed \n",
      "2000 founder(s) included \n",
      "\n",
      "2226 variant(s) not found in previous data \n",
      "88836 variant(s) included \n",
      "\n",
      "There are a total of 1 phenotype to process \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "257564 MB RAM detected; reserving 1 MB for clumping \n",
      "\n",
      "Allocated 1 MB successfully \n",
      "\n",
      "Number of variant(s) after clumping : 88836 \n",
      "\n",
      "Phenotype is a binary phenotype \n",
      "1000 control(s) \n",
      "1000 case(s) \n",
      "\n",
      "There are 1 region(s) with p-value less than 1e-5. Please \n",
      "note that these results are inflated due to the overfitting \n",
      "inherent in finding the best-fit PRS (but it's still best \n",
      "to find the best-fit PRS!). \n",
      "You can use the --perm option (see manual) to calculate an \n",
      "empirical P-value. \n",
      "\n"
     ]
    }
   ],
   "source": [
    "cat PRSice.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "33c86ac7-4845-4a6b-8e7a-3f329f0a7efb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FID IID In_Regression PRS\n",
      "CAS_1 CAS_1 Yes -0.00599501328\n",
      "CAS_2 CAS_2 Yes -0.00631017938\n",
      "CAS_3 CAS_3 Yes -0.00227495325\n",
      "CAS_4 CAS_4 Yes -0.00204360007\n",
      "CAS_5 CAS_5 Yes -0.000830676955\n",
      "CAS_6 CAS_6 Yes -0.00224943517\n",
      "CAS_7 CAS_7 Yes -0.000687589983\n",
      "CAS_8 CAS_8 Yes -0.00413102565\n",
      "CAS_9 CAS_9 Yes 0.00256661049\n"
     ]
    }
   ],
   "source": [
    "head PRSice.best"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "e073a829-4eb0-4cdb-98d4-0d6d7ee37ea7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Set\tThreshold\tR2\tP\tCoefficient\tStandard.Error\tNum_SNP\n",
      "Base\t0.00025005\t0.0133696\t0.000119198\t-0.197266\t0.0512675\t2\n",
      "Base\t0.00030005\t0.00824473\t0.00155473\t-0.225204\t0.0711709\t3\n",
      "Base\t0.00040005\t0.0089725\t0.000519956\t-0.350267\t0.100934\t5\n",
      "Base\t0.00045005\t0.0101339\t0.000174268\t-0.445497\t0.118683\t6\n",
      "Base\t0.00065005\t0.00532975\t0.00605014\t-0.402003\t0.146446\t8\n",
      "Base\t0.00070005\t0.00876654\t0.000367565\t-0.549246\t0.154181\t9\n",
      "Base\t0.00080005\t0.00233607\t0.0618813\t-0.369219\t0.197745\t13\n",
      "Base\t0.00085005\t0.00153157\t0.130151\t-0.342923\t0.226575\t15\n",
      "Base\t0.00095005\t0.000124324\t0.666585\t-0.100725\t0.233789\t16\n"
     ]
    }
   ],
   "source": [
    "head PRSice.prsice"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "2382f496-6703-4e82-8d80-f02cd3824eff",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;35m/home/hmiwa/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic/PRSice_BARPLOT_2023-11-07.png\u001b[0m\u001b[K\n",
      "\u001b[01;35m/home/hmiwa/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic/PRSice_HIGH-RES_PLOT_2023-11-07.png\u001b[0m\u001b[K\n"
     ]
    }
   ],
   "source": [
    "cp *2023-11-07.png ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic\n",
    "ls ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/pic/*2023-11-07.png"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fece99cb-1e67-49a3-b0f0-de468e9b6710",
   "metadata": {},
   "source": [
    "<img src=\"pic/PRSice_BARPLOT_2023-11-07.png\" width=\"49%\"> <img src=\"pic/PRSice_HIGH-RES_PLOT_2023-11-07.png\" width=\"49%\"> "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e41e1fa8-0ede-4c5d-aa21-758405dd42f1",
   "metadata": {},
   "source": [
    "**https://github.com/MareesAT/GWA_tutorial.git にあるチュートリアルは以上！**　**お疲れ様でした！！**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f1aae64-1e92-4733-9eb0-bb42add9930c",
   "metadata": {},
   "source": [
    "## 演習\n",
    "\n",
    "~~1-3の結果を使ってPRSiceを動かしてみよう~~ -> とりあえず別のチュートリアルや実解析をしてみながらの方が良さそう。\n",
    "\n",
    "実際のところ、GWAS(assocなど)は先行研究から引っ張ってくることになるはず。"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f5878fb0-7333-459f-9cbc-31354b085314",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1kG_MDS2.bed\n",
      "1kG_MDS2.bim\n",
      "1kG_MDS2.fam\n",
      "1kG_MDS2.log\n",
      "1kG_MDS2.nosex\n",
      "1kG_MDS3.bed\n",
      "1kG_MDS3.bim\n",
      "1kG_MDS3.fam\n",
      "1kG_MDS3.log\n",
      "1kG_MDS3.nosex\n",
      "1kG_MDS4.bed\n",
      "1kG_MDS4.bim\n",
      "1kG_MDS4.fam\n",
      "1kG_MDS4.log\n",
      "1kG_MDS4.nosex\n",
      "1kG_MDS5.bed\n",
      "1kG_MDS5.bim\n",
      "1kG_MDS5.fam\n",
      "1kG_MDS5.log\n",
      "1kG_MDS5.nosex\n",
      "1kG_MDS6.bed\n",
      "1kG_MDS6.bim\n",
      "1kG_MDS6.fam\n",
      "1kG_MDS6.log\n",
      "1kG_MDS6.nosex\n",
      "1kG_MDS6_SNPs.txt\n",
      "1kG_MDS7.bed\n",
      "1kG_MDS7.bim\n",
      "1kG_MDS7.fam\n",
      "1kG_MDS7.log\n",
      "1kG_MDS7.nosex\n",
      "1kGMDS7_tmp\n",
      "1kG_MDS8.bed\n",
      "1kG_MDS8.bim\n",
      "1kG_MDS8.fam\n",
      "1kG_MDS8.log\n",
      "1kG_MDS8.nosex\n",
      "1kG_MDS.bed\n",
      "1kG_MDS.bim\n",
      "1kG_MDS.fam\n",
      "1kG_MDS.log\n",
      "1kG_MDS.nosex\n",
      "1kg_ref-list.txt\n",
      "20100804.ALL.panel\n",
      "20100804.ALL.panel.1\n",
      "adjusted_assoc_results.assoc\n",
      "adjusted_assoc_results.assoc.adjusted\n",
      "adjusted_assoc_results.log\n",
      "ALL.2of4intersection.20100804.genotypes.bed\n",
      "ALL.2of4intersection.20100804.genotypes.bim\n",
      "ALL.2of4intersection.20100804.genotypes.fam\n",
      "ALL.2of4intersection.20100804.genotypes.log\n",
      "ALL.2of4intersection.20100804.genotypes_no_missing_IDs.bed\n",
      "ALL.2of4intersection.20100804.genotypes_no_missing_IDs.bim\n",
      "ALL.2of4intersection.20100804.genotypes_no_missing_IDs.fam\n",
      "ALL.2of4intersection.20100804.genotypes_no_missing_IDs.log\n",
      "ALL.2of4intersection.20100804.genotypes_no_missing_IDs.nosex\n",
      "ALL.2of4intersection.20100804.genotypes.nosex\n",
      "\u001b[0m\u001b[01;31mALL.2of4intersection.20100804.genotypes.vcf.gz\u001b[0m\n",
      "all_differences.txt\n",
      "\u001b[01;35massoc_manhattan.jpeg\u001b[0m\n",
      "assoc_results.assoc\n",
      "assoc_results.log\n",
      "buildhapmap.txt\n",
      "corrected_hapmap.bed\n",
      "corrected_hapmap.bim\n",
      "corrected_hapmap.fam\n",
      "corrected_hapmap.log\n",
      "corrected_hapmap_tmp\n",
      "covar_mds.txt\n",
      "EUR_MDS_merge2\n",
      "flip_list.txt\n",
      "HapMap_3_r3_12.bed\n",
      "HapMap_3_r3_12.bim\n",
      "HapMap_3_r3_12.fam\n",
      "HapMap_3_r3_13.bed\n",
      "HapMap_3_r3_13.bim\n",
      "HapMap_3_r3_13.fam\n",
      "HapMap_3_r3_13.genome\n",
      "HapMap_3_r3_13.log\n",
      "HapMap_3_r3_13_mds.cluster1\n",
      "HapMap_3_r3_13_mds.cluster2\n",
      "HapMap_3_r3_13_mds.cluster3\n",
      "HapMap_3_r3_13_mds.log\n",
      "HapMap_3_r3_13_mds.mds\n",
      "HapMap-adj.bed\n",
      "HapMap-adj.bim\n",
      "HapMap-adj.fam\n",
      "HapMap-adj.log\n",
      "HapMap-adj_tmp\n",
      "HapMap_MDS2.bed\n",
      "HapMap_MDS2.bim\n",
      "HapMap_MDS2.fam\n",
      "HapMap_MDS2.log\n",
      "HapMap_MDS.bed\n",
      "HapMap_MDS.bim\n",
      "HapMap_MDS.fam\n",
      "HapMap_MDS.log\n",
      "HapMap_MDS.map\n",
      "HapMap_MDS.ped\n",
      "HapMap_SNPs.txt\n",
      "HapMap_subset_for_perm.bed\n",
      "HapMap_subset_for_perm.bim\n",
      "HapMap_subset_for_perm.fam\n",
      "HapMap_subset_for_perm.log\n",
      "indepSNP.prune.in\n",
      "\u001b[01;35mLogistic_manhattan.jpeg\u001b[0m\n",
      "logistic_results.assoc_2.logistic\n",
      "logistic_results.assoc.logistic\n",
      "logistic_results.log\n",
      "MDS_merge2.bed\n",
      "MDS_merge2.bim\n",
      "MDS_merge2.cluster1\n",
      "MDS_merge2.cluster2\n",
      "MDS_merge2.cluster3\n",
      "MDS_merge2.fam\n",
      "MDS_merge2.genome\n",
      "MDS_merge2.log\n",
      "MDS_merge2.mds\n",
      "MDS_merge2.nosex\n",
      "MDS.pdf\n",
      "\u001b[01;35mQQ-Plot_assoc.jpeg\u001b[0m\n",
      "\u001b[01;35mQQ-Plot_logistic.jpeg\u001b[0m\n",
      "race_1kG10.txt\n",
      "race_1kG11.txt\n",
      "race_1kG12.txt\n",
      "race_1kG13.txt\n",
      "race_1kG14.txt\n",
      "race_1kG2.txt\n",
      "race_1kG3.txt\n",
      "race_1kG4.txt\n",
      "race_1kG5.txt\n",
      "race_1kG6.txt\n",
      "race_1kG7.txt\n",
      "race_1kG8.txt\n",
      "race_1kG9.txt\n",
      "race_1kG.txt\n",
      "racefile_own.txt\n",
      "racefile.txt\n",
      "SNPs_for_exlusion.txt\n",
      "sorted_subset.txt\n",
      "subset_1M_perm_result.assoc\n",
      "subset_1M_perm_result.assoc.mperm\n",
      "subset_1M_perm_result.log\n",
      "subset_snp_chr_22.txt\n",
      "uncorresponding_SNPs.txt\n"
     ]
    }
   ],
   "source": [
    "#いまあるファイルの確認\n",
    "cd ~/hmiwa1/storage/hmiwa-lab/02_231105_TutorialPRS\n",
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "939cfbcb-0f6e-4d12-8048-255a85a07d51",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "adjusted_\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K_results.\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K\n",
      "adjusted_\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K_results.\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K.adjusted\n",
      "adjusted_\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K_results.log\n",
      "\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K_manhattan.jpeg\n",
      "\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K_results.\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K\n",
      "\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K_results.log\n",
      "logistic_results.\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K_2.logistic\n",
      "logistic_results.\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K.logistic\n",
      "QQ-Plot_\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K.jpeg\n",
      "subset_1M_perm_result.\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K\n",
      "subset_1M_perm_result.\u001b[01;31m\u001b[Kassoc\u001b[m\u001b[K.mperm\n"
     ]
    }
   ],
   "source": [
    "#assocファイル\n",
    "ls | grep assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d62cf5f3-47ee-4f06-81f7-40e3979db013",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR         SNP      UNADJ         GC       BONF       HOLM   SIDAK_SS   SIDAK_SD     FDR_BH     FDR_BY\n",
      "   3   rs1097157   1.66e-06  1.861e-06          1          1     0.8316     0.8316     0.2871          1 \n",
      "   3   rs1840290   1.66e-06  1.861e-06          1          1     0.8316     0.8316     0.2871          1 \n",
      "   8    rs279466  2.441e-06  2.727e-06          1          1     0.9272     0.9272     0.2871          1 \n",
      "   8    rs279460  2.441e-06  2.727e-06          1          1     0.9272     0.9272     0.2871          1 \n",
      "   8    rs279453  2.441e-06  2.727e-06          1          1     0.9272     0.9272     0.2871          1 \n",
      "   8   rs2623297  2.441e-06  2.727e-06          1          1     0.9272     0.9272     0.2871          1 \n",
      "   3   rs1598120  2.527e-06  2.821e-06          1          1     0.9336     0.9336     0.2871          1 \n",
      "   3   rs2085079  3.162e-06  3.523e-06          1          1     0.9664     0.9664     0.2871          1 \n",
      "   3    rs834858  3.162e-06  3.523e-06          1          1     0.9664     0.9664     0.2871          1 \n"
     ]
    }
   ],
   "source": [
    "head adjusted_assoc_results.assoc.adjusted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b79b9af9-e191-46f5-ae14-7e9f2ab75312",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR         SNP         BP   A1      F_A      F_U   A2        CHISQ            P           OR \n",
      "   1   rs3131972     742584    A   0.1944   0.1455    G       0.9281       0.3354        1.418 \n",
      "   1   rs3131969     744045    A   0.1759      0.1    G        2.647       0.1037        1.921 \n",
      "   1   rs1048488     750775    C   0.1981   0.1455    T        1.054       0.3045        1.451 \n",
      "   1  rs12562034     758311    A  0.06481   0.1182    G        1.863       0.1723       0.5171 \n",
      "   1  rs12124819     766409    G    0.287      0.3    A      0.04416       0.8336       0.9394 \n",
      "   1   rs4040617     769185    G   0.1667      0.1    A          2.1       0.1473          1.8 \n",
      "   1   rs4970383     828418    A   0.2222   0.1727    C       0.8431       0.3585        1.368 \n",
      "   1   rs4475691     836671    T   0.1574   0.1273    C       0.4057       0.5242        1.281 \n",
      "   1   rs1806509     843817    C   0.4259   0.3455    A         1.49       0.2222        1.406 \n"
     ]
    }
   ],
   "source": [
    "head adjusted_assoc_results.assoc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1c5ecd2f-6474-4d83-9144-619dba651cef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR         SNP         BP   A1       TEST    NMISS         OR         STAT            P \n",
      "   1   rs3131972     742584    A        ADD      109      1.957        1.552       0.1207\n",
      "   1   rs3131969     744045    A        ADD      109      2.322          1.8      0.07179\n",
      "   1   rs1048488     750775    C        ADD      108      1.986        1.591       0.1117\n",
      "   1  rs12562034     758311    A        ADD      109     0.3292       -1.912      0.05586\n",
      "   1  rs12124819     766409    G        ADD      109     0.9143      -0.2787       0.7805\n",
      "   1   rs4040617     769185    G        ADD      109      2.059        1.549       0.1214\n",
      "   1   rs4970383     828418    A        ADD      109      1.825         1.41       0.1587\n",
      "   1   rs4475691     836671    T        ADD      109      1.669        1.116       0.2643\n",
      "   1   rs1806509     843817    C        ADD      109      1.817        1.854      0.06371\n",
      " CHR         SNP         BP   A1       TEST    NMISS         OR         STAT            P \n",
      "   1   rs3131972     742584    A        ADD      109      1.957        1.552       0.1207\n",
      "   1   rs3131969     744045    A        ADD      109      2.322          1.8      0.07179\n",
      "   1   rs1048488     750775    C        ADD      108      1.986        1.591       0.1117\n",
      "   1  rs12562034     758311    A        ADD      109     0.3292       -1.912      0.05586\n",
      "   1  rs12124819     766409    G        ADD      109     0.9143      -0.2787       0.7805\n",
      "   1   rs4040617     769185    G        ADD      109      2.059        1.549       0.1214\n",
      "   1   rs4970383     828418    A        ADD      109      1.825         1.41       0.1587\n",
      "   1   rs4475691     836671    T        ADD      109      1.669        1.116       0.2643\n",
      "   1   rs1806509     843817    C        ADD      109      1.817        1.854      0.06371\n"
     ]
    }
   ],
   "source": [
    "head logistic_results.assoc.logistic\n",
    "head logistic_results.assoc_2.logistic #NAが除去されている"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "802e4263-fc1c-4116-95c7-f60514f7bfca",
   "metadata": {},
   "source": [
    "それぞれの項目などについてある程度まとめてあって参考になる\n",
    "\n",
    "https://gist.github.com/kokitsuyuzaki/7907646"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
