{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "defd54a4-e796-4cf2-9cd6-80fb2511a6f7",
   "metadata": {},
   "source": [
    "# 02_231105_TutorialPRS\n",
    "\n",
    "Polygenic Risk Scoreの計算実習をしてみましょう"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "86aab064-4d84-467f-af87-1fb5504a2875",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'GWA_tutorial'...\n",
      "remote: Enumerating objects: 315, done.\u001b[K\n",
      "remote: Counting objects: 100% (19/19), done.\u001b[K\n",
      "remote: Compressing objects: 100% (17/17), done.\u001b[K\n",
      "remote: Total 315 (delta 7), reused 12 (delta 2), pack-reused 296\u001b[K\n",
      "Receiving objects: 100% (315/315), 31.71 MiB | 25.71 MiB/s, done.\n",
      "Resolving deltas: 100% (121/121), done.\n"
     ]
    }
   ],
   "source": [
    "git clone https://github.com/MareesAT/GWA_tutorial.git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2ca4f684-5145-49ee-bccd-9329124f9019",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "02_231105_TutorialPRS.ipynb  \u001b[0m\u001b[01;34mGWA_tutorial\u001b[0m  README.md\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3e72074e-8760-4869-b2f0-dc4955e09415",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;31m1_QC_GWAS.zip\u001b[0m                    \u001b[01;31m3_Association_GWAS.zip\u001b[0m  README.md\n",
      "\u001b[01;31m2_Population_stratification.zip\u001b[0m  4_PRS.doc\n"
     ]
    }
   ],
   "source": [
    "ls GWA_tutorial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2df51214-cf6f-4adb-9d9e-226fc0fb51a5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;31m1_QC_GWAS.zip\u001b[0m                    \u001b[01;31m3_Association_GWAS.zip\u001b[0m  README.md\n",
      "\u001b[01;31m2_Population_stratification.zip\u001b[0m  4_PRS.doc\n"
     ]
    }
   ],
   "source": [
    "cd ~/hmiwa/hmiwa-lab/d_20231023/02_231105_TutorialPRS/GWA_tutorial\n",
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "5ed35dba-2432-4457-a5f7-c0c389da94ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Archive:  1_QC_GWAS.zip\n",
      "  inflating: 1_QC_GWAS/1_Main_script_QC_GWAS.txt  \n",
      "  inflating: 1_QC_GWAS/check_heterozygosity_rate.R  \n",
      "  inflating: 1_QC_GWAS/gender_check.R  \n",
      "  inflating: 1_QC_GWAS/HapMap_3_r3_1.bed  \n",
      "  inflating: 1_QC_GWAS/HapMap_3_r3_1.bim  \n",
      "  inflating: 1_QC_GWAS/HapMap_3_r3_1.fam  \n",
      "  inflating: 1_QC_GWAS/heterozygosity_outliers_list.R  \n",
      "  inflating: 1_QC_GWAS/hist_miss.R   \n",
      "  inflating: 1_QC_GWAS/hwe.R         \n",
      "  inflating: 1_QC_GWAS/inversion.txt  \n",
      "  inflating: 1_QC_GWAS/MAF_check.R   \n",
      "  inflating: 1_QC_GWAS/pops_HapMap_3_r3  \n",
      "  inflating: 1_QC_GWAS/Relatedness.R  \n",
      "Archive:  2_Population_stratification.zip\n",
      "  inflating: 2_Population_stratification/1_Main_script_QC_GWAS.txt  \n",
      "  inflating: 2_Population_stratification/2_Main_script_MDS.txt  \n",
      "  inflating: 2_Population_stratification/MDS_merged.R  \n",
      "Archive:  3_Association_GWAS.zip\n",
      "  inflating: 3_Association_GWAS/3_Main_script_association_GWAS.txt  \n",
      "  inflating: 3_Association_GWAS/Manhattan_plot.R  \n",
      "  inflating: 3_Association_GWAS/QQ_plot.R  \n"
     ]
    }
   ],
   "source": [
    "unzip 1_QC_GWAS.zip\n",
    "unzip 2_Population_stratification.zip\n",
    "unzip 3_Association_GWAS.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "9bcf8a9d-03d7-44df-9c0c-d3d341f57169",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1_Main_script_QC_GWAS.txt    HapMap_3_r3_1.fam               MAF_check.R\n",
      "check_heterozygosity_rate.R  heterozygosity_outliers_list.R  pops_HapMap_3_r3\n",
      "gender_check.R               hist_miss.R                     Relatedness.R\n",
      "HapMap_3_r3_1.bed            hwe.R\n",
      "HapMap_3_r3_1.bim            inversion.txt\n"
     ]
    }
   ],
   "source": [
    "cd 1_QC_GWAS\n",
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4ce62670-f0bb-481f-a517-055baacd4867",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################ Explanation of the main script ##########################\n",
      "\n",
      "# This tutorial uses freely available HapMap data: hapmap3_r3_b36_fwd.consensus.qc. We simulated a binary outcome measure (i.e., a binary phenotypic trait) and added this to the dataset. The outcome measure was only simulated for the founders in the HapMap data. This data set will be referred to as HapMap_3_r3_1. \n",
      "# The HapMap data, without our simulated outcome measure, can also be obtained from http://hapmap.ncbi.nlm.nih.gov/downloads/genotypes/2010-05_phaseIII/plink_format/ \n",
      "# It is essential for the execution of the tutorial that that all scripts belonging to this tutorial are in the same directory on your UNIX workstation.\n",
      "# Many scripts include comments which explain how these scripts work. Note, in order to complete the tutorial it is essential to execute all commands in this tutorial.\n",
      "# This script can also be used for your own data analysis, to use it as such, replace the name of the HapMap file with the name of your own data file. \n",
      "# Furthermore, this script is based on a binary outcome measure, and is therefore not applicable for quantitative outcome measures (this would require some adaptations)\n",
      "# Note, most GWAS studies are performed on an ethnic homogenous population, in which population outliers are removed. The HapMap data, used for this tutorial, contains multiple distinct ethnic groups, which makes it problematic for analysis.\n",
      "# Therefore, we have selected only the EUR individuals of the complete HapMap sample for the tutorials 1-3. This selection is already performed in the HapMap_3_r3_1 file from our GitHub page.\n",
      "# The Rscripts used in this tutorial are all executed from the Unix command line.\n",
      "# Therefore, this tutorial and the other tutorials from our GitHub page, can be completed simply by copy-and-pasting all commands from the �main scripts� into the Unix terminal.\n",
      "# For a thorough theoretical explanation of all QC steps we refer to the article accompanying this tutorial entitled �A tutorial on conducting Genome-Wide-Association Studies: Quality control and statistical analysis� (https://www.ncbi.nlm.nih.gov/pubmed/29484742).\n",
      "\n",
      "\n",
      "##############################################################\n",
      "############### START ANALISIS ###############################\n",
      "##############################################################\n",
      "\n",
      "# Change directory to a folder on your UNIX device containing all files from �1_QC_GWAS.zip�. \n",
      "cd HOME/{user}/{path/folder containing your files}\n",
      "\n",
      "### Step 1 ### \n",
      "\n",
      "# Investigate missingness per individual and per SNP and make histograms.\n",
      "plink --bfile HapMap_3_r3_1 --missing    \n",
      "# output: plink.imiss and plink.lmiss, these files show respectively the proportion of missing SNPs per individual and the proportion of missing individuals per SNP.\n",
      "\n",
      "\n",
      "# Generate plots to visualize the missingness results.\n",
      "Rscript --no-save hist_miss.R\n",
      "\n",
      "# Delete SNPs and individuals with high levels of missingness, explanation of this and all following steps can be found in box 1 and table 1 of the article mentioned in the comments of this script.\n",
      "# The following two QC commands will not remove any SNPs or individuals. However, it is good practice to start the QC with these non-stringent thresholds.  \n",
      "# Delete SNPs with missingness >0.2.\n",
      "plink --bfile HapMap_3_r3_1 --geno 0.2 --make-bed --out HapMap_3_r3_2\n",
      "\n",
      "# Delete individuals with missingness >0.2.\n",
      "plink --bfile HapMap_3_r3_2 --mind 0.2 --make-bed --out HapMap_3_r3_3\n",
      "\n",
      "# Delete SNPs with missingness >0.02.\n",
      "plink --bfile HapMap_3_r3_3 --geno 0.02 --make-bed --out HapMap_3_r3_4\n",
      "\n",
      "# Delete individuals with missingness >0.02.\n",
      "plink --bfile HapMap_3_r3_4 --mind 0.02 --make-bed --out HapMap_3_r3_5\n",
      "\n",
      "###################################################################\n",
      "### Step2 ####\n",
      "\n",
      "# Check for sex discrepancy.\n",
      "# Subjects who were a priori determined as females must have a F value of <0.2, and subjects who were a priori determined as males must have a F value >0.8. This F value is based on the X chromosome inbreeding (homozygosity) estimate.\n",
      "# Subjects who do not fulfil these requirements are flagged \"PROBLEM\" by PLINK.\n",
      "\n",
      "plink --bfile HapMap_3_r3_5 --check-sex \n",
      "\n",
      "# Generate plots to visualize the sex-check results.\n",
      "Rscript --no-save gender_check.R\n",
      "# These checks indicate that there is one woman with a sex discrepancy, F value of 0.99. (When using other datasets often a few discrepancies will be found). \n",
      "\n",
      "# The following two scripts can be used to deal with individuals with a sex discrepancy.\n",
      "# Note, please use one of the two options below to generate the bfile hapmap_r23a_6, this file we will use in the next step of this tutorial.\n",
      "\n",
      "# 1) Delete individuals with sex discrepancy.\n",
      "grep \"PROBLEM\" plink.sexcheck| awk '{print$1,$2}'> sex_discrepancy.txt\n",
      "# This command generates a list of individuals with the status �PROBLEM�.\n",
      "plink --bfile HapMap_3_r3_5 --remove sex_discrepancy.txt --make-bed --out HapMap_3_r3_6 \n",
      "# This command removes the list of individuals with the status �PROBLEM�.\n",
      "\n",
      "# 2) impute-sex.\n",
      "#plink --bfile HapMap_3_r3_5 --impute-sex --make-bed --out HapMap_3_r3_6\n",
      "# This imputes the sex based on the genotype information into your data set.\n",
      "\n",
      "###################################################\n",
      "### Step 3 ### \n",
      "\n",
      "# Generate a bfile with autosomal SNPs only and delete SNPs with a low minor allele frequency (MAF).\n",
      "\n",
      "# Select autosomal SNPs only (i.e., from chromosomes 1 to 22).\n",
      "awk '{ if ($1 >= 1 && $1 <= 22) print $2 }' HapMap_3_r3_6.bim > snp_1_22.txt\n",
      "plink --bfile HapMap_3_r3_6 --extract snp_1_22.txt --make-bed --out HapMap_3_r3_7\n",
      "\n",
      "# Generate a plot of the MAF distribution.\n",
      "plink --bfile HapMap_3_r3_7 --freq --out MAF_check\n",
      "Rscript --no-save MAF_check.R\n",
      "\n",
      "# Remove SNPs with a low MAF frequency.\n",
      "plink --bfile HapMap_3_r3_7 --maf 0.05 --make-bed --out HapMap_3_r3_8\n",
      "# 1073226 SNPs are left\n",
      "# A conventional MAF threshold for a regular GWAS is between 0.01 or 0.05, depending on sample size.\n",
      "\n",
      "\n",
      "####################################################\n",
      "### Step 4 ###\n",
      "\n",
      "# Delete SNPs which are not in Hardy-Weinberg equilibrium (HWE).\n",
      "# Check the distribution of HWE p-values of all SNPs.\n",
      "\n",
      "plink --bfile HapMap_3_r3_8 --hardy\n",
      "# Selecting SNPs with HWE p-value below 0.00001, required for one of the two plot generated by the next Rscript, allows to zoom in on strongly deviating SNPs. \n",
      "awk '{ if ($9 <0.00001) print $0 }' plink.hwe>plinkzoomhwe.hwe\n",
      "Rscript --no-save hwe.R\n",
      "\n",
      "# By default the --hwe option in plink only filters for controls.\n",
      "# Therefore, we use two steps, first we use a stringent HWE threshold for controls, followed by a less stringent threshold for the case data.\n",
      "plink --bfile HapMap_3_r3_8 --hwe 1e-6 --make-bed --out HapMap_hwe_filter_step1\n",
      "\n",
      "# The HWE threshold for the cases filters out only SNPs which deviate extremely from HWE. \n",
      "# This second HWE step only focusses on cases because in the controls all SNPs with a HWE p-value < hwe 1e-6 were already removed\n",
      "plink --bfile HapMap_hwe_filter_step1 --hwe 1e-10 --hwe-all --make-bed --out HapMap_3_r3_9\n",
      "\n",
      "# Theoretical background for this step is given in our accompanying article: https://www.ncbi.nlm.nih.gov/pubmed/29484742 .\n",
      "\n",
      "############################################################\n",
      "### step 5 ###\n",
      "\n",
      "# Generate a plot of the distribution of the heterozygosity rate of your subjects.\n",
      "# And remove individuals with a heterozygosity rate deviating more than 3 sd from the mean.\n",
      "\n",
      "# Checks for heterozygosity are performed on a set of SNPs which are not highly correlated.\n",
      "# Therefore, to generate a list of non-(highly)correlated SNPs, we exclude high inversion regions (inversion.txt [High LD regions]) and prune the SNPs using the command --indep-pairwise�.\n",
      "# The parameters �50 5 0.2� stand respectively for: the window size, the number of SNPs to shift the window at each step, and the multiple correlation coefficient for a SNP being regressed on all other SNPs simultaneously.\n",
      "\n",
      "plink --bfile HapMap_3_r3_9 --exclude inversion.txt --range --indep-pairwise 50 5 0.2 --out indepSNP\n",
      "# Note, don't delete the file indepSNP.prune.in, we will use this file in later steps of the tutorial.\n",
      "\n",
      "plink --bfile HapMap_3_r3_9 --extract indepSNP.prune.in --het --out R_check\n",
      "# This file contains your pruned data set.\n",
      "\n",
      "# Plot of the heterozygosity rate distribution\n",
      "Rscript --no-save check_heterozygosity_rate.R\n",
      "\n",
      "# The following code generates a list of individuals who deviate more than 3 standard deviations from the heterozygosity rate mean.\n",
      "# For data manipulation we recommend using UNIX. However, when performing statistical calculations R might be more convenient, hence the use of the Rscript for this step:\n",
      "Rscript --no-save heterozygosity_outliers_list.R\n",
      "\n",
      "# Output of the command above: fail-het-qc.txt .\n",
      "# When using our example data/the HapMap data this list contains 2 individuals (i.e., two individuals have a heterozygosity rate deviating more than 3 SD's from the mean).\n",
      "# Adapt this file to make it compatible for PLINK, by removing all quotation marks from the file and selecting only the first two columns.\n",
      "sed 's/\"// g' fail-het-qc.txt | awk '{print$1, $2}'> het_fail_ind.txt\n",
      "\n",
      "# Remove heterozygosity rate outliers.\n",
      "plink --bfile HapMap_3_r3_9 --remove het_fail_ind.txt --make-bed --out HapMap_3_r3_10\n",
      "\n",
      "\n",
      "############################################################\n",
      "### step 6 ###\n",
      "\n",
      "# It is essential to check datasets you analyse for cryptic relatedness.\n",
      "# Assuming a random population sample we are going to exclude all individuals above the pihat threshold of 0.2 in this tutorial.\n",
      "\n",
      "# Check for relationships between individuals with a pihat > 0.2.\n",
      "plink --bfile HapMap_3_r3_10 --extract indepSNP.prune.in --genome --min 0.2 --out pihat_min0.2\n",
      "\n",
      "# The HapMap dataset is known to contain parent-offspring relations. \n",
      "# The following commands will visualize specifically these parent-offspring relations, using the z values. \n",
      "awk '{ if ($8 >0.9) print $0 }' pihat_min0.2.genome>zoom_pihat.genome\n",
      "\n",
      "# Generate a plot to assess the type of relationship.\n",
      "Rscript --no-save Relatedness.R\n",
      "\n",
      "# The generated plots show a considerable amount of related individuals (explentation plot; PO = parent-offspring, UN = unrelated individuals) in the Hapmap data, this is expected since the dataset was constructed as such.\n",
      "# Normally, family based data should be analyzed using specific family based methods. In this tutorial, for demonstrative purposes, we treat the relatedness as cryptic relatedness in a random population sample.\n",
      "# In this tutorial, we aim to remove all 'relatedness' from our dataset.\n",
      "# To demonstrate that the majority of the relatedness was due to parent-offspring we only include founders (individuals without parents in the dataset).\n",
      "\n",
      "plink --bfile HapMap_3_r3_10 --filter-founders --make-bed --out HapMap_3_r3_11\n",
      "\n",
      "# Now we will look again for individuals with a pihat >0.2.\n",
      "plink --bfile HapMap_3_r3_11 --extract indepSNP.prune.in --genome --min 0.2 --out pihat_min0.2_in_founders\n",
      "# The file 'pihat_min0.2_in_founders.genome' shows that, after exclusion of all non-founders, only 1 individual pair with a pihat greater than 0.2 remains in the HapMap data.\n",
      "# This is likely to be a full sib or DZ twin pair based on the Z values. Noteworthy, they were not given the same family identity (FID) in the HapMap data.\n",
      "\n",
      "# For each pair of 'related' individuals with a pihat > 0.2, we recommend to remove the individual with the lowest call rate. \n",
      "plink --bfile HapMap_3_r3_11 --missing\n",
      "# Use an UNIX text editor (e.g., vi(m) ) to check which individual has the highest call rate in the 'related pair'. \n",
      "\n",
      "# Generate a list of FID and IID of the individual(s) with a Pihat above 0.2, to check who had the lower call rate of the pair.\n",
      "# In our dataset the individual 13291  NA07045 had the lower call rate.\n",
      "vi 0.2_low_call_rate_pihat.txt\n",
      "i \n",
      "13291  NA07045\n",
      "# Press esc on keyboard!\n",
      ":x\n",
      "# Press enter on keyboard\n",
      "# In case of multiple 'related' pairs, the list generated above can be extended using the same method as for our lone 'related' pair.\n",
      "\n",
      "# Delete the individuals with the lowest call rate in 'related' pairs with a pihat > 0.2 \n",
      "plink --bfile HapMap_3_r3_11 --remove 0.2_low_call_rate_pihat.txt --make-bed --out HapMap_3_r3_12\n",
      "\n",
      "################################################################################################################################\n",
      "\n",
      "# CONGRATULATIONS!! You've just succesfully completed the first tutorial! You are now able to conduct a proper genetic QC. \n",
      "\n",
      "# For the next tutorial, using the script: 2_Main_script_MDS.txt, you need the following files:\n",
      "# - The bfile HapMap_3_r3_12 (i.e., HapMap_3_r3_12.fam,HapMap_3_r3_12.bed, and HapMap_3_r3_12.bim\n",
      "# - indepSNP.prune.in\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cat 1_Main_script_QC_GWAS.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "f7f50342-a55a-41c5-afa8-fce3debd0151",
   "metadata": {},
   "outputs": [],
   "source": [
    "plink=~/local/plink/plink"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c5759d93-8995-4311-b8de-766def4949f4",
   "metadata": {},
   "source": [
    "### Step 1 ### \n",
    "\n",
    "Investigate missingness per individual and per SNP and make histograms.\n",
    "\n",
    "    plink --bfile HapMap_3_r3_1 --missing    \n",
    "\n",
    "output: plink.imiss and plink.lmiss, these files show respectively the proportion of missing SNPs per individual and the proportion of missing individuals per SNP.\n",
    "\n",
    "\n",
    "Generate plots to visualize the missingness results.\n",
    "\n",
    "    Rscript --no-save hist_miss.R\n",
    "\n",
    "Delete SNPs and individuals with high levels of missingness, explanation of this and all following steps can be found in box 1 and table 1 of the article mentioned in the comments of this script.\n",
    "\n",
    "The following two QC commands will not remove any SNPs or individuals. However, it is good practice to start the QC with these non-stringent thresholds.  \n",
    "\n",
    "Delete SNPs with missingness >0.2.\n",
    "\n",
    "    plink --bfile HapMap_3_r3_1 --geno 0.2 --make-bed --out HapMap_3_r3_2\n",
    "\n",
    "Delete individuals with missingness >0.2.\n",
    "\n",
    "    plink --bfile HapMap_3_r3_2 --mind 0.2 --make-bed --out HapMap_3_r3_3\n",
    "\n",
    "Delete SNPs with missingness >0.02.\n",
    "\n",
    "    plink --bfile HapMap_3_r3_3 --geno 0.02 --make-bed --out HapMap_3_r3_4\n",
    "\n",
    "Delete individuals with missingness >0.02.\n",
    "\n",
    "    plink --bfile HapMap_3_r3_4 --mind 0.02 --make-bed --out HapMap_3_r3_5\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "cff7ff1e-d141-47c6-9ee4-b56f4c0bd7ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plink.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_1\n",
      "  --missing\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1457897 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 225 het. haploid genotypes present (see plink.hh ); many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.997378.\n",
      "--missing: Sample missing data report written to plink.imiss, and variant-based\n",
      "missing data report written to plink.lmiss.\n"
     ]
    }
   ],
   "source": [
    "# Investigate missingness per individual and per SNP and make histograms.\n",
    "$plink --bfile HapMap_3_r3_1 --missing    \n",
    "# output: plink.imiss and plink.lmiss, these files show respectively the proportion of missing SNPs per individual and the proportion of missing individuals per SNP."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "6be9a08e-692c-4df7-b80c-9ec3b30c5894",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "indmiss<-read.table(file=\"plink.imiss\", header=TRUE)\n",
      "snpmiss<-read.table(file=\"plink.lmiss\", header=TRUE)\n",
      "# read data into R \n",
      "\n",
      "pdf(\"histimiss.pdf\") #indicates pdf format and gives title to file\n",
      "hist(indmiss[,6],main=\"Histogram individual missingness\") #selects column 6, names header of file\n",
      "\n",
      "pdf(\"histlmiss.pdf\") \n",
      "hist(snpmiss[,5],main=\"Histogram SNP missingness\")  \n",
      "dev.off() # shuts down the current device\n",
      "pdf \n",
      "  2 \n"
     ]
    }
   ],
   "source": [
    "# Generate plots to visualize the missingness results.\n",
    "cat hist_miss.R\n",
    "Rscript --no-save hist_miss.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "fc25c3e1-a013-4f23-a073-bb265e3391a3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_2.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_1\n",
      "  --geno 0.2\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_2\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1457897 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 225 het. haploid genotypes present (see HapMap_3_r3_2.hh ); many\n",
      "commands treat these as missing.\n",
      "Total genotyping rate is 0.997378.\n",
      "0 variants removed due to missing genotype data (--geno).\n",
      "1457897 variants and 165 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (53 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_2.bed + HapMap_3_r3_2.bim + HapMap_3_r3_2.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_3.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_2\n",
      "  --make-bed\n",
      "  --mind 0.2\n",
      "  --out HapMap_3_r3_3\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1457897 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 225 het. haploid genotypes present (see HapMap_3_r3_3.hh ); many\n",
      "commands treat these as missing.\n",
      "Total genotyping rate is 0.997378.\n",
      "1457897 variants and 165 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (53 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_3.bed + HapMap_3_r3_3.bim + HapMap_3_r3_3.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_4.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_3\n",
      "  --geno 0.02\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_4\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1457897 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 225 het. haploid genotypes present (see HapMap_3_r3_4.hh ); many\n",
      "commands treat these as missing.\n",
      "Total genotyping rate is 0.997378.\n",
      "27454 variants removed due to missing genotype data (--geno).\n",
      "1430443 variants and 165 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (53 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_4.bed + HapMap_3_r3_4.bim + HapMap_3_r3_4.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_5.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_4\n",
      "  --make-bed\n",
      "  --mind 0.02\n",
      "  --out HapMap_3_r3_5\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1430443 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "0 people removed due to missing genotype data (--mind).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 179 het. haploid genotypes present (see HapMap_3_r3_5.hh ); many\n",
      "commands treat these as missing.\n",
      "Total genotyping rate is 0.997899.\n",
      "1430443 variants and 165 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (53 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_5.bed + HapMap_3_r3_5.bim + HapMap_3_r3_5.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# Delete SNPs and individuals with high levels of missingness, explanation of this and all following steps can be found in box 1 and table 1 of the article mentioned in the comments of this script.\n",
    "# The following two QC commands will not remove any SNPs or individuals. However, it is good practice to start the QC with these non-stringent thresholds.  \n",
    "# Delete SNPs with missingness >0.2.\n",
    "$plink --bfile HapMap_3_r3_1 --geno 0.2 --make-bed --out HapMap_3_r3_2\n",
    "\n",
    "# Delete individuals with missingness >0.2.\n",
    "$plink --bfile HapMap_3_r3_2 --mind 0.2 --make-bed --out HapMap_3_r3_3\n",
    "\n",
    "# Delete SNPs with missingness >0.02.\n",
    "$plink --bfile HapMap_3_r3_3 --geno 0.02 --make-bed --out HapMap_3_r3_4\n",
    "\n",
    "# Delete individuals with missingness >0.02.\n",
    "$plink --bfile HapMap_3_r3_4 --mind 0.02 --make-bed --out HapMap_3_r3_5"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9e5b53e-dac4-4f27-97dc-9a8e00e0c362",
   "metadata": {},
   "source": [
    "### Step2 ####\n",
    "\n",
    "Check for sex discrepancy.\n",
    "\n",
    "Subjects who were a priori determined as females must have a F value of <0.2, and subjects who were a priori determined as males must have a F value >0.8. This F value is based on the X chromosome inbreeding (homozygosity) estimate.\n",
    "\n",
    "Subjects who do not fulfil these requirements are flagged \"PROBLEM\" by PLINK.\n",
    "\n",
    "    plink --bfile HapMap_3_r3_5 --check-sex \n",
    "\n",
    "Generate plots to visualize the sex-check results.\n",
    "\n",
    "    Rscript --no-save gender_check.R\n",
    "\n",
    "These checks indicate that there is one woman with a sex discrepancy, F value of 0.99. (When using other datasets often a few discrepancies will be found). \n",
    "\n",
    "The following two scripts can be used to deal with individuals with a sex discrepancy.\n",
    "\n",
    "Note, please use one of the two options below to generate the bfile hapmap_r23a_6, this file we will use in the next step of this tutorial.\n",
    "\n",
    "1. Delete individuals with sex discrepancy.\n",
    "\n",
    "        grep \"PROBLEM\" plink.sexcheck| awk '{print$1,$2}'> sex_discrepancy.txt\n",
    "\n",
    "    This command generates a list of individuals with the status �PROBLEM�.\n",
    "\n",
    "        plink --bfile HapMap_3_r3_5 --remove sex_discrepancy.txt --make-bed --out HapMap_3_r3_6 \n",
    "\n",
    "    This command removes the list of individuals with the status �PROBLEM�.\n",
    "\n",
    "2. impute-sex.\n",
    "\n",
    "        plink --bfile HapMap_3_r3_5 --impute-sex --make-bed --out HapMap_3_r3_6\n",
    "   \n",
    "    This imputes the sex based on the genotype information into your data set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "e57ba06f-3b14-43b2-9791-b794866ec5b8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plink.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_5\n",
      "  --check-sex\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1430443 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 53 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 179 het. haploid genotypes present (see plink.hh ); many commands\n",
      "treat these as missing.\n",
      "Total genotyping rate is 0.997899.\n",
      "1430443 variants and 165 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (53 phenotypes\n",
      "are missing.)\n",
      "--check-sex: 23424 Xchr and 0 Ychr variant(s) scanned, 1 problem detected.\n",
      "Report written to plink.sexcheck .\n"
     ]
    }
   ],
   "source": [
    "# Check for sex discrepancy.\n",
    "# Subjects who were a priori determined as females must have a F value of <0.2, and subjects who were a priori determined as males must have a F value >0.8. This F value is based on the X chromosome inbreeding (homozygosity) estimate.\n",
    "# Subjects who do not fulfil these requirements are flagged \"PROBLEM\" by PLINK.\n",
    "$plink --bfile HapMap_3_r3_5 --check-sex "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "c501ad47-3e86-4d38-8602-59e375f56582",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gender <- read.table(\"plink.sexcheck\", header=T,as.is=T)\n",
      "\n",
      "pdf(\"Gender_check.pdf\")\n",
      "hist(gender[,6],main=\"Gender\", xlab=\"F\")\n",
      "dev.off()\n",
      "\n",
      "pdf(\"Men_check.pdf\")\n",
      "male=subset(gender, gender$PEDSEX==1)\n",
      "hist(male[,6],main=\"Men\",xlab=\"F\")\n",
      "dev.off()\n",
      "\n",
      "pdf(\"Women_check.pdf\")\n",
      "female=subset(gender, gender$PEDSEX==2)\n",
      "hist(female[,6],main=\"Women\",xlab=\"F\")\n",
      "dev.off()\n",
      "\n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n",
      "null device \n",
      "          1 \n"
     ]
    }
   ],
   "source": [
    "# Generate plots to visualize the sex-check results.\n",
    "cat gender_check.R\n",
    "Rscript --no-save gender_check.R\n",
    "# These checks indicate that there is one woman with a sex discrepancy, F value of 0.99. (When using other datasets often a few discrepancies will be found). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "ba12b07a-0401-4031-b1cf-49b9238f0202",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to HapMap_3_r3_6.log.\n",
      "Options in effect:\n",
      "  --bfile HapMap_3_r3_5\n",
      "  --make-bed\n",
      "  --out HapMap_3_r3_6\n",
      "  --remove sex_discrepancy.txt\n",
      "\n",
      "257564 MB RAM detected; reserving 128782 MB for main workspace.\n",
      "1430443 variants loaded from .bim file.\n",
      "165 people (80 males, 85 females) loaded from .fam.\n",
      "112 phenotype values loaded from .fam.\n",
      "--remove: 164 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 112 founders and 52 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Warning: 179 het. haploid genotypes present (see HapMap_3_r3_6.hh ); many\n",
      "commands treat these as missing.\n",
      "Total genotyping rate in remaining samples is 0.99798.\n",
      "1430443 variants and 164 people pass filters and QC.\n",
      "Among remaining phenotypes, 56 are cases and 56 are controls.  (52 phenotypes\n",
      "are missing.)\n",
      "--make-bed to HapMap_3_r3_6.bed + HapMap_3_r3_6.bim + HapMap_3_r3_6.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "# The following two scripts can be used to deal with individuals with a sex discrepancy.\n",
    "# Note, please use one of the two options below to generate the bfile hapmap_r23a_6, this file we will use in the next step of this tutorial.\n",
    "\n",
    "# 1) Delete individuals with sex discrepancy.\n",
    "grep \"PROBLEM\" plink.sexcheck| awk '{print$1,$2}'> sex_discrepancy.txt\n",
    "# This command generates a list of individuals with the status �PROBLEM�.\n",
    "$plink --bfile HapMap_3_r3_5 --remove sex_discrepancy.txt --make-bed --out HapMap_3_r3_6 \n",
    "# This command removes the list of individuals with the status �PROBLEM�.\n",
    "\n",
    "# 2) impute-sex.\n",
    "#$plink --bfile HapMap_3_r3_5 --impute-sex --make-bed --out HapMap_3_r3_6\n",
    "# This imputes the sex based on the genotype information into your data set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1567fd5-7612-4e88-801a-fe7699d090ad",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
